<div class="container">

<table style="width: 100%;"><tr>
<td>doACNE</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>(ACNE)</h2>

<h3>Description</h3>

<p>(ACNE) based on [1].
The algorithm is processed in bounded memory, meaning virtually
any number of arrays can be analyzed on also very limited computer
systems.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  ## S3 method for class 'AffymetrixCelSet'
doACNE(csR, fln=FALSE, drop=TRUE, verbose=FALSE, ...)
  ## Default S3 method:
doACNE(dataSet, ..., verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>csR, dataSet</code></td>
<td>
<p>An <code>AffymetrixCelSet</code> (or the name of an
<code>AffymetrixCelSet</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fln</code></td>
<td>
<p>If <code>TRUE</code>, CRMAv2-style PCR fragment-length normalization
is performed, otherwise not.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop</code></td>
<td>
<p>If <code>TRUE</code>, the RMA summaries are returned, otherwise
a named <code>list</code> of all intermediate and final results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>See <code>Verbose</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments used to set up <code>AffymetrixCelSet</code>
(when argument <code>dataSet</code> is specified).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a named <code>list</code>, iff <code>drop == FALSE</code>, otherwise
a named <code>list</code> of <code>AromaUnitTotalCnBinarySet</code>
and <code>AromaUnitFracBCnBinarySet</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] M. Ortiz-Estevez, H. Bengtsson, A. Rubio, <em>ACNE: a summarization method to estimate allele-specific copy numbers for Affymetrix SNP arrays</em>, Bioinformatics, 2010 [PMC2913655].
<br></p>


</div>