<div class="container">

<table style="width: 100%;"><tr>
<td>NmfPlm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The NmfPlm class</h2>

<h3>Description</h3>

<p>Package:  ACNE <br><b>Class NmfPlm</b><br></p>
<p><code>Object</code><br><code>~~|</code><br><code>~~+--</code><code>ParametersInterface</code><br><code>~~~~~~~|</code><br><code>~~~~~~~+--</code><code>Model</code><br><code>~~~~~~~~~~~~|</code><br><code>~~~~~~~~~~~~+--</code><code>UnitModel</code><br><code>~~~~~~~~~~~~~~~~~|</code><br><code>~~~~~~~~~~~~~~~~~+--</code><code>MultiArrayUnitModel</code><br><code>~~~~~~~~~~~~~~~~~~~~~~|</code><br><code>~~~~~~~~~~~~~~~~~~~~~~+--</code><code>ProbeLevelModel</code><br><code>~~~~~~~~~~~~~~~~~~~~~~~~~~~|</code><br><code>~~~~~~~~~~~~~~~~~~~~~~~~~~~+--</code><em><code>NmfPlm</code></em><br></p>
<p><b>Directly known subclasses:</b><br><em>NmfSnpPlm</em><br></p>
<p>public abstract static class <b>NmfPlm</b><br>
extends <em>ProbeLevelModel</em><br></p>
<p>This class represents the NMF model of [REF].
</p>


<h3>Usage</h3>

<pre><code class="language-R">NmfPlm(..., maxIter=10L, maxIterRlm=20L, refs=NULL, flavor=c("v4", "v3", "v2", "v1"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments passed to <code>ProbeLevelModel</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxIter</code></td>
<td>
<p>The maximum number of iteration in the NMF step.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxIterRlm</code></td>
<td>
<p>A positive <code>integer</code> specifying the maximum number of
iterations used in rlm.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refs</code></td>
<td>
<p>An index <code>vector</code> (<code>integer</code> or <code>logical</code>) specifying the
reference samples.  If <code>NULL</code>, all samples are used as a reference.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flavor</code></td>
<td>
<p>(Internal/developmental only)
A <code>character</code> string specifying which algorithm to use.</p>
</td>
</tr>
</table>
<h3>Fields and Methods</h3>

<p><b>Methods:</b><br></p>

<table>
<tr>
<td style="text-align: right;">
 </td>
<td style="text-align: left;"> <code>getAsteriskTags</code> </td>
<td style="text-align: left;">  -</td>
</tr>
<tr>
<td style="text-align: right;">
</td>
</tr>
</table>
<p><b>Methods inherited from ProbeLevelModel</b>:<br>
calculateResidualSet, calculateWeights, fit, getAsteriskTags, getCalculateResidualsFunction, getChipEffectSet, getProbeAffinityFile, getResidualSet, getRootPath, getWeightsSet
</p>
<p><b>Methods inherited from MultiArrayUnitModel</b>:<br>
getListOfPriors, setListOfPriors, validate
</p>
<p><b>Methods inherited from UnitModel</b>:<br>
findUnitsTodo, getAsteriskTags, getFitSingleCellUnitFunction, getParameters
</p>
<p><b>Methods inherited from Model</b>:<br>
as.character, fit, getAlias, getAsteriskTags, getDataSet, getFullName, getName, getPath, getRootPath, getTags, setAlias, setTags
</p>
<p><b>Methods inherited from ParametersInterface</b>:<br>
getParameterSets, getParameters, getParametersAsString
</p>
<p><b>Methods inherited from Object</b>:<br>
$, $&lt;-, [[, [[&lt;-, as.character, attach, attachLocally, clearCache, clearLookupCache, clone, detach, equals, extend, finalize, getEnvironment, getFieldModifier, getFieldModifiers, getFields, getInstantiationTime, getStaticInstance, hasField, hashCode, ll, load, names, objectSize, print, save, asThis
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] M. Ortiz-Estevez, H. Bengtsson, A. Rubio, <em>ACNE: a summarization method to estimate allele-specific copy numbers for Affymetrix SNP arrays</em>, Bioinformatics, 2010 [PMC2913655].
<br></p>


<h3>See Also</h3>

<p>Internally, for each SNP the NMF model is fitted using the
<code>fitSnpNmf</code>() function.
</p>


</div>