<div class="container">

<table style="width: 100%;"><tr>
<td>GSE29172_H1395</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Annotated copy-number regions from the GEO GSE29172 (and GSE26302) data sets.</h2>

<h3>Description</h3>

<p>The GEO GSE29172 data set is a dilution series from the Affymetrix 
GenomeWideSNP_6 chip type. The GEO GSE26302 data set contains the experiment 
corresponding to the matched normal (i.e. 0% dilution).
</p>


<h3>Format</h3>

<p>A data frame with 770668 observations of 7 variables: </p>
 
<dl>
<dt>c</dt>
<dd>
<p>total copy number (not log-scaled)</p>
</dd> <dt>b</dt>
<dd>
<p>allelic ratios in the
diluted tumor sample (after TumorBoost)</p>
</dd> <dt>genotype</dt>
<dd>
<p>germline 
genotypes</p>
</dd> <dt>bT</dt>
<dd>
<p>allelic ratios in the diluted tumor sample (before 
TumorBoost)</p>
</dd> <dt>bN</dt>
<dd>
<p>allelic ratios in the matched normal sample</p>
</dd> 
<dt>region</dt>
<dd>
<p>a character value, annotation label for the region. Should be
encoded as <code>"(C1,C2)"</code>, where <code>C1</code> denotes the minor copy number 
and <code>C2</code> denotes the major copy number.  For example, </p>
 
<dl>
<dt>(1,1)</dt>
<dd>
<p>Normal</p>
</dd> <dt>(0,1)</dt>
<dd>
<p>Hemizygous deletion</p>
</dd> 
<dt>(0,0)</dt>
<dd>
<p>Homozygous deletion</p>
</dd> <dt>(1,2)</dt>
<dd>
<p>Single copy gain</p>
</dd> 
<dt>(0,2)</dt>
<dd>
<p>Copy-neutral LOH</p>
</dd> <dt>(2,2)</dt>
<dd>
<p>Balanced two-copy gain</p>
</dd> 
<dt>(1,3)</dt>
<dd>
<p>Unbalanced two-copy gain</p>
</dd> <dt>(0,3)</dt>
<dd>
<p>Single-copy gain with 
LOH</p>
</dd> </dl>
</dd> <dt>genotype</dt>
<dd>
<p>the (germline) genotype of SNPs. By definition, rows 
with missing genotypes are interpreted as non-polymorphic loci (a.k.a. copy
number probes).</p>
</dd> <dt>cellularity</dt>
<dd>
<p>A numeric value between 0 and 1, the
percentage of tumor cells in the  sample.</p>
</dd> </dl>
<h3>Details</h3>

<p>These data have been processed from the files available from GEO using 
scripts that are included in the 'inst/preprocessing/GSE29172' directory of
this package. This processing includes normalization of the raw CEL files
using the CRMAv2 method implemented in the aroma.affymetrix package.
</p>


<h3>Source</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE29172">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE29172</a> 
<a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE26302">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE26302</a>
</p>


<h3>References</h3>

<p>Rasmussen, M., Sundstr\"om, M., Kultima, H. G., Botling, J., 
Micke, P., Birgisson, H., Glimelius, B. &amp; Isaksson, A. (2011). 
Allele-specific copy number analysis of tumor samples with aneuploidy and 
tumor heterogeneity. Genome Biology, 12(10), R108.#'
</p>
<p>Bengtsson, H., Wirapati , P. &amp; Speed, T.P. (2009). A single-array
preprocessing method for estimating full-resolution raw copy numbers from 
all Affymetrix genotyping arrays including GenomeWideSNP 5 &amp; 6, 
Bioinformatics 25(17), pp. 2149-56.
</p>
<p>Bengtsson H., Neuvial, P. and Speed, T. P. (2010) TumorBoost:
normalization of allele-specific tumor copy numbers from a single pair of
tumor-normal genotyping microarrays. BMC bioinformatics 11 (2010), p. 245.
</p>


<h3>Examples</h3>

<pre><code class="language-R">dat &lt;- loadCnRegionData("GSE29172_H1395")
unique(dat$region)
</code></pre>


</div>