<div class="container">

<table style="width: 100%;"><tr>
<td>loadCnRegionData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>loadCnRegionData</h2>

<h3>Description</h3>

<p>Load real, annotated copy number data
</p>


<h3>Usage</h3>

<pre><code class="language-R">loadCnRegionData(dataSet, tumorFraction = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataSet</code></td>
<td>
<p>name of one of the data sets of the package, see
<code>listDataSets</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tumorFraction</code></td>
<td>
<p>proportion of tumor cells in the "tumor" sample (a.k.a.
tumor cellularity). See <code>listTumorFractions</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is a wrapper to load real genotyping array data taken from
</p>
<p>* a dilution series from the Affymetrix GenomeWideSNP_6 chip type (Rasmussen
et al, 2011), see <code>GSE29172_H1395</code> * a dilution series from the
Illumina HumanCNV370v1 chip type (Staaf et al, 2008), see
<code>GSE11976_CRL2324</code> * a tumor/normal pair from the Affymetrix
GenomeWideSNP_6 chip type (Chiang et al, 2008), see
<code>GSE13372_HCC1143</code>
</p>


<h3>Value</h3>

<p>a data.frame containing copy number data for different types of copy
number regions.  Columns:</p>
 <dl>
<dt>c</dt>
<dd>
<p>Total copy number</p>
</dd>
<dt>b</dt>
<dd>
<p>Allele B fraction (a.k.a. BAF)</p>
</dd> <dt>region</dt>
<dd>
<p>a character value,
annotation label for the region.  Should be encoded as <code>"(C1,C2)"</code>,
where <code>C1</code> denotes the minor copy number and <code>C2</code> denotes the
major copy number.  For example, </p>
 <dl>
<dt>(1,1)</dt>
<dd>
<p>Normal</p>
</dd>
<dt>(0,1)</dt>
<dd>
<p>Hemizygous deletion</p>
</dd> <dt>(0,0)</dt>
<dd>
<p>Homozygous deletion</p>
</dd>
<dt>(1,2)</dt>
<dd>
<p>Single copy gain</p>
</dd> <dt>(0,2)</dt>
<dd>
<p>Copy-neutral LOH</p>
</dd>
<dt>(2,2)</dt>
<dd>
<p>Balanced two-copy gain</p>
</dd> <dt>(1,3)</dt>
<dd>
<p>Unbalanced two-copy gain</p>
</dd>
<dt>(0,3)</dt>
<dd>
<p>Single-copy gain with LOH</p>
</dd> </dl>
</dd> <dt>muN</dt>
<dd>
<p>the (germline)
genotype of SNPs. By definition, rows with missing genotypes are
interpreted as non-polymorphic loci (a.k.a. copy number probes).</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Morgane Pierre-Jean and Pierre Neuvial
</p>


<h3>Examples</h3>

<pre><code class="language-R">
affyDat &lt;- loadCnRegionData(dataSet="GSE29172_H1395", tumorFraction=1)
str(affyDat)

illuDat &lt;- loadCnRegionData(dataSet="GSE11976_CRL2324", tumorFraction=.79)
str(illuDat)

affyDat2 &lt;- loadCnRegionData(dataSet="GSE13372_HCC1143", tumorFraction=1)
str(affyDat2)

</code></pre>


</div>