<div class="container">

<table style="width: 100%;"><tr>
<td>spillToConvergence</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Spillover to convergence</h2>

<h3>Description</h3>

<p>Build an n-pass spillover matrix, continuing until the results converge into clusters of cell types
</p>
<p>deconMatrices &lt;- spillToConvergence(sigMatrix, geneExpr, 100, FALSE, TRUE)
</p>


<h3>Usage</h3>

<pre><code class="language-R">spillToConvergence(
  sigMatrix,
  geneExpr,
  nPasses = 100,
  plotIt = FALSE,
  imputNAs = FALSE,
  method = "DCQ"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sigMatrix</code></td>
<td>
<p>The deconvolution matrix, e.g. LM22 or MGSM27</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneExpr</code></td>
<td>
<p>The source gene expression matrix used to calculate sigMatrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPasses</code></td>
<td>
<p>The maximum number of iterations (DEFAULT: 100)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotIt</code></td>
<td>
<p>Set to TRUE to plot it (DEFAULT: FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imputNAs</code></td>
<td>
<p>Set to TRUE to imput genes with missing values &amp; cache the imputed.  FALSE will just remove them (DEFAULT: FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>One of 'DCQ', 'SVMDECON', 'DeconRNASeq', 'proportionsInAdmixture', 'nnls' (DEFAULT: DCQ)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of signature matrices
</p>


<h3>Examples</h3>

<pre><code class="language-R">#This toy example 
library(ADAPTS)
fullLM22 &lt;- ADAPTS::LM22[1:30, 1:4]
smallLM22 &lt;- fullLM22[1:25,] 

deconMatrices &lt;- spillToConvergence(sigMatrix=smallLM22, geneExpr=fullLM22, nPasses=10, plotIt=TRUE)
</code></pre>


</div>