<div class="container">

<table style="width: 100%;"><tr>
<td>rankByT</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Rank genes for each cell type</h2>

<h3>Description</h3>

<p>Use a t-test to rank to features for each cell type
</p>
<p>gList &lt;- rankByT(geneExpr, qCut=0.3)
</p>


<h3>Usage</h3>

<pre><code class="language-R">rankByT(
  geneExpr,
  qCut = 0.3,
  oneCore = FALSE,
  secondPval = TRUE,
  remZinf = FALSE,
  reqRatGT1 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>geneExpr</code></td>
<td>
<p>The gene expression data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qCut</code></td>
<td>
<p>(DEFAULT: 0.3)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>oneCore</code></td>
<td>
<p>Set to TRUE to disable paralellization (DEFAULT: FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>secondPval</code></td>
<td>
<p>Set to TRUE to use p-Values as a second sort criteria (DEFAULT: TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remZinf</code></td>
<td>
<p>Set to TRUE to remove any ratio with zero or infinity.  Good for scRNAseq. (DEFAULT: FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reqRatGT1</code></td>
<td>
<p>Set to TRUE to remove any gene with a ratio with less than 1.  Good for scRNAseq. (DEFAULT: FALSE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list of cell types with data frames ranking genes
</p>


<h3>Examples</h3>

<pre><code class="language-R">#This toy example treats the LM22 deconvolution matrix as if it were all of the data
#  For a real example, look at the vignette or comments in exprData, fullLM22, small LM22
library(ADAPTS)
fullLM22 &lt;- ADAPTS::LM22[1:200, 1:8]
#Make a fake signature matrix out of 100 genes and the first 8 cell types
smallLM22 &lt;- fullLM22[1:100, 1:8] 

#Make fake data representing two replicates of purified Mast.cells 
exprData &lt;- ADAPTS::LM22[1:200, c("Mast.cells.resting","Mast.cells.activated")]
colnames(exprData) &lt;- c("Mast.cells", "Mast.cells")

#Fake source data with replicates for all purified cell types.
#  Note in this fake data set, many cell types have exactly one replicate
fakeAllData &lt;- cbind(fullLM22, as.data.frame(exprData)) 
gList &lt;- rankByT(geneExpr = fakeAllData, qCut=0.3, oneCore=TRUE, reqRatGT1=FALSE)
</code></pre>


</div>