<div class="container">

<table style="width: 100%;"><tr>
<td>hierarchicalSplit</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Build hierarchical cell clusters.</h2>

<h3>Description</h3>

<p>Attempt to deconvolve cell types by building a hierarchy of cell types using
spillToConvergence to determine cell types that are not signficantly different.
First deconvolve those clusters of cell types.
Deconvolution matrices are then built to separate the cell types that formerly could
not be separated.
</p>


<h3>Usage</h3>

<pre><code class="language-R">hierarchicalSplit(
  sigMatrix,
  geneExpr,
  oneCore = FALSE,
  nPasses = 100,
  deconMatrices = NULL,
  remZinf = TRUE,
  method = "DCQ",
  useRF = TRUE,
  incNonCluster = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sigMatrix</code></td>
<td>
<p>The deconvolution matrix, e.g. LM22 or MGSM27</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneExpr</code></td>
<td>
<p>The source gene expression matrix used to calculate sigMatrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>oneCore</code></td>
<td>
<p>Set to TRUE to disable parallelization (DEFAULT: FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPasses</code></td>
<td>
<p>The maximum number of iterations for spillToConvergence (DEFAULT: 100)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deconMatrices</code></td>
<td>
<p>Optional pre-computed results from spillToConvergence (DEFAULT: NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remZinf</code></td>
<td>
<p>Set to TRUE to remove any ratio with zero or infinity when generating gList (DEFAULT: FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>One of 'DCQ', 'SVMDECON', 'DeconRNASeq', 'proportionsInAdmixture', 'nnls' (DEFAULT: DCQ)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useRF</code></td>
<td>
<p>Set to TRUE to use ranger random forests to build the seed matrix (DEFAULT: TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>incNonCluster</code></td>
<td>
<p>Set to TRUE to include a 'nonCluster' in each of the sub matrices (DEFAULT: TRUE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of clusters and a list of signature matrices for breaking those clusters
</p>


<h3>Examples</h3>

<pre><code class="language-R">#This toy example 
library(ADAPTS)
fullLM22 &lt;- ADAPTS::LM22[1:30, 1:4]
smallLM22 &lt;- fullLM22[1:25,] 

clusters &lt;- hierarchicalSplit(sigMatrix=smallLM22, geneExpr=fullLM22, oneCore=TRUE, nPasses=10,
    deconMatrices=NULL, remZinf=TRUE, method='DCQ', useRF=TRUE, incNonCluster=TRUE)
</code></pre>


</div>