<div class="container">

<table style="width: 100%;"><tr>
<td>hierarchicalClassify</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Hierarchical Deconvolution</h2>

<h3>Description</h3>

<p>Deconvolve cell types based on clusters detected by an n-pass spillover matrix
</p>


<h3>Usage</h3>

<pre><code class="language-R">hierarchicalClassify(
  sigMatrix,
  geneExpr,
  toPred,
  hierarchData = NULL,
  pdfDir = tempdir(),
  oneCore = FALSE,
  nPasses = 100,
  remZinf = TRUE,
  method = "DCQ",
  useRF = TRUE,
  incNonCluster = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sigMatrix</code></td>
<td>
<p>The deconvolution matrix, e.g. LM22 or MGSM27</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneExpr</code></td>
<td>
<p>The source gene expression matrix used to calculate sigMatrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>toPred</code></td>
<td>
<p>The gene expression to ultimately deconvolve</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hierarchData</code></td>
<td>
<p>The results of hierarchicalSplit OR hierarchicalSplit.sc (DEFAULT: NULL, ie hierarchicalSplit)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdfDir</code></td>
<td>
<p>A fold to write the pdf file to (DEFAULT: tempdir())</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>oneCore</code></td>
<td>
<p>Set to TRUE to disable parallelization (DEFAULT: FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPasses</code></td>
<td>
<p>The maximum number of iterations for spillToConvergence (DEFAULT: 100)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remZinf</code></td>
<td>
<p>Set to TRUE to remove any ratio with zero or infinity when generating gList (DEFAULT: FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>One of 'DCQ', 'SVMDECON', 'DeconRNASeq', 'proportionsInAdmixture', 'nnls' (DEFAULT: DCQ)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useRF</code></td>
<td>
<p>Set to TRUE to use ranger random forests to build the seed matrix (DEFAULT: TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>incNonCluster</code></td>
<td>
<p>Set to TRUE to include a 'nonCluster' in each of the sub matrices (DEFAULT: TRUE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a matrix of cell counts
</p>


<h3>Examples</h3>

<pre><code class="language-R">#This toy example 
library(ADAPTS)
fullLM22 &lt;- ADAPTS::LM22[1:30, 1:4]
smallLM22 &lt;- fullLM22[1:25,] 

cellCounts &lt;- hierarchicalClassify(sigMatrix=smallLM22, geneExpr=fullLM22, toPred=fullLM22, 
    oneCore=TRUE, nPasses=10, method='DCQ')
</code></pre>


</div>