<div class="container">

<table style="width: 100%;"><tr>
<td>clustWspillOver</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Cluster with spillover</h2>

<h3>Description</h3>

<p>Build clusters based on n-pass spillover matrix
</p>


<h3>Usage</h3>

<pre><code class="language-R">clustWspillOver(
  sigMatrix,
  geneExpr,
  nPasses = 100,
  deconMatrices = NULL,
  method = "DCQ"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sigMatrix</code></td>
<td>
<p>The deconvolution matrix, e.g. LM22 or MGSM27</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneExpr</code></td>
<td>
<p>The source gene expression matrix used to calculate sigMatrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPasses</code></td>
<td>
<p>The maximum number of iterations for spillToConvergence (DEFAULT: 100)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deconMatrices</code></td>
<td>
<p>Optional pre-computed results from spillToConvergence (DEFAULT: NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>One of 'DCQ', 'SVMDECON', 'DeconRNASeq', 'proportionsInAdmixture', 'nnls' (DEFAULT: DCQ)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Cell types grouped by cluster
</p>


<h3>Examples</h3>

<pre><code class="language-R">#This toy example 
library(ADAPTS)
fullLM22 &lt;- ADAPTS::LM22[1:30, 1:4]
smallLM22 &lt;- fullLM22[1:25,] 

clusters &lt;- clustWspillOver(sigMatrix=smallLM22, geneExpr=fullLM22, nPasses=10)
</code></pre>


</div>