<div class="container">

<table style="width: 100%;"><tr>
<td>buildSpilloverMat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Build a spillover matrix</h2>

<h3>Description</h3>

<p>Build a spillover matrix, i.e. what do purified samples deconvolve as?
</p>
<p>spillExpr &lt;- buildSpilloverMat(refExpr, geneExpr, method='DCQ')
</p>


<h3>Usage</h3>

<pre><code class="language-R">buildSpilloverMat(refExpr, geneExpr, method = "DCQ")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>refExpr</code></td>
<td>
<p>The deconvolution matrix, e.g. LM22 or MGSM27</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneExpr</code></td>
<td>
<p>The full gene expression for purified cell types.  Multiple columns (examples) for each column in the reference expr.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>One of 'DCQ', 'SVMDECON', 'DeconRNASeq', 'proportionsInAdmixture', 'nnls' (DEFAULT: DCQ)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A spillover matrix showing how purified cell types deconvolve
</p>


<h3>Examples</h3>

<pre><code class="language-R">#This toy example 
library(ADAPTS)
fullLM22 &lt;- ADAPTS::LM22[1:30, 1:4]
smallLM22 &lt;- fullLM22[1:25,] 

spillover &lt;- buildSpilloverMat(refExpr=smallLM22, geneExpr=fullLM22, method='DCQ')
</code></pre>


</div>