<div class="container">

<table style="width: 100%;"><tr>
<td>aminoacyl</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Codon usage</h2>

<h3>Description</h3>

<p><code>aminoacyl</code> is a list containing the codon counts of 36 genes encoding yeast aminoacyl-tRNA-synthetase(S.Cerevisiae).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(aminoacyl)</code></pre>


<h3>Format</h3>

<p><code>aminoacyl</code> is a list containing the 5 following objects:  
</p>
     
<dl>
<dt>genes</dt>
<dd>
<p>is a vector giving the gene names.</p>
</dd> 
<dt>localisation</dt>
<dd>
<p>is a vector giving the cellular localisation
of the proteins (M = mitochondrial, C = cytoplasmic,
I = indetermined, CI = cyto and mito).</p>
</dd>    
<dt>codon</dt>
<dd>
<p>is a vector containing the 64 triplets.</p>
</dd>
<dt>AA</dt>
<dd>
<p>is a factor giving the amino acid names for each codon.</p>
</dd>
<dt>usage.codon</dt>
<dd>
<p>is a dataframe containing the codon counts for each gene.</p>
</dd>
</dl>
<h3>Source</h3>

<p>Data prepared by D. Charif <a href="mailto:Delphine.Charif@versailles.inra.fr">Delphine.Charif@versailles.inra.fr</a> starting from:<br><a href="http://www.expasy.org/sprot/">http://www.expasy.org/sprot/</a>
</p>


<h3>References</h3>

<p>Chiapello H., Olivier E., Landes-Devauchelle C., Nitschké P. and Risler J.L (1999) 
Codon usage as a tool to predict the cellular localisation of eukariotic ribosomal 
proteins and aminoacyl-tRNA synthetases. <em>Nucleic Acids Res.</em>, <b>27</b>, 14, 2848–2851.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(aminoacyl)
aminoacyl$genes
aminoacyl$usage.codon
dudi.coa(aminoacyl$usage.codon, scannf = FALSE)
</code></pre>


</div>