<div class="container">

<table style="width: 100%;"><tr>
<td>extract.PLINKmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reading PLINK Single Nucleotide Polymorphism data</h2>

<h3>Description</h3>

<p>The function <code>read.PLINK</code> reads a data file exported by the PLINK
software with extension '.raw' and converts it into a <code>"genlight"</code>
object. Optionally, information about SNPs can be read from a ".map" file,
either by specifying the argument <code>map.file</code> in <code>read.PLINK</code>, or
using <code>extract.PLINKmap</code> to add information to an existing
<code>"genlight"</code> object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extract.PLINKmap(file, x = NULL)

read.PLINK(
  file,
  map.file = NULL,
  quiet = FALSE,
  chunkSize = 1000,
  parallel = FALSE,
  n.cores = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>for <code>read.PLINK</code> a character string giving the path to the
file to convert, with the extension ".raw"; for <code>extract.PLINKmap</code>, a
character string giving the path to a file with extension ".map".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an optional object of the class <code>"genlight"</code>, in which
the information read is stored; if provided, information is matched against
the names of the loci in <code>x</code>, as returned by <code>locNames(x)</code>; if not
provided, a list of two components is returned, containing chromosome and
position information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.file</code></td>
<td>
<p>an optional character string indicating the path to a ".map"
file, which contains information about the SNPs (chromosome, position). If
provided, this information is processed by <code>extract.PLINKmap</code> and
stored in the <code>@other</code> slot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>logical stating whether a conversion messages should be printed
(TRUE,default) or not (FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chunkSize</code></td>
<td>
<p>an integer indicating the number of genomes to be read at a
time; larger values require more RAM but decrease the time needed to read
the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>a logical indicating whether multiple cores -if available-
should be used for the computations (TRUE, default), or not (FALSE);
requires the package <code>parallel</code> to be installed (see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p>if <code>parallel</code> is TRUE, the number of cores to be used in
the computations; if NULL, then the maximum number of cores available on the
computer is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other arguments to be passed to other functions - currently not
used.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function reads data by chunks of several genomes (minimum 1, no maximum)
at a time, which allows one to read massive datasets with negligible RAM
requirements (albeit at a cost of computational time). The argument
<code>chunkSize</code> indicates the number of genomes read at a time. Increasing
this value decreases the computational time required to read data in, while
increasing memory requirements.
</p>
<p>See details for the documentation about how to export data using PLINK to
the '.raw' format.
</p>
<p>=== Exporting data from PLINK ===
</p>
<p>Data need to be exported from PLINK using the option "–recodeA" (and NOT
"–recodeAD"). The PLINK command should therefore look like: <code>plink
--file data --recodeA</code>. For more information on this topic, please look at
this webpage: <a href="http://zzz.bwh.harvard.edu/plink/">http://zzz.bwh.harvard.edu/plink/</a>
</p>


<h3>Value</h3>

<p>- read.PLINK: an object of the class <code>"genlight"</code>
</p>
<p>- extract.PLINKmap: if a <code>"genlight"</code> is provided as argument
<code>x</code>, this object incorporating the new information about SNPs in the
<code>@other</code> slot (with new components 'chromosome' and 'position');
otherwise, a list with two components containing chromosome and position
information.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>

<p>- <code>?genlight</code> for a description of the class
<code>"genlight"</code>.
</p>
<p>- <code>read.snp</code>: read SNPs in adegenet's '.snp' format.
</p>
<p>- <code>fasta2genlight</code>: extract SNPs from alignments with fasta
format.
</p>
<p>- other import function in adegenet: <code>import2genind</code>,
<code>df2genind</code>, <code>read.genetix</code>
<code>read.fstat</code>, <code>read.structure</code>,
<code>read.genepop</code>.
</p>
<p>- another function <code>read.plink</code> is available in the package
<code>snpMatrix</code>.
</p>


</div>