<div class="container">

<table style="width: 100%;"><tr>
<td>H3N2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Seasonal influenza (H3N2) HA segment data</h2>

<h3>Description</h3>

<p>The dataset <code>H3N2</code> consists of 1903 strains of seasonal influenza
(H3N2) distributed worldwide, and typed at 125 SNPs located in the
hemagglutinin (HA) segment. It is stored as an R object with class
genind and can be accessed as usual using <code>data(H3N2)</code>
(see example). These data were gathered from DNA sequences available from
Genbank (http://www.ncbi.nlm.nih.gov/Genbank/).
</p>


<h3>Format</h3>

<p><code>H3N2</code> is a genind object with several data frame as
supplementary components (<code>H3N2@other) slort</code>, which contains the
following items: </p>
 <dl>
<dt>x</dt>
<dd>
<p>a <code>data.frame</code> containing
miscellaneous annotations of the sequences.</p>
</dd> <dt>xy</dt>
<dd>
<p>a matrix with two
columns indicating the geographic coordinates of the strains, as longitudes
and latitudes.</p>
</dd> <dt>epid</dt>
<dd>
<p>a character vector indicating the epidemic of
the strains.</p>
</dd> </dl>
<h3>Details</h3>

<p>The data file <code>usflu.fasta</code> is a toy dataset also gathered from
Genbank, consisting of the aligned sequences of 80 seasonal influenza
isolates (HA segment) sampled in the US, in <code>fasta</code> format. This file
is installed alongside the package; the path to this file is automatically
determined by R using <code>system.file</code> (see example in this manpage and in
?fasta2genlight) as well.
</p>


<h3>Source</h3>

<p>This dataset was prepared by Thibaut Jombart
(t.jombart@imperia.ac.uk), from annotated sequences available on Genbank
(http://www.ncbi.nlm.nih.gov/Genbank/).
</p>


<h3>References</h3>

<p>Jombart, T., Devillard, S. and Balloux, F. Discriminant analysis
of principal components: a new method for the analysis of genetically
structured populations. Submitted to <em>BMC genetics</em>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 
#### H3N2 ####
## LOAD DATA
data(H3N2)
H3N2

## set population to yearly epidemics
pop(H3N2) &lt;- factor(H3N2$other$epid)



## PERFORM DAPC - USE POPULATIONS AS CLUSTERS
## to reproduce exactly analyses from the paper, use "n.pca=1000"
dapc1 &lt;- dapc(H3N2, all.contrib=TRUE, scale=FALSE, n.pca=150, n.da=5)
dapc1

## (see ?dapc for details about the output)


## SCREEPLOT OF EIGENVALUES
barplot(dapc1$eig, main="H3N2 - DAPC eigenvalues")


## SCATTERPLOT (axes 1-2)
scatter(dapc1, posi.da="topleft", cstar=FALSE, cex=2, pch=17:22,
solid=.5, bg="white")




#### usflu.fasta ####
myPath &lt;- system.file("files/usflu.fasta",package="adegenet")
myPath

## extract SNPs from alignments using fasta2genlight
## see ?fasta2genlight for more details
obj &lt;- fasta2genlight(myPath, chunk=10) # process 10 sequences at a time
obj

## End(Not run)

</code></pre>


</div>