<div class="container">

<table style="width: 100%;"><tr>
<td>snpposi</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Analyse the position of polymorphic sites</h2>

<h3>Description</h3>

<p>These functions are used to describe the distribution of polymorphic
sites (SNPs) in an alignment.
</p>
<p>The function <code>snpposi.plot</code> plots the positions and density of
SNPs in the alignment.
</p>
<p>The function <code>snpposi.test</code> tests whether SNPs are randomly
distributed in the genome, the alternative hypothesis being that they
are clustered. This test is based on the distances of each SNP to the
closest SNP. This provides one measure of clustering for each SNP.
Different statistics can be used to summarise these values (argument
<code>stat</code>), but by default the statistics used is the median.
</p>
<p><code>snpposi.plot</code> and <code>snpposi.test</code> are generic functions with
methods for vectors of integers or numeric (indicating SNP position),
and for <code>DNAbin</code> objects.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snpposi.plot(...)

## S3 method for class 'integer'
snpposi.plot(x, genome.size, smooth=0.1,
              col="royalblue", alpha=.2, codon=TRUE, start.at=1, ...)

## S3 method for class 'numeric'
snpposi.plot(x, ...)

## S3 method for class 'DNAbin'
snpposi.plot(x, ...)


snpposi.test(...)

## S3 method for class 'integer'
snpposi.test(x, genome.size, n.sim=999, stat=median, ...)

## S3 method for class 'numeric'
snpposi.test(x, ...)

## S3 method for class 'DNAbin'
snpposi.test(x, ...)



</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a vector of integers or numerics containing SNP positions, or
a set of aligned sequences in a <code>DNAbin</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome.size</code></td>
<td>
<p>an integer indicating the length of genomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>smooth</code></td>
<td>
<p>a smoothing parameter for the density estimation;
smaller values will give more local peaks; values have to be
positive but can be less than 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>the color to be used for the plot; ignored if codon
positions are represented.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>the alpha level to be used for transparency (density curve).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>codon</code></td>
<td>
<p>a logical indicating if codon position should be
indicated (TRUE, default) or not.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start.at</code></td>
<td>
<p>an integer indicating at which base of a codon the
alignment starts (defaults to 1); values other than 1, 2 and 3 will
be ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.sim</code></td>
<td>
<p>an integer indicating the number of randomizations to be
used in the Monte Carlo test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat</code></td>
<td>
<p>a function used to summarize the measure of physical
proximity between SNPs; by default, the median is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to the <code>integer</code>
method.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A Monte Carlo test of the class <code>randtest</code>.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>.
</p>


<h3>See Also</h3>

<p>The <code>fasta2DNAbin</code> to read fasta alignments with minimum
RAM use.
</p>


<h3>Examples</h3>

<pre><code class="language-R">if(require(ape)){
data(woodmouse)
snpposi.plot(woodmouse, codon=FALSE)
snpposi.plot(woodmouse)

## Not run: 
snpposi.test(c(1,3,4,5), 100)
snpposi.test(woodmouse)

## End(Not run)
}
</code></pre>


</div>