<div class="container">

<table style="width: 100%;"><tr>
<td>genlight auxiliary functions</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Auxiliary functions for genlight objects</h2>

<h3>Description</h3>

<p>These functions provide facilities for usual computations using
genlight objects. When ploidy varies across individuals,
the outputs of these functions depend on whether the information
units are individuals, or alleles within individuals (see details).
</p>
<p>These functions are:
</p>
<p>- <code>glSum</code>: computes the sum of the number of second allele in each SNP.
</p>
<p>- <code>glNA</code>: computes the number of missing values in each SNP.
</p>
<p>- <code>glMean</code>: computes the mean number of second allele in each SNP.
</p>
<p>- <code>glVar</code>: computes the variance of the number of second allele in each SNP.
</p>
<p>- <code>glDotProd</code>: computes dot products between (possibly
centred/scaled) vectors of individuals - uses compiled C code - used
by glPca.
</p>


<h3>Usage</h3>

<pre><code class="language-R">glSum(x, alleleAsUnit = TRUE, useC = FALSE)
glNA(x, alleleAsUnit = TRUE)
glMean(x, alleleAsUnit = TRUE)
glVar(x, alleleAsUnit = TRUE)
glDotProd(x, center = FALSE, scale = FALSE, alleleAsUnit = FALSE,
                parallel = FALSE, n.cores = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a genlight object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleleAsUnit</code></td>
<td>
<p>a logical indicating whether alleles are
considered as units (i.e., a diploid genotype equals two samples, a
triploid, three, etc.) or whether individuals are considered as
units of information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>center</code></td>
<td>
<p>a logical indicating whether SNPs should be centred to
mean zero.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>a logical indicating whether SNPs should be scaled to
unit variance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useC</code></td>
<td>
<p>a logical indicating whether compiled C code should be
used (TRUE) or not (FALSE, default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>a logical indicating whether multiple cores -if
available- should be used for the computations (TRUE, default), or
not (FALSE); requires the package <code>parallel</code> to be installed
(see details); this option cannot be used alongside useCoption.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p>if <code>parallel</code> is TRUE, the number of cores to
be used in the computations; if NULL, then the maximum number of
cores available on the computer is used.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>=== On the unit of information ===
</p>
<p>In the cases where individuals can have different ploidy, computation
of sums, means, etc. of allelic data depends on what we consider as a
unit of information.
</p>
<p>To estimate e.g. allele frequencies, unit of information can be
considered as the allele, so that a diploid genotype contains two
samples, a triploid individual, three samples, etc. In such a case,
all computations are done directly on the number of alleles. This
corresponds to <code>alleleAsUnit = TRUE</code>.
</p>
<p>However, when the focus is put on studying differences/similarities
between individuals, the unit of information is the individual, and
all genotypes possess the same information no matter what their ploidy
is. In this case, computations are made after standardizing
individual genotypes to relative allele frequencies. This
corresponds to <code>alleleAsUnit = FALSE</code>.
</p>
<p>Note that when all individuals have the same ploidy, this distinction
does not hold any more.
</p>


<h3>Value</h3>

<p>A numeric vector containing the requested information.
</p>


<h3>Author(s)</h3>

<p> Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a> </p>


<h3>See Also</h3>

<p>- <code>genlight</code>: class of object for storing massive binary
SNP data.
</p>
<p>- <code>dapc</code>: Discriminant Analysis of Principal Components.
</p>
<p>- <code>glPca</code>: PCA for genlight objects.
</p>
<p>- <code>glSim</code>: a simple simulator for genlight objects.
</p>
<p>- <code>glPlot</code>: plotting genlight objects.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
x &lt;- new("genlight", list(c(0,0,1,1,0), c(1,1,1,0,0,1), c(2,1,1,1,1,NA)))
x
as.matrix(x)
ploidy(x)

## compute statistics - allele as unit ##
glNA(x)
glSum(x)
glMean(x)

## compute statistics - individual as unit ##
glNA(x, FALSE)
glSum(x, FALSE)
glMean(x, FALSE)

## explanation: data are taken as relative frequencies
temp &lt;- as.matrix(x)/ploidy(x)
apply(temp,2, function(e) sum(is.na(e))) # NAs
apply(temp,2,sum, na.rm=TRUE) # sum
apply(temp,2,mean, na.rm=TRUE) # mean

## End(Not run)
</code></pre>


</div>