<div class="container">

<table style="width: 100%;"><tr>
<td>adegenet.package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The adegenet package</h2>

<h3>Description</h3>

<p>This package is devoted to the multivariate analysis of genetic markers
data. These data can be codominant markers (e.g. microsatellites) or
presence/absence data (e.g. AFLP), and have any level of ploidy.  'adegenet'
defines three formal (S4) classes:<br> - genind: a class for
data of individuals ("genind" stands for genotypes-individuals).<br> -
genpop: a class for data of groups of individuals ("genpop"
stands for genotypes-populations)<br> - genlight: a class for
genome-wide SNP data<br></p>


<h3>Details</h3>

<p>For more information about these classes, type "class ? genind", "class ?
genpop", or "?genlight".<br></p>
<p>Essential functionalities of the package are presented througout 4
tutorials, accessible using <code>adegenetTutorial(which="name-below")</code>:<br>
- <code>basics</code>: introduction to the package.<br> - <code>spca</code>: multivariate
analysis of spatial genetic patterns.<br> - <code>dapc</code>: population structure
and group assignment using DAPC.<br> - <code>genomics</code>: introduction to the
class genlight for the handling and analysis of genome-wide
SNP data.<br></p>
<p>Note: In older versions of adegenet, these tutorials were avilable as
vignettes, accessible through the function <code>vignette("name-below",
package="adegenet")</code>:<br> - <code>adegenet-basics</code>.<br> -
<code>adegenet-spca</code>.<br> - <code>adegenet-dapc</code>.<br> -
<code>adegenet-genomics</code>.<br></p>
<p>Important functions are also summarized below.<br></p>
<p>=== IMPORTING DATA ===<br> = TO GENIND OBJECTS = <br><code>adegenet</code> imports
data to genind object from the following softwares:<br> -
STRUCTURE: see <code>read.structure</code><br> - GENETIX: see
<code>read.genetix</code><br> - FSTAT: see <code>read.fstat</code><br> -
Genepop: see <code>read.genepop</code><br> To import data from any of these
formats, you can also use the general function
<code>import2genind</code>.<br></p>
<p>In addition, it can extract polymorphic sites from nucleotide and amino-acid
alignments:<br> - DNA files: use <code>read.dna</code> from the ape
package, and then extract SNPs from DNA alignments using
<code>DNAbin2genind</code>. <br></p>
<p>- protein sequences alignments: polymorphic sites can be extracted from
protein sequences alignments in <code>alignment</code> format (package
<code>seqinr</code>, see <code>as.alignment</code>) using the function
<code>alignment2genind</code>. <br></p>
<p>The function <code>fasta2DNAbin</code> allows for reading fasta files into
DNAbin object with minimum RAM requirements.<br></p>
<p>It is also possible to read genotypes coded by character strings from a
data.frame in which genotypes are in rows, markers in columns. For this, use
<code>df2genind</code>. Note that <code>df2genind</code> can be used for
any level of ploidy.<br></p>
<p>= TO GENLIGHT OBJECTS = <br> SNP data can be read from the following
formats:<br> - PLINK: see function <code>read.PLINK</code><br> - .snp
(adegenet's own format): see function <code>read.snp</code><br></p>
<p>SNP can also be extracted from aligned DNA sequences with the fasta format,
using <code>fasta2genlight</code><br></p>
<p>=== EXPORTING DATA ===<br><code>adegenet</code> exports data from
</p>
<p>Genotypes can also be recoded from a genind object into a
data.frame of character strings, using any separator between alleles. This
covers formats from many softwares like GENETIX or STRUCTURE. For this, see
<code>genind2df</code>.<br></p>
<p>Also note that the <code>pegas</code> package imports genind objects
using the function <code>as.loci</code>.
</p>
<p>=== MANIPULATING DATA ===<br> Several functions allow one to manipulate
genind or genpop objects<br> -
<code>genind2genpop</code>: convert a genind object to a
genpop <br> - <code>seploc</code>: creates one object per
marker; for genlight objects, creates blocks of SNPs.<br> -
<code>seppop</code>: creates one object per population <br> -
- <code>tab</code>: access the allele data (counts or frequencies) of an object
(genind and genpop) <br> -
x[i,j]: create a new object keeping only genotypes (or populations) indexed
by 'i' and the alleles indexed by 'j'.<br> - <code>makefreq</code>: returns
a table of allelic frequencies from a genpop object.<br> -
<code>repool</code> merges genoptypes from different gene pools into one
single genind object.<br> - <code>propTyped</code> returns the
proportion of available (typed) data, by individual, population, and/or
locus.<br> - <code>selPopSize</code> subsets data, retaining only genotypes
from a population whose sample size is above a given level.<br> -
<code>pop</code> sets the population of a set of genotypes.<br></p>
<p>=== ANALYZING DATA ===<br> Several functions allow to use usual, and less
usual analyses:<br> - <code>HWE.test.genind</code>: performs HWE test for all
populations and loci combinations <br> - <code>dist.genpop</code>: computes 5
genetic distances among populations.  <br> - <code>monmonier</code>:
implementation of the Monmonier algorithm, used to seek genetic boundaries
among individuals or populations. Optimized boundaries can be obtained using
<code>optimize.monmonier</code>. Object of the class <code>monmonier</code> can be
plotted and printed using the corresponding methods. <br> -
<code>spca</code>: implements Jombart et al.  (2008) spatial Principal
Component Analysis <br> - <code>global.rtest</code>: implements Jombart et
al. (2008) test for global spatial structures <br> -
<code>local.rtest</code>: implements Jombart et al. (2008) test for local
spatial structures <br> - <code>propShared</code>: computes the proportion of
shared alleles in a set of genotypes (i.e. from a genind object)<br> -
<code>propTyped</code>: function to investigate missing data in several ways
<br> - <code>scaleGen</code>: generic method to scale genind or
genpop before a principal component analysis <br> -
<code>Hs</code>: computes the average expected heterozygosity by population
in a genpop. Classically Used as a measure of genetic
diversity.<br> - <code>find.clusters</code> and <code>dapc</code>: implement
the Discriminant Analysis of Principal Component (DAPC, Jombart et al.,
2010).<br> - <code>seqTrack</code>: implements the SeqTrack algorithm for
recontructing transmission trees of pathogens (Jombart et al., 2010) .<br><code>glPca</code>: implements PCA for genlight objects.<br> -
<code>gengraph</code>: implements some simple graph-based clustering using
genetic data.  - <code>snpposi.plot</code> and <code>snpposi.test</code>:
visualize the distribution of SNPs on a genetic sequence and test their
randomness.  - <code>adegenetServer</code>: opens up a web interface for
some functionalities of the package (DAPC with cross validation and feature
selection).<br></p>
<p>=== GRAPHICS ===<br> - <code>colorplot</code>: plots points with associated
values for up to three variables represented by colors using the RGB system;
useful for spatial mapping of principal components.<br> -
<code>loadingplot</code>: plots loadings of variables. Useful for
representing the contribution of alleles to a given principal component in a
multivariate method. <br> - <code>scatter.dapc</code>: scatterplots for DAPC
results.<br> - <code>compoplot</code>: plots membership probabilities from a
DAPC object. <br></p>
<p>=== SIMULATING DATA ===<br> - <code>hybridize</code>: implements
hybridization between two populations. <br> - <code>haploGen</code>:
simulates genealogies of haplotypes, storing full genomes. <br><code>glSim</code>: simulates simple genlight objects.<br></p>
<p>=== DATASETS ===<br> - <code>H3N2</code>: Seasonal influenza (H3N2) HA
segment data. <br> - <code>dapcIllus</code>: Simulated data illustrating the
DAPC. <br> - <code>eHGDP</code>: Extended HGDP-CEPH dataset. <br> -
<code>microbov</code>: Microsatellites genotypes of 15 cattle breeds. <br> -
<code>nancycats</code>: Microsatellites genotypes of 237 cats from 17
colonies of Nancy (France). <br> - <code>rupica</code>: Microsatellites
genotypes of 335 chamois (Rupicapra rupicapra) from the Bauges mountains
(France).<br> - <code>sim2pop</code>: Simulated genotypes of two
georeferenced populations.<br> - <code>spcaIllus</code>: Simulated data
illustrating the sPCA. <br></p>
<p>For more information, visit the adegenet website using the function
<code>adegenetWeb</code>.<br></p>
<p>Tutorials are available via the command <code>adegenetTutorial</code>.<br></p>
<p>To cite adegenet, please use the reference given by
<code>citation("adegenet")</code> (or see references below).
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart &lt;t.jombart@imperial.ac.uk&gt;<br>
Developers: Zhian N. Kamvar &lt;zkamvar@gmail.com&gt;,
Caitlin Collins &lt;caitiecollins17@gmail.com&gt;,
Ismail Ahmed &lt;ismail.ahmed@inserm.fr&gt;,
Federico Calboli, Tobias Erik Reiners, Peter
Solymos, Anne Cori, <br> Contributed datasets from: Katayoun
Moazami-Goudarzi, Denis LaloÃ«, Dominique Pontier, Daniel Maillard, Francois
Balloux.
</p>


<h3>References</h3>

<p>Jombart T. (2008) adegenet: a R package for the multivariate
analysis of genetic markers <em>Bioinformatics</em> 24: 1403-1405. doi:
10.1093/bioinformatics/btn129<br></p>
<p>Jombart T. and Ahmed I. (2011) adegenet 1.3-1: new tools for the analysis of
genome-wide SNP data.  <em>Bioinformatics</em>. doi:
10.1093/bioinformatics/btr521
</p>
<p>Jombart T, Devillard S and Balloux F (2010) Discriminant analysis of
principal components: a new method for the analysis of genetically
structured populations. BMC Genetics 11:94.  doi:10.1186/1471-2156-11-94<br></p>
<p>Jombart T, Eggo R, Dodd P, Balloux F (2010) Reconstructing disease outbreaks
from genetic data: a graph approach. <em>Heredity</em>. doi:
10.1038/hdy.2010.78.<br></p>
<p>Jombart, T., Devillard, S., Dufour, A.-B. and Pontier, D. (2008) Revealing
cryptic spatial patterns in genetic variability by a new multivariate
method. <em>Heredity</em>, <b>101</b>, 92â103.<br></p>
<p>See adegenet website: <a href="http://adegenet.r-forge.r-project.org/">http://adegenet.r-forge.r-project.org/</a><br></p>
<p>Please post your questions on 'the adegenet forum':
adegenet-forum@lists.r-forge.r-project.org
</p>


<h3>See Also</h3>

<p>adegenet is related to several packages, in particular:<br> -
<code>ade4</code> for multivariate analysis<br> - <code>pegas</code> for population
genetics tools<br> - <code>ape</code> for phylogenetics and DNA data handling<br> -
<code>seqinr</code> for handling nucleic and proteic sequences<br> - <code>shiny</code>
for R-based web interfaces<br></p>


</div>