<div class="container">

<table style="width: 100%;"><tr>
<td>genind class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>adegenet formal class (S4) for individual genotypes</h2>

<h3>Description</h3>

<p>The S4 class <code>genind</code> is used to store individual genotypes.<br>
It contains several components described in the 'slots' section).<br>
The <code>summary</code> of a <code>genind</code> object invisibly returns a list of component.
The function <code>.valid.genind</code> is for internal use.
The function <code>genind</code> creates a genind object from a valid table
of alleles corresponding to the <code>@tab</code> slot.
Note that as in other S4 classes, slots are accessed using @ instead
of $.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>tab</code>:</dt>
<dd> <p><strong>(accessor: <code>tab</code>)</strong>
matrix integers containing genotypes data for
individuals (in rows) for all alleles (in columns). The table
differs depending on the <code>@type</code> slot:<br>
- 'codom': values
are numbers of alleles, summing up to the
individuals' ploidies.<br>
- 'PA': values are presence/absence of alleles.<br>
In all cases, rows and columns are given generic names.</p>
</dd>
<dt>
<code>loc.fac</code>:</dt>
<dd>
<p><strong>(accessor: <code>locFac</code>)</strong> locus 
factor for the columns of <code>tab</code></p>
</dd>
<dt>
<code>loc.n.all</code>:</dt>
<dd>
<p><strong>(accessor: <code>nAll</code>)</strong> integer 
vector giving the number of observed alleles per locus (see note)</p>
</dd>
<dt>
<code>all.names</code>:</dt>
<dd>
<p><strong>(accessor: <code>alleles</code>)</strong> list 
having one component per locus, each containing a character vector of allele
names</p>
</dd>
<dt>
<code>ploidy</code>:</dt>
<dd>
<p><strong>(accessor: <code>ploidy</code>)</strong> an integer 
vector indicating the degree of ploidy of the genotypes. Beware: 2 is not 
an integer, but 2L or as.integer(2) is.</p>
</dd>
<dt>
<code>type</code>:</dt>
<dd>
<p> a character string indicating the type of
marker: 'codom' stands for 'codominant' (e.g. microstallites,
allozymes); 'PA' stands for 'presence/absence' (e.g. AFLP).</p>
</dd>
<dt>
<code>call</code>:</dt>
<dd>
<p>the matched call</p>
</dd>
<dt>
<code>strata</code>:</dt>
<dd>
<p><strong>(accessor: <code>strata</code>)</strong> (optional) 
data frame giving levels of population stratification for each individual</p>
</dd>
<dt>
<code>hierarchy</code>:</dt>
<dd>
<p><strong>(accessor: <code>hier</code>)</strong> (optional, 
currently unused) a hierarchical <code>formula</code> defining the 
hierarchical levels in the <code>@@strata</code> slot.</p>
</dd>
<dt>
<code>pop</code>:</dt>
<dd>
<p><strong>(accessor: <code>pop</code>)</strong> (optional) factor 
giving the population of each individual</p>
</dd>
<dt>
<code>other</code>:</dt>
<dd>
<p><strong>(accessor: <code>other</code>)</strong> (optional) a 
list containing other information</p>
</dd>
</dl>
<h3>Note:</h3>

<p>The <code>loc.n.all</code> slot will reflect the number of columns per locus that 
contain at least one observation. This means that the sum of the this vector 
will not necessarily equal the number of columns in the data unless you use 
<code>drop = TRUE</code> when subsetting. 
</p>


<h3>Extends</h3>

<p>Class <code>"gen"</code>, directly.
Class <code>"indInfo"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>names</dt>
<dd>
<p><code>signature(x = "genind")</code>: give the names of the
components of a genind object</p>
</dd>
<dt>print</dt>
<dd>
<p><code>signature(x = "genind")</code>: prints a genind object</p>
</dd>
<dt>show</dt>
<dd>
<p><code>signature(object = "genind")</code>: shows a genind
object (same as print)</p>
</dd>
<dt>summary</dt>
<dd>
<p><code>signature(object = "genind")</code>: summarizes a
genind object, invisibly returning its content or suppress printing of auxiliary information by specifying <code>verbose = FALSE</code></p>
</dd>
</dl>
<h3>Author(s)</h3>

<p> Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a> </p>


<h3>See Also</h3>

<p><code>as.genind</code>, <code>genind2genpop</code>,
<code>genpop</code>, <code>import2genind</code>,
<code>read.genetix</code>, <code>read.genepop</code>,
<code>read.fstat</code><br></p>
<p>Related classes:<br>
- genpop for storing data per populations<br></p>
<p>- genlight for an efficient storage of binary SNPs genotypes<br></p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("genind")

obj &lt;- read.genetix(system.file("files/nancycats.gtx",package="adegenet"))
obj
validObject(obj)
summary(obj)

## Not run: 
# test inter-colonies structuration
if(require(hierfstat)){
gtest &lt;- gstat.randtest(obj,nsim=99)
gtest
plot(gtest)
}

# perform a between-class PCA
pca1 &lt;- dudi.pca(scaleGen(obj, NA.method="mean"),scannf=FALSE,scale=FALSE)
pcabet1 &lt;- between(pca1,obj@pop,scannf=FALSE)
pcabet1

s.class(pcabet1$ls,obj@pop,sub="Inter-class PCA",possub="topleft",csub=2)
add.scatter.eig(pcabet1$eig,2,xax=1,yax=2)


## End(Not run)
</code></pre>


</div>