<div class="container">

<table style="width: 100%;"><tr>
<td>read.snp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reading Single Nucleotide Polymorphism data</h2>

<h3>Description</h3>

<p>The function <code>read.snp</code> reads a SNP data file with extension '.snp' and
converts it into a genlight object. This format is devoted to
handle biallelic SNP only, but can accommodate massive datasets such as
complete genomes with considerably less memory than other formats.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.snp(
  file,
  quiet = FALSE,
  chunkSize = 1000,
  parallel = FALSE,
  n.cores = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>a character string giving the path to the file to convert, with
the extension ".snp".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>logical stating whether a conversion messages should be printed
(TRUE,default) or not (FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chunkSize</code></td>
<td>
<p>an integer indicating the number of genomes to be read at a
time; larger values require more RAM but decrease the time needed to read
the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>a logical indicating whether multiple cores -if available-
should be used for the computations (TRUE, default), or not (FALSE);
requires the package <code>parallel</code> to be installed (see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p>if <code>parallel</code> is TRUE, the number of cores to be used in
the computations; if NULL, then the maximum number of cores available on the
computer is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other arguments to be passed to other functions - currently not
used.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function reads data by chunks of a few genomes (minimum 1, no maximum)
at a time, which allows one to read massive datasets with negligible RAM
requirements (albeit at a cost of computational time). The argument
<code>chunkSize</code> indicates the number of genomes read at a time. Increasing
this value decreases the computational time required to read data in, while
increasing memory requirements.
</p>
<p>A description of the .snp format is provided in an example file distributed
with adegenet (see example below).
</p>
<p>=== The .snp format ===
</p>
<p>Details of the .snp format can be found in the example file distributed with
adegenet (see below), or on the adegenet website (type <code>adegenetWeb()</code>
in R).
</p>


<h3>Value</h3>

<p>an object of the class <code>"genlight"</code>
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>
</p>


<h3>See Also</h3>

<p>- <code>?genlight</code> for a description of the class
<code>"genlight"</code>.
</p>
<p>- <code>read.PLINK</code>: read SNPs in PLINK's '.raw' format.
</p>
<p>- <code>fasta2genlight</code>: extract SNPs from alignments with fasta
format.
</p>
<p>- <code>df2genind</code>: convert any multiallelic markers into adegenet
<code>"genlight"</code>.
</p>
<p>- <code>import2genind</code>: read multiallelic markers from various
software into adegenet.<br></p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 
## show the example file ##
## this is the path to the file:
system.file("files/exampleSnpDat.snp",package="adegenet")

## show its content:
file.show(system.file("files/exampleSnpDat.snp",package="adegenet"))


## read the file
obj &lt;-
read.snp(system.file("files/exampleSnpDat.snp",package="adegenet"), chunk=2)
obj
as.matrix(obj)
ploidy(obj)
alleles(obj)
locNames(obj)

## End(Not run)

</code></pre>


</div>