<div class="container">

<table style="width: 100%;"><tr>
<td>SequencesToGenind</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Importing data from an alignement of sequences to a genind object</h2>

<h3>Description</h3>

<p>These functions take an alignement of sequences and translate SNPs
into a genind object. Note that only polymorphic loci
are retained.<br></p>
<p>Currently, accepted sequence formats are:<br>
- DNAbin (ape package): function DNAbin2genind<br>
- alignment (seqinr package): function alignment2genind<br></p>


<h3>Usage</h3>

<pre><code class="language-R">DNAbin2genind(x, pop=NULL, exp.char=c("a","t","g","c"), polyThres=1/100)

alignment2genind(x, pop=NULL, exp.char=c("a","t","g","c"), na.char="-",
                 polyThres=1/100)

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object containing aligned sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>an optional factor giving the population to which each sequence belongs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exp.char</code></td>
<td>
<p>a vector of single character providing expected values;
all other characters will be turned to NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.char</code></td>
<td>
<p>a vector of single characters providing values that should be
considered as NA. If not NULL, this is used instead of <code>exp.char</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>polyThres</code></td>
<td>
<p>the minimum frequency of a minor allele for a locus to
be considered as polymorphic (defaults to 0.01).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>an object of the class genind</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a> </p>


<h3>See Also</h3>

<p><code>import2genind</code>, <code>read.genetix</code>,
<code>read.fstat</code>, <code>read.structure</code>,
<code>read.genepop</code>, <code>DNAbin</code>,
<code>as.alignment</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data(woodmouse)
x &lt;- DNAbin2genind(woodmouse)
x
genind2df(x)

## End(Not run)

if(require(seqinr)){
mase.res   &lt;- read.alignment(file=system.file("sequences/test.mase",package="seqinr"),
format = "mase")
mase.res
x &lt;- alignment2genind(mase.res)
x
locNames(x) # list of polymorphic sites
genind2df(x)

## look at Euclidean distances
D &lt;- dist(tab(x))
D

## summarise with a PCoA
pco1 &lt;- dudi.pco(D, scannf=FALSE,nf=2)
scatter(pco1, posi="bottomright")
title("Principal Coordinate Analysis\n-based on proteic distances-")

}
</code></pre>


</div>