<div class="container">

<table style="width: 100%;"><tr>
<td>fasta2DNAbin</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Read large DNA alignments into R</h2>

<h3>Description</h3>

<p>The function <code>fasta2DNAbin</code> reads alignments with the fasta
format (extensions ".fasta", ".fas", or ".fa"), and outputs a
<code>DNAbin</code> object (the efficient DNA representation from the
ape package). The output contains either the full alignments, or only
SNPs. This implementation is designed for memory-efficiency,
and can read in larger datasets than Ape's <code>read.dna</code>.
</p>
<p>The function reads data by chunks of a few genomes (minimum 1, no
maximum) at a time, which allows one to read massive datasets with
negligible RAM requirements (albeit at a cost of computational
time). The argument <code>chunkSize</code> indicates the number of genomes
read at a time. Increasing this value decreases the computational time
required to read data in, while increasing memory requirements.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fasta2DNAbin(file, quiet=FALSE, chunkSize=10, snpOnly=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> a character string giving the path to the file to
convert, with the extension ".fa", ".fas", or ".fasta".
</p>
<p>Can also be a connection (which will be opened for reading if
necessary, and if so <code>close</code>d (and hence destroyed) at the
end of the function call).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>a logical stating whether a conversion messages should be
printed (FALSE, default) or not (TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chunkSize</code></td>
<td>
<p>an integer indicating the number of genomes to be
read at a time; larger values require more RAM but decrease the time
needed to read the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpOnly</code></td>
<td>
<p>a logical indicating whether SNPs only should be returned.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>an object of the class <code>DNAbin</code></p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a> </p>


<h3>See Also</h3>

<p>- <code>?DNAbin</code> for a description of the class <code>DNAbin</code>.
</p>
<p>- <code>read.snp</code>: read SNPs in adegenet's '.snp' format.
</p>
<p>- <code>read.PLINK</code>: read SNPs in PLINK's '.raw' format.
</p>
<p>- <code>df2genind</code>: convert any multiallelic markers into
adegenet genind.
</p>
<p>- <code>import2genind</code>: read multiallelic markers from various
software into adegenet.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## show the example file ##
## this is the path to the file:
myPath &lt;- system.file("files/usflu.fasta",package="adegenet")
myPath

## read the file
obj &lt;- fasta2DNAbin(myPath, chunk=10) # process 10 sequences at a time
obj

## End(Not run)
</code></pre>


</div>