<div class="container">

<table style="width: 100%;"><tr>
<td>findMutations</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identify mutations between DNA sequences</h2>

<h3>Description</h3>

<p>The function <code>findMutations</code> identifies mutations (position and
nature) of pairs of aligned DNA sequences. The function
<code>graphMutations</code> does the same thing but plotting mutations on a
directed graph.<br></p>
<p>Both functions are generics, but the only methods implemented in
adegenet so far is for <code>DNAbin</code> objects.
</p>


<h3>Usage</h3>

<pre><code class="language-R">findMutations(...)

## S3 method for class 'DNAbin'
findMutations(x, from=NULL, to=NULL, allcomb=TRUE, ...)

graphMutations(...)

## S3 method for class 'DNAbin'
graphMutations(x, from=NULL, to=NULL, allcomb=TRUE, plot=TRUE,
               curved.edges=TRUE, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a <code>DNAbin</code> object containing aligned sequences, as a matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>from</code></td>
<td>
<p>a vector indicating the DNA sequences from which mutations
should be found. If <code>NULL</code>, all sequences are considered (i.e.,
<code>1:nrow(x)</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>to</code></td>
<td>
<p>a vector indicating the DNA sequences to which mutations
should be found. If <code>NULL</code>, all sequences are considered (i.e.,
<code>1:nrow(x)</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allcomb</code></td>
<td>
<p>a logical indicating whether all combinations of
sequences (from and to) should be considered (TRUE, default), or not
(FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>a logical indicating whether the graph should be plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>curved.edges</code></td>
<td>
<p>a logical indicating whether the edges of the
graph should be curved.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to other methods. Used in
<code>graphMutations</code> where it is passed to the plot method for
<code>igraph</code> objects.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>For <code>findMutations</code>, a named list indicating the mutations from
one sequence to another. For each comparison, a three-column matrix is
provided, corresponding to the nucleotides in first and second
sequence, and a summary of the mutation provided as:
[position]:[nucleotide in first sequence]-&gt;[nucleotide in second
sequence].
</p>
<p>For <code>graphMutations</code>, a graph with the class <code>igraph</code>.
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a>.
</p>


<h3>See Also</h3>

<p>The <code>fasta2DNAbin</code> to read fasta alignments with minimum
RAM use.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data(woodmouse)

## mutations between first 3 sequences
findMutations(woodmouse[1:3,])

## mutations from the first to sequences 2 and 3
findMutations(woodmouse[1:3,], from=1)

## same, graphical display
g &lt;- graphMutations(woodmouse[1:3,], from=1)

## some manual checks
as.character(woodmouse)[1:3,35]
as.character(woodmouse)[1:3,36]
as.character(woodmouse)[1:3,106]


## End(Not run)
</code></pre>


</div>