<div class="container">

<table style="width: 100%;"><tr>
<td>glSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulation of simple genlight objects</h2>

<h3>Description</h3>

<p>The function <code>glSim</code> simulates simple SNP data with the
possibility of contrasted structures between two groups
as well as background ancestral population structure. 
Returned objects are instances of the class genlight.
</p>


<h3>Usage</h3>

<pre><code class="language-R">glSim(n.ind, n.snp.nonstruc, n.snp.struc = 0, grp.size = c(0.5, 0.5), k = NULL,
                    pop.freq = NULL, ploidy = 1, alpha = 0, parallel = FALSE,
                    LD = TRUE, block.minsize = 10, block.maxsize = 1000, theta = NULL,
                    sort.pop = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n.ind</code></td>
<td>
<p>an integer indicating the number of individuals to be simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.snp.nonstruc</code></td>
<td>
<p>an integer indicating the number of
non-structured SNPs to be simulated; for these SNPs, all individuals
are drawn from the same binomial distribution.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.snp.struc</code></td>
<td>
<p>an integer indicating the number of
structured SNPs to be simulated; for these SNPs, different
binomial distributions are used for the two simulated groups;
frequencies of the derived alleles in groups A and B are built to
differ (see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grp.size</code></td>
<td>
<p>a vector of length 2 specifying the proportions of the two 
phenotypic groups (must sum to 1). By default, both groups have the same
size.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>an integer specifying the number of ancestral populations to be 
generated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop.freq</code></td>
<td>
<p>a vector of length <code>k</code> specifying the proportions of the
k ancestral populations (must sum to 1). If, as by default, <code>pop.freq</code> 
is null, and <code>k</code> is non-null, <code>pop.freq</code> will be the result of
random sampling into k population groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>an integer indicating the ploidy of the simulated
genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>asymmetry parameter: a numeric value between 0 and 0.5,
used to enforce allelic differences between the groups. Differences
between groups are strongest when alpha = 0.5 and weakest when 
alpha = 0 (see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>a logical indicating whether multiple cores should be used in 
generating the simulated data (TRUE). This option can reduce the amount of 
computational time required to simulate the data, but is not supported on 
Windows.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LD</code></td>
<td>
<p>a logical indicating whether loci should be displaying
linkage disequilibrium (TRUE) or be generated independently
(FALSE, default). When set to TRUE, data are generated by blocks of
correlated SNPs (see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block.minsize</code></td>
<td>
<p>an optional integer indicating the minimum number of 
SNPs to be handled at a time during the simulation of linked SNPs (when 
<code>LD=TRUE</code>. Increasing the minimum block size will increase 
the RAM requirement but decrease the amount of computational time 
required to simulate the genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block.maxsize</code></td>
<td>
<p>an optional integer indicating the maximum number of SNPs
to be handled at a time during the simulation of linked SNPs. Note: if LD
blocks of equal size are desired, set block.minsize = block.maxsize.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>an optional numeric value between 0 and 0.5 specifying 
the extent to which linkage should be diluted.
Linkage is strongest when theta = 0 and weakest when theta = 0.5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort.pop</code></td>
<td>
<p>a logical specifying whether individuals should be ordered by
ancestral population (<code>sort.pop=TRUE</code>) or phenotypic population 
(<code>sort.pop=FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments to be passed to the genlight constructor.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>=== Allele frequencies in contrasted groups ===
</p>
<p>When <code>n.snp.struc</code> is greater than 0, some SNPs are simulated in
order to differ between groups (noted 'A' and 'B'). Different patterns 
between groups are achieved by using different
frequencies of the second allele for A and B, denoted <code class="reqn">p_A</code> and
<code class="reqn">p_B</code>. For a given SNP, <code class="reqn">p_A</code> is drawn from a uniform
distribution between 0 and (0.5 - alpha). <code class="reqn">p_B</code> is then computed
as 1 - <code class="reqn">p_A</code>. Therefore, differences between groups are mild for
alpha=0, and total for alpha = 0.5.
</p>
<p>=== Linked or independent loci ===
</p>
<p>Independent loci (<code>LD=FALSE</code>) are simulated using the standard
binomial distribution, with randomly generated allele
frequencies. Linked loci (<code>LD=FALSE</code>) are trickier towe need to
simulate discrete variables with pre-defined correlation structure.
</p>
<p>Here, we first generate deviates from multivariate normal
distributions with randomly generated correlation structures. These
variables are then discretized using the quantiles of the
distribution. Further improvement of the procedure will aim at i)
specifying the strength of the correlations between blocks of alleles
and ii) enforce contrasted structures between groups.
</p>


<h3>Value</h3>

<p>A genlight object.
</p>


<h3>Author(s)</h3>

<p> Caitlin Collins <a href="mailto:caitlin.collins12@imperial.ac.uk">caitlin.collins12@imperial.ac.uk</a>,
Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a></p>


<h3>See Also</h3>

<p>- <code>genlight</code>: class of object for storing massive binary
SNP data.
</p>
<p>- <code>glPlot</code>: plotting genlight objects.
</p>
<p>- <code>glPca</code>: PCA for genlight objects.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## no structure
x &lt;- glSim(100, 1e3, ploid=2)
plot(x)

## 1,000 non structured SNPs, 100 structured SNPs
x &lt;- glSim(100, 1e3, n.snp.struc=100, ploid=2)
plot(x)

## 1,000 non structured SNPs, 100 structured SNPs, ploidy=4
x &lt;- glSim(100, 1e3, n.snp.struc=100, ploid=4)
plot(x)

## same thing, stronger differences between groups
x &lt;- glSim(100, 1e3, n.snp.struc=100, ploid=2, alpha=0.4)
plot(x)

##  same thing, loci with LD structures
x &lt;- glSim(100, 1e3, n.snp.struc=100, ploid=2, alpha=0.4, LD=TRUE, block.minsize=100)
plot(x)

## End(Not run)
</code></pre>


</div>