<div class="container">

<table style="width: 100%;"><tr>
<td>dist.genpop</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Genetic distances between populations </h2>

<h3>Description</h3>

<p>This function computes measures of genetic distances
between populations using a <code>genpop</code> object. <br>
Currently, five distances are available, some of which are euclidian
(see details).<br></p>
<p>A non-euclidian distance can be transformed into an Euclidean one
using <code>cailliez</code> in order to perform a
Principal Coordinate Analysis <code>dudi.pco</code> (both
functions in <code>ade4</code>). <br></p>
<p>The function <code>dist.genpop</code> is based on former <code>dist.genet</code>
function of <code>ade4</code> package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dist.genpop(x, method = 1, diag = FALSE, upper = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> a list of class <code>genpop</code>  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p> an integer between 1 and 5. See details </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diag</code></td>
<td>
<p> a logical value indicating whether the diagonal of the distance matrix should be printed by <code>print.dist</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upper</code></td>
<td>
<p> a logical value indicating whether the upper triangle of the distance matrix should be printed by <code>print.dist</code> </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Let <b>A</b> a table containing allelic frequencies with <em>t</em> populations (rows) and <em>m</em> alleles (columns).<br>
Let  <code class="reqn">\nu</code> the number of loci. The locus <em>j</em> gets <em>m(j)</em> alleles. 
<code class="reqn">m=\sum_{j=1}^{\nu} m(j)</code><br></p>
<p>For the row <em>i</em> and the modality <em>k</em> of the variable <em>j</em>, notice the value <code class="reqn">a_{ij}^k</code> (<code class="reqn">1 \leq i \leq t</code>, <code class="reqn">1 \leq j \leq \nu</code>,
<code class="reqn">1 \leq k \leq m(j)</code>) the value of the initial table.<br></p>
<p><code class="reqn">a_{ij}^+=\sum_{k=1}^{m(j)}a_{ij}^k</code> and <code class="reqn">p_{ij}^k=\frac{a_{ij}^k}{a_{ij}^+}</code><br></p>
<p>Let <b>P</b> the table of general term <code class="reqn">p_{ij}^k</code><br><code class="reqn">p_{ij}^+=\sum_{k=1}^{m(j)}p_{ij}^k=1</code>, <code class="reqn">p_{i+}^+=\sum_{j=1}^{\nu}p_{ij}^+=\nu</code>, <code class="reqn">p_{++}^+=\sum_{j=1}^{\nu}p_{i+}^+=t\nu</code><br></p>
<p>The option <code>method</code> computes the distance matrices between populations using the frequencies <code class="reqn">p_{ij}^k</code>. <br></p>
<p>1. Nei's distance (not Euclidean): <br><code class="reqn">D_1(a,b)=- \ln(\frac{\sum_{k=1}^{\nu} \sum_{j=1}^{m(k)}
p_{aj}^k p_{bj}^k}{\sqrt{\sum_{k=1}^{\nu} \sum_{j=1}^{m(k)}
{(p_{aj}^k) }^2}\sqrt{\sum_{k=1}^{\nu} \sum_{j=1}^{m(k)}
{(p_{bj}^k)}^2}})</code><br></p>
<p>2. Angular distance or Edwards' distance (Euclidean):<br><code class="reqn">D_2(a,b)=\sqrt{1-\frac{1}{\nu} \sum_{k=1}^{\nu}
\sum_{j=1}^{m(k)} \sqrt{p_{aj}^k  p_{bj}^k}}</code><br></p>
<p>3. Coancestrality coefficient or Reynolds' distance (Eucledian):<br><code class="reqn">D_3(a,b)=\sqrt{\frac{\sum_{k=1}^{\nu}
\sum_{j=1}^{m(k)}{(p_{aj}^k - p_{bj}^k)}^2}{2 \sum_{k=1}^{\nu} (1-
\sum_{j=1}^{m(k)}p_{aj}^k p_{bj}^k)}}</code><br></p>
<p>4. Classical Euclidean distance or Rogers' distance (Eucledian):<br><code class="reqn">D_4(a,b)=\frac{1}{\nu} \sum_{k=1}^{\nu} \sqrt{\frac{1}{2}
\sum_{j=1}^{m(k)}{(p_{aj}^k - p_{bj}^k)}^2}</code><br></p>
<p>5. Absolute genetics distance or Provesti 's distance (not Euclidean):<br><code class="reqn">D_5(a,b)=\frac{1}{2{\nu}} \sum_{k=1}^{\nu} \sum_{j=1}^{m(k)}
|p_{aj}^k - p_{bj}^k|</code>
</p>


<h3>Value</h3>

<p>returns a distance matrix of class <code>dist</code> between the rows of the data frame
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart <a href="mailto:t.jombart@imperial.ac.uk">t.jombart@imperial.ac.uk</a><br>
Former dist.genet code by Daniel Chessel <a href="mailto:chessel@biomserv.univ-lyon1.fr">chessel@biomserv.univ-lyon1.fr</a><br>
and documentation by Anne B. Dufour <a href="mailto:dufour@biomserv.univ-lyon1.fr">dufour@biomserv.univ-lyon1.fr</a>
</p>


<h3>References</h3>

 
<p>To complete informations about distances:<br></p>
<p>Distance 1:<br>
Nei, M. (1972) Genetic distances between populations. <em>American Naturalist</em>, <b>106</b>, 283–292. <br>
Nei M. (1978) Estimation of average heterozygosity and genetic distance from a small number of individuals. <em>Genetics</em>, <b>23</b>, 341–369. <br>
Avise, J. C. (1994) Molecular markers, natural history and evolution. Chapman &amp; Hall, London. 
</p>
<p>Distance 2:<br>
Edwards, A.W.F. (1971) Distance between populations on the basis of gene frequencies. <em>Biometrics</em>, <b>27</b>, 873–881. <br>
Cavalli-Sforza L.L. and Edwards A.W.F. (1967) Phylogenetic analysis: models and estimation procedures. <em>Evolution</em>, <b>32</b>, 550–570. <br>
Hartl, D.L. and Clark, A.G. (1989) Principles of population genetics. Sinauer Associates, Sunderland, Massachussetts (p. 303).
</p>
<p>Distance 3:<br>
Reynolds, J. B., B. S. Weir, and C. C. Cockerham. (1983) Estimation of the coancestry coefficient: basis for a short-term genetic distance. <em>Genetics</em>, <b>105</b>, 767–779.
</p>
<p>Distance 4:<br>
Rogers, J.S. (1972) Measures of genetic similarity and genetic distances. <em>Studies in Genetics</em>, Univ. Texas Publ., <b>7213</b>, 145–153.  <br>
Avise, J. C. (1994) Molecular markers, natural history and evolution. Chapman &amp; Hall, London. 
</p>
<p>Distance 5:<br>
Prevosti A. (1974) La distancia genetica entre poblaciones. <em>Miscellanea Alcobe</em>, <b>68</b>, 109–118. <br>
Prevosti A., Ocaña J. and Alonso G. (1975) Distances between populations of Drosophila subobscura, based on chromosome arrangements frequencies. <em>Theoretical and Applied Genetics</em>, <b>45</b>, 231–241. <br></p>
<p>For more information on dissimilarity indexes:<br>
Gower J. and Legendre P. (1986) Metric and Euclidean properties of
dissimilarity coefficients. <em>Journal of Classification</em>, <b>3</b>,
5–48 <br></p>
<p>Legendre P. and Legendre L. (1998) <em>Numerical Ecology</em>, Elsevier
Science B.V. 20, pp274–288.<br></p>


<h3>See Also</h3>

 <p><code>cailliez</code>,<code>dudi.pco</code> 
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data(microsatt)
obj &lt;- as.genpop(microsatt$tab)

listDist &lt;- lapply(1:5, function(i) cailliez(dist.genpop(obj,met=i)))
for(i in 1:5) {attr(listDist[[i]],"Labels") &lt;- popNames(obj)}
listPco &lt;- lapply(listDist, dudi.pco,scannf=FALSE)

par(mfrow=c(2,3))
for(i in 1:5) {scatter(listPco[[i]],sub=paste("Dist:", i))}


## End(Not run)
</code></pre>


</div>