<div class="container">

<table style="width: 100%;"><tr>
<td>admix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulated Admixed Population Data</h2>

<h3>Description</h3>

<p>The example dataset provides simulated allele calls for 50 individuals from 3 populations. We consider a simple admixture scenario where two ancestral populations, popA (sample size <code class="reqn">n_A</code> = 15) and popB (sample size <code class="reqn">n_B</code> = 15), merged 160 generations ago to form the admixed population, popAB (sample size <code class="reqn">n_{AB}</code> = 20). The ancestral populations contributed to the admixed population with probability 0.5.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(admix)</code></pre>


<h3>Format</h3>

<p>A list with the following entries:
</p>

<ul>
<li> <p><b><code>data</code></b>:  3,000 x 50 data matrix with 3,000 genotype calls (rows) for 50 individuals (columns). Raw entries take the value 0 if heterozygous, and -1 or 1 if homozygous. Polymorphisms are ordered by their physical positions. Column names list individual IDs, row names list polymorphism IDs.
</p>
</li>
<li> <p><b><code>map</code></b>:  3,000 x 2 data matrix providing a mapping to genetic distance (‘recombination distance’) for each polymorphism.
</p>
</li>
<li> <p><b><code>colplot</code></b>:  vector of length 50. Plotting color for each individual. 
</p>
</li>
<li> <p><b><code>populations</code></b>:  list of length 3. Character vectors giving the IDs of individuals in each population. IDs must map to the column names of the data matrix.
</p>
</li>
</ul>
<h3>Details</h3>

<p>Further description of the dataset can be found in Sanderson et al. (2015). The data were simulated using MaCS (Chen et al. 2009).
</p>


<h3>References</h3>

<p>Sanderson J, H Sudoyo, TM Karafet, MF Hammer and MP Cox. 2015. Reconstructing past admixture processes from local genomic ancestry using wavelet transformation. <em>Genetics</em> 200:469-481. <a href="https://doi.org/10.1534/genetics.115.176842">https://doi.org/10.1534/genetics.115.176842</a>
</p>
<p>Chen GK, P Marjoram and JD Wall. 2009. Fast and flexible simulation of DNA sequence data. <em>Genome Research</em> 19:136-142. <a href="https://doi.org/10.1101/gr.083634.108">https://doi.org/10.1101/gr.083634.108</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(admix)
str(admix) 
</code></pre>


</div>