<div class="container">

<table style="width: 100%;"><tr>
<td>read_alignment</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read a protein sequence multiple alignment</h2>

<h3>Description</h3>

<p>Reads a protein sequence multiple alignment (PSMA) from either a set of
pre-bundled alignments, by gene name, or from a Multi-FASTA file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_alignment(
  gene = c("ATM", "BRCA1", "BRCA2", "CHEK2", "MRE11", "MSH6", "NBN", "PALB2", "PMS2",
    "RAD50", "RAD51", "XRCC2"),
  file = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>
<p>The gene name for which an alignment is provided with this
package. Use the function <code>alignment_file()</code> to list the pre-bundled
alignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>The path to a Multi-FASTA file. If this argument is given, it
takes precedence over the <code>gene</code> parameter.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An alignment object; essentially, a character matrix, whose elements
are protein residues in one-letter notation. Rows are sequences and columns
are alignment positions.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Read in the alignment for the gene XRCC2
read_alignment('XRCC2')

# Also read in the alignment for the gene XRCC2, but now by specifying
# directly the path to the file.
path &lt;- system.file("extdata", alignment_file("XRCC2"), package = "agvgd")
read_alignment(file = path)

</code></pre>


</div>