<div class="container">

<table style="width: 100%;"><tr>
<td>alakazam</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The Alakazam package</h2>

<h3>Description</h3>

<p><code>alakazam</code> in a member of the Immcantation framework of tools and serves five main 
purposes:
</p>

<ul>
<li>
<p>  Providing core functionality for other R packages in Immcantation. This
includes common tasks such as file I/O, basic DNA sequence manipulation, and
interacting with V(D)J segment and gene annotations.
</p>
</li>
<li>
<p>  Providing an R interface for interacting with the output of the pRESTO and 
Change-O tool suites.
</p>
</li>
<li>
<p>  Performing clonal abundance and diversity analysis on lymphocyte repertoires.
</p>
</li>
<li>
<p>  Performing lineage reconstruction on clonal populations of immunoglobulin 
(Ig) sequences.
</p>
</li>
<li>
<p>  Performing physicochemical property analyses of lymphocyte receptor sequences.
</p>
</li>
</ul>
<p>For additional details regarding the use of the <code>alakazam</code> package see the 
vignettes:<br><code>browseVignettes("alakazam")</code>
</p>


<h3>File I/O</h3>


<ul>
<li>  <p>readChangeoDb:        Input Change-O style files.
</p>
</li>
<li>  <p>writeChangeoDb:       Output Change-O style files.
</p>
</li>
</ul>
<h3>Sequence cleaning</h3>


<ul>
<li>  <p>maskSeqEnds:          Mask ragged ends.
</p>
</li>
<li>  <p>maskSeqGaps:          Mask gap characters.
</p>
</li>
<li>  <p>collapseDuplicates:   Remove duplicate sequences.
</p>
</li>
</ul>
<h3>Lineage reconstruction</h3>


<ul>
<li>  <p>makeChangeoClone:     Clean sequences for lineage reconstruction.
</p>
</li>
<li>  <p>buildPhylipLineage:   Perform lineage reconstruction of Ig sequences.
</p>
</li>
</ul>
<h3>Lineage topology analysis</h3>


<ul>
<li>  <p>tableEdges:           Tabulate annotation relationships over edges.
</p>
</li>
<li>  <p>testEdges:            Significance testing of annotation edges.
</p>
</li>
<li>  <p>testMRCA:             Significance testing of MRCA annotations.
</p>
</li>
<li>  <p>summarizeSubtrees:    Various summary statistics for subtrees.
</p>
</li>
<li>  <p>plotSubtrees:         Plot distributions of summary statistics 
for a population of trees.
</p>
</li>
</ul>
<h3>Diversity analysis</h3>


<ul>
<li>  <p>countClones:          Calculate clonal abundance.
</p>
</li>
<li>  <p>estimateAbundance:  	 Bootstrap clonal abundance curves.
</p>
</li>
<li>  <p>alphaDiversity:  	 Generate clonal alpha diversity curves.
</p>
</li>
<li>  <p>plotAbundanceCurve:   Plot clone size distribution as a rank-abundance 
</p>
</li>
<li>  <p>plotDiversityCurve:   Plot clonal diversity curves.
</p>
</li>
<li>  <p>plotDiversityTest:    Plot testing at given diversity hill indicex. 
</p>
</li>
</ul>
<h3>Ig and TCR sequence annotation</h3>


<ul>
<li>  <p>countGenes:           Calculate Ig and TCR allele, gene and family usage.
</p>
</li>
<li>  <p>extractVRegion:       Extract CDRs and FWRs sub-sequences.
</p>
</li>
<li>  <p>getAllele:            Get V(D)J allele names.
</p>
</li>
<li>  <p>getGene:              Get V(D)J gene names.
</p>
</li>
<li>  <p>getFamily:            Get V(D)J family names.
</p>
</li>
<li>  <p>junctionAlignment: Junction alignment properties 
</p>
</li>
</ul>
<h3>Sequence distance calculation</h3>


<ul>
<li>  <p>seqDist:        Calculate Hamming distance between two sequences.
</p>
</li>
<li>  <p>seqEqual:       Test two sequences for equivalence.
</p>
</li>
<li>  <p>pairwiseDist:   Calculate a matrix of pairwise Hamming distances for a 
set of sequences.
</p>
</li>
<li>  <p>pairwiseEqual:  Calculate a logical matrix of pairwise equivalence for a 
set of sequences.
</p>
</li>
</ul>
<h3>Amino acid propertes</h3>


<ul>
<li>  <p>translateDNA:         Translate DNA sequences to amino acid sequences.
</p>
</li>
<li>  <p>aminoAcidProperties:  Calculate various physicochemical properties of amino acid 
sequences.
</p>
</li>
<li>  <p>countPatterns:        Count patterns in sequences.
</p>
</li>
</ul>
<h3>References</h3>


<ol>
<li>
<p>  Vander Heiden JA, Yaari G, et al. pRESTO: a toolkit for processing 
high-throughput sequencing raw reads of lymphocyte receptor repertoires. 
Bioinformatics. 2014 30(13):1930-2.
</p>
</li>
<li>
<p>  Stern JNH, Yaari G, Vander Heiden JA, et al. B cells populating the multiple 
sclerosis brain mature in the draining cervical lymph nodes. 
Sci Transl Med. 2014 6(248):248ra107.
</p>
</li>
<li>
<p>  Wu Y-CB, et al. Influence of seasonal exposure to grass pollen on local and 
peripheral blood IgE repertoires in patients with allergic rhinitis. 
J Allergy Clin Immunol. 2014 134(3):604-12.
</p>
</li>
<li>
<p>  Gupta NT, Vander Heiden JA, et al. Change-O: a toolkit for analyzing 
large-scale B cell immunoglobulin repertoire sequencing data.
Bioinformatics. 2015 Oct 15;31(20):3356-8.
</p>
</li>
</ol>
</div>