<div class="container">

<table style="width: 100%;"><tr>
<td>readIgphyml</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read in output from IgPhyML</h2>

<h3>Description</h3>

<p><code>readIgphyml</code> reads output from the IgPhyML phylogenetics inference package for 
B cell repertoires
</p>


<h3>Usage</h3>

<pre><code class="language-R">readIgphyml(
  file,
  id = NULL,
  format = c("graph", "phylo"),
  collapse = FALSE,
  branches = c("mutations", "distance")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>IgPhyML output file (.tab).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>ID to assign to output object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>if <code>"graph"</code> return trees as igraph <code>graph</code> objects. 
if <code>"phylo"</code> return trees as ape <code>phylo</code> objects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>collapse</code></td>
<td>
<p>if <code>TRUE</code> transform branch lengths to units of substitutions, 
rather than substitutions per site, and collapse internal nodes
separated by branches &lt; 0.1 substitutions. Will also remove all
internal node labels, as it makes them inconsistent.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>branches</code></td>
<td>
<p>if <code>"distance"</code> branch lengths are in expected mutations per
site. If <code>"mutations"</code> branches are in expected mutations.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>readIgphyml</code> reads output from the IgPhyML repertoire phylogenetics inference package. 
The resulting object is divded between parameter estimates (usually under the HLP19 model),
which provide information about mutation and selection pressure operating on the sequences.
</p>
<p>Trees returned from this function are either igraph objects or phylo objects, and each may be 
visualized accordingly. Futher, branch lengths in tree may represent either the expected number of
substitutions per site (codon, if estimated under HLP or GY94 models), or the total number of 
expected substitutions per site. If the latter, internal nodes - but not tips - separated by branch
lengths less than 0.1 are collapsed to simplify viewing.
</p>


<h3>Value</h3>

<p>A list containing IgPhyML model parameters and estimated lineage trees. 
</p>
<p>Object attributes:
</p>

<ul>
<li>  <p><code>param</code>:     Data.frame of parameter estimates for each clonal 
lineage. Columns include: <code>CLONE</code>, which is the 
clone id; <code>NSEQ</code>, the total number of sequences in 
the lineage; <code>NSITE</code>, the number of codon sites;
<code>TREE_LENGTH</code>, the sum of all branch lengths in 
the estimated lineage tree; and <code>LHOOD</code>, the log 
likelihood of the clone's sequences given the tree and
parameters. Subsequent columns are parameter estimates 
from IgPhyML, which will depend on the model used. 
Parameter columns ending with <code>_MLE</code> are maximum 
likelihood estimates; those ending with <code>_LCI</code> are 
the lower 95
with <code>_UCI</code> are the upper 95
estimate. The first line of <code>param</code> is for clone 
<code>REPERTOIRE</code>, 
which is a summary of all lineages within the repertoire.
For this row, <code>NSEQ</code> is the total number of sequences, 
<code>NSITE</code> is the average number of sites, and
<code>TREE_LENGTH</code> is the mean tree length. For most 
applications, parameter values will be the same for all 
lineages within the repertoire, so access them simply by:
<code>&lt;object&gt;$param$OMEGA_CDR_MLE[1]</code> to, for instance,
get the estimate of dN/dS on the CDRs at the repertoire level.
</p>
</li>
<li>  <p><code>trees</code>:     List of tree objects estimated by IgPhyML. If 
<code>format="graph"</code> these are igraph <code>graph</code> objects. 
If <code>format="phylo"</code>, these are ape <code>phylo</code> objects.
</p>
</li>
<li>  <p><code>command</code>:   Command used to run IgPhyML.
</p>
</li>
</ul>
<h3>References</h3>


<ol>
<li>
<p>  Hoehn KB, Lunter G, Pybus OG - A Phylogenetic Codon Substitution Model for Antibody 
Lineages. Genetics 2017 206(1):417-427
https://doi.org/10.1534/genetics.116.196303 
</p>
</li>
<li>
<p>  Hoehn KB, Vander Heiden JA, Zhou JQ, Lunter G, Pybus OG, Kleinstein SHK - 
Repertoire-wide phylogenetic models of B cell molecular evolution reveal 
evolutionary signatures of aging and vaccination. bioRxiv 2019  
https://doi.org/10.1101/558825 
</p>
</li>
</ol>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
   # Read in and plot a tree from an igphyml run
   library(igraph)
   s1 &lt;- readIgphyml("IB+7d_lineages_gy.tsv_igphyml_stats_hlp.tab", id="+7d")
   print(s1$param$OMEGA_CDR_MLE[1])
   plot(s1$trees[[1]], layout=layout_as_tree, edge.label=E(s1$trees[[1]])$weight)

## End(Not run)

</code></pre>


</div>