<div class="container">

<table style="width: 100%;"><tr>
<td>readFastqDb</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Load sequencing quality scores from a FASTQ file</h2>

<h3>Description</h3>

<p><code>readFastqDb</code> adds the sequencing quality scores to a data.frame
from a FASTQ file. Matching is done by 'sequence_id'.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readFastqDb(
  data,
  fastq_file,
  quality_offset = -33,
  header = c("presto", "asis"),
  sequence_id = "sequence_id",
  sequence = "sequence",
  sequence_alignment = "sequence_alignment",
  v_cigar = "v_cigar",
  d_cigar = "d_cigar",
  j_cigar = "j_cigar",
  np1_length = "np1_length",
  np2_length = "np2_length",
  v_sequence_end = "v_sequence_end",
  d_sequence_end = "d_sequence_end",
  style = c("num", "ascii", "both"),
  quality_sequence = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p><code>data.frame</code> containing sequence data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fastq_file</code></td>
<td>
<p>path to the fastq file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quality_offset</code></td>
<td>
<p>offset value to be used by ape::read.fastq. It is 
the value to be added to the quality scores 
(the default -33 applies to the Sanger format and 
should work for most recent FASTQ files).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header</code></td>
<td>
<p>FASTQ file header format; one of <code>"presto"</code> or 
<code>"asis"</code>. Use <code>"presto"</code> to specify 
that the fastq file headers are using the pRESTO
format and can be parsed to extract 
the <code>sequence_id</code>. Use <code>"asis"</code> to skip 
any processing and use the sequence names as they are.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence_id</code></td>
<td>
<p>column in <code>data</code> that contains sequence 
identifiers to be matched to sequence identifiers in 
<code>fastq_file</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>column in <code>data</code> that contains sequence data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence_alignment</code></td>
<td>
<p>column in <code>data</code> that contains IMGT aligned sequence data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_cigar</code></td>
<td>
<p>column in <code>data</code> that contains CIGAR 
strings for the V gene alignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_cigar</code></td>
<td>
<p>column in <code>data</code> that contains CIGAR 
strings for the D gene alignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_cigar</code></td>
<td>
<p>column in <code>data</code> that contains CIGAR 
strings for the J gene alignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>np1_length</code></td>
<td>
<p>column in <code>data</code> that contains the number
of nucleotides between the V gene and first D gene 
alignments or between the V gene and J gene alignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>np2_length</code></td>
<td>
<p>column in <code>data</code> that contains the number
of nucleotides between either the first D gene and J 
gene alignments or the first D gene and second D gene
alignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_sequence_end</code></td>
<td>
<p>column in <code>data</code> that contains the 
end position of the V gene in <code>sequence</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_sequence_end</code></td>
<td>
<p>column in <code>data</code> that contains the 
end position of the D gene in <code>sequence</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>style</code></td>
<td>
<p>how the sequencing quality should be returned;
one of <code>"num"</code>, <code>"phred"</code>, or <code>"both"</code>.
Specify <code>"num"</code> to store the quality scores as strings of 
comma separated numeric values. Use <code>"phred"</code> to have
the function return the scores as Phred (ASCII) scores. 
Use <code>"both"</code> to retrieve both.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quality_sequence</code></td>
<td>
<p>specify <code>TRUE</code> to keep the quality scores for 
<code>sequence</code>. If false, only the quality score
for <code>sequence_alignment</code> will be added to <code>data</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Modified <code>data</code> with additional fields:
</p>

<ol>
<li> <p><code>quality_alignment</code>:     A character vector with ASCII Phred 
scores for <code>sequence_alignment</code>.
</p>
</li>
<li> <p><code>quality_alignment_num</code>: A character vector, with comma separated 
numerical quality values for each 
position in <code>sequence_alignment</code>.
</p>
</li>
<li> <p><code>quality</code>:      A character vector with ASCII Phred 
scores for <code>sequence</code>.
</p>
</li>
<li> <p><code>quality_num</code>:  A character vector, with comma separated 
numerical quality values for each 
position in <code>sequence</code>.
</p>
</li>
</ol>
<h3>See Also</h3>

<p>maskPositionsByQuality and getPositionQuality
</p>


<h3>Examples</h3>

<pre><code class="language-R">db &lt;- airr::read_rearrangement(system.file("extdata", "example_quality.tsv", package="alakazam"))
fastq_file &lt;- system.file("extdata", "example_quality.fastq", package="alakazam")
db &lt;- readFastqDb(db, fastq_file, quality_offset=-33)

</code></pre>


</div>