<div class="container">

<table style="width: 100%;"><tr>
<td>junctionAlignment</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate junction region alignment properties</h2>

<h3>Description</h3>

<p><code>junctionAlignment</code> determines the number of deleted germline nucleotides in the 
junction region and the number of V gene and J gene nucleotides in the CDR3.
</p>


<h3>Usage</h3>

<pre><code class="language-R">junctionAlignment(
  data,
  germline_db,
  v_call = "v_call",
  d_call = "d_call",
  j_call = "j_call",
  v_germline_start = "v_germline_start",
  v_germline_end = "v_germline_end",
  d_germline_start = "d_germline_start",
  d_germline_end = "d_germline_end",
  j_germline_start = "j_germline_start",
  j_germline_end = "j_germline_end",
  np1_length = "np1_length",
  np2_length = "np2_length",
  junction = "junction",
  junction_length = "junction_length",
  sequence_alignment = "sequence_alignment"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p><code>data.frame</code> containing sequence data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>germline_db</code></td>
<td>
<p>reference germline database for the V, D and J genes.
in <code>data</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_call</code></td>
<td>
<p>V gene assignment column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_call</code></td>
<td>
<p>D gene assignment column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_call</code></td>
<td>
<p>J gene assignment column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_germline_start</code></td>
<td>
<p>column containing the start position of the alignment 
in the V reference germline.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_germline_end</code></td>
<td>
<p>column containing the end position of the alignment in the 
V reference germline.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_germline_start</code></td>
<td>
<p>column containing the start position of the alignment 
in the D reference germline.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_germline_end</code></td>
<td>
<p>column containing the start position of the alignment 
in the D reference germline.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_germline_start</code></td>
<td>
<p>column containing the start position of the alignment 
in the J reference germline.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_germline_end</code></td>
<td>
<p>column containing the start position of the alignment 
in the J reference germline.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>np1_length</code></td>
<td>
<p>combined length of the N and P regions between the 
V and D regions (heavy chain) or V and J regions (light chain).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>np2_length</code></td>
<td>
<p>combined length of the N and P regions between the 
D and J regions (heavy chain).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junction</code></td>
<td>
<p>column containing the junction sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junction_length</code></td>
<td>
<p>column containing the length of the junction region in nucleotides.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence_alignment</code></td>
<td>
<p>column containing the aligned sequence.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A modified input <code>data.frame</code> with the following additional columns storing 
junction alignment information:
</p>

<ol>
<li>  <p><code>e3v_length</code>:     number of 3' V germline nucleotides deleted.
</p>
</li>
<li>  <p><code>e5d_length</code>:     number of 5' D germline nucleotides deleted.
</p>
</li>
<li>  <p><code>e3d_length</code>:     number of 3' D germline nucleotides deleted.
</p>
</li>
<li>  <p><code>e5j_length</code>:     number of 5' J germline nucleotides deleted.
</p>
</li>
<li>  <p><code>v_cdr3_length</code>:  number of sequence_alignment V nucleotides in the CDR3.
</p>
</li>
<li>  <p><code>j_cdr3_length</code>:  number of sequence_alignment J nucleotides in the CDR3.
</p>
</li>
</ol>
<h3>Examples</h3>

<pre><code class="language-R">germline_db &lt;- list(
"IGHV3-11*05"="CAGGTGCAGCTGGTGGAGTCTGGGGGA...GGCTTGGTCAAGCCTGGAGGGTCCCTGAGACT
CTCCTGTGCAGCCTCTGGATTCACCTTC............AGTGACTACTACATGAGCTGGATCCGCCAGGCTCCAG
GGAAGGGGCTGGAGTGGGTTTCATACATTAGTAGTAGT......AGTAGTTACACAAACTACGCAGACTCTGTGAAG
...GGCCGATTCACCATCTCCAGAGACAACGCCAAGAACTCACTGTATCTGCAAATGAACAGCCTGAGAGCCGAGGA
CACGGCCGTGTATTACTGTGCGAGAGA",
"IGHD3-10*01"="GTATTACTATGGTTCGGGGAGTTATTATAAC",
"IGHJ5*02"="ACAACTGGTTCGACCCCTGGGGCCAGGGAACCCTGGTCACCGTCTCCTCAG"
)

db &lt;- junctionAlignment(SingleDb, germline_db)

</code></pre>


</div>