<div class="container">

<table style="width: 100%;"><tr>
<td>seqDist</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate distance between two sequences</h2>

<h3>Description</h3>

<p><code>seqDist</code> calculates the distance between two DNA sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">seqDist(seq1, seq2, dist_mat = getDNAMatrix())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq1</code></td>
<td>
<p>character string containing a DNA sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq2</code></td>
<td>
<p>character string containing a DNA sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist_mat</code></td>
<td>
<p>Character distance matrix. Defaults to a Hamming distance 
matrix returned by getDNAMatrix. If gap 
characters, <code>c("-", ".")</code>, are assigned a value of -1 
in <code>dist_mat</code> then contiguous gaps of any run length,
which are not present in both sequences, will be counted as a 
distance of 1. Meaning, indels of any length will increase
the sequence distance by 1. Gap values other than -1 will 
return a distance that does not consider indels as a special case.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Numerical distance between <code>seq1</code> and <code>seq2</code>.
</p>


<h3>See Also</h3>

<p>Nucleotide distance matrix may be built with 
getDNAMatrix. Amino acid distance matrix may be built
with getAAMatrix. Used by pairwiseDist for generating
distance matrices. See seqEqual for testing sequence equivalence.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Ungapped examples
seqDist("ATGGC", "ATGGG")
seqDist("ATGGC", "ATG??")

# Gaps will be treated as Ns with a gap=0 distance matrix
seqDist("ATGGC", "AT--C", dist_mat=getDNAMatrix(gap=0))

# Gaps will be treated as universally non-matching characters with gap=1
seqDist("ATGGC", "AT--C", dist_mat=getDNAMatrix(gap=1))

# Gaps of any length will be treated as single mismatches with a gap=-1 distance matrix
seqDist("ATGGC", "AT--C", dist_mat=getDNAMatrix(gap=-1))

# Gaps of equivalent run lengths are not counted as gaps
seqDist("ATG-C", "ATG-C", dist_mat=getDNAMatrix(gap=-1))

# Overlapping runs of gap characters are counted as a single gap
seqDist("ATG-C", "AT--C", dist_mat=getDNAMatrix(gap=-1))
seqDist("A-GGC", "AT--C", dist_mat=getDNAMatrix(gap=-1))
seqDist("AT--C", "AT--C", dist_mat=getDNAMatrix(gap=-1))

# Discontiguous runs of gap characters each count as separate gaps
seqDist("-TGGC", "AT--C", dist_mat=getDNAMatrix(gap=-1))

</code></pre>


</div>