<div class="container">

<table style="width: 100%;"><tr>
<td>buildPhylipLineage</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Infer an Ig lineage using PHYLIP</h2>

<h3>Description</h3>

<p><code>buildPhylipLineage</code> reconstructs an Ig lineage via maximum parsimony using the 
dnapars application, or maximum liklihood using the dnaml application of the PHYLIP package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">buildPhylipLineage(
  clone,
  phylip_exec,
  dist_mat = getDNAMatrix(gap = 0),
  rm_temp = FALSE,
  verbose = FALSE,
  temp_path = NULL,
  onetree = FALSE,
  branch_length = c("mutations", "distance")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>clone</code></td>
<td>
<p>ChangeoClone object containing clone data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phylip_exec</code></td>
<td>
<p>absolute path to the PHYLIP dnapars executable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist_mat</code></td>
<td>
<p>character distance matrix to use for reassigning edge weights. 
Defaults to a Hamming distance matrix returned by getDNAMatrix 
with <code>gap=0</code>. If gap characters, <code>c("-", ".")</code>, are assigned 
a value of -1 in <code>dist_mat</code> then contiguous gaps of any run length,
which are not present in both sequences, will be counted as a 
distance of 1. Meaning, indels of any length will increase
the sequence distance by 1. Gap values other than -1 will 
return a distance that does not consider indels as a special case.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rm_temp</code></td>
<td>
<p>if <code>TRUE</code> delete the temporary directory after running dnapars;
if <code>FALSE</code> keep the temporary directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>if <code>FALSE</code> suppress the output of dnapars; 
if <code>TRUE</code> STDOUT and STDERR of dnapars will be passed to 
the console.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>temp_path</code></td>
<td>
<p>specific path to temp directory if desired.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>onetree</code></td>
<td>
<p>if <code>TRUE</code> save only one tree.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>branch_length</code></td>
<td>
<p>specifies how to define branch lengths; one of <code>"mutations"</code> 
or <code>"distance"</code>. If set to <code>"mutations"</code> (default), then branch 
lengths represent the number of mutations between nodes. 
If set to <code>"distance"</code>, then branch lengths represent
the expected number of mutations per site, unaltered from PHYLIP output.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>buildPhylipLineage</code> builds the lineage tree of a set of unique Ig sequences via
maximum parsimony through an external call to the dnapars application of the PHYLIP
package. dnapars is called with default algorithm options, except for the search option, 
which is set to "Rearrange on one best tree". The germline sequence of the clone is used 
for the outgroup. 
</p>
<p>Following tree construction using dnapars, the dnapars output is modified to allow
input sequences to appear as internal nodes of the tree. Intermediate sequences 
inferred by dnapars are replaced by children within the tree having a Hamming distance 
of zero from their parent node. With the default <code>dist_mat</code>, the distance calculation 
allows IUPAC ambiguous character matches, where an ambiguous character has distance zero 
to any character in the set of characters it represents. Distance calculation and movement of 
child nodes up the tree is repeated until all parent-child pairs have a distance greater than zero 
between them. The germline sequence (outgroup) is moved to the root of the tree and
excluded from the node replacement processes, which permits the trunk of the tree to be
the only edge with a distance of zero. Edge weights of the resultant tree are assigned 
as the distance between each sequence.
</p>


<h3>Value</h3>

<p>An igraph <code>graph</code> object defining the Ig lineage tree. Each unique input 
sequence in <code>clone</code> is a vertex of the tree, with additional vertices being
either the germline (root) sequences or inferred intermediates. The <code>graph</code> 
object has the following attributes.
</p>
<p>Vertex attributes:
</p>

<ul>
<li>  <p><code>name</code>:      value in the <code>sequence_id</code> column of the <code>data</code> 
slot of the input <code>clone</code> for observed sequences. 
The germline (root) vertex is assigned the name 
"Germline" and inferred intermediates are assigned
names with the format "Inferred1", "Inferred2", ....
</p>
</li>
<li>  <p><code>sequence</code>:  value in the <code>sequence</code> column of the <code>data</code> 
slot of the input <code>clone</code> for observed sequences.
The germline (root) vertex is assigned the sequence
in the <code>germline</code> slot of the input <code>clone</code>.
The sequence of inferred intermediates are extracted
from the dnapars output.
</p>
</li>
<li>  <p><code>label</code>:     same as the <code>name</code> attribute.
</p>
</li>
</ul>
<p>Additionally, each other column in the <code>data</code> slot of the input 
<code>clone</code> is added as a vertex attribute with the attribute name set to 
the source column name. For the germline and inferred intermediate vertices,
these additional vertex attributes are all assigned a value of <code>NA</code>.
</p>
<p>Edge attributes:
</p>

<ul>
<li>  <p><code>weight</code>:    Hamming distance between the <code>sequence</code> attributes
of the two vertices.
</p>
</li>
<li>  <p><code>label</code>:     same as the <code>weight</code> attribute.
</p>
</li>
</ul>
<p>Graph attributes:
</p>

<ul>
<li>  <p><code>clone</code>:     clone identifier from the <code>clone</code> slot of the
input <code>ChangeoClone</code>.
</p>
</li>
<li>  <p><code>v_gene</code>:    V-segment gene call from the <code>v_gene</code> slot of 
the input <code>ChangeoClone</code>.
</p>
</li>
<li>  <p><code>j_gene</code>:    J-segment gene call from the <code>j_gene</code> slot of 
the input <code>ChangeoClone</code>.
</p>
</li>
<li>  <p><code>junc_len</code>:  junction length (nucleotide count) from the 
<code>junc_len</code> slot of the input <code>ChangeoClone</code>.
</p>
<p>Alternatively, this function will return an <code>phylo</code> object, which is compatible
with the ape package. This object will contain reconstructed ancestral sequences in
<code>nodes</code> attribute.
</p>
</li>
</ul>
<h3>References</h3>


<ol>
<li>
<p>  Felsenstein J. PHYLIP - Phylogeny Inference Package (Version 3.2). 
Cladistics. 1989 5:164-166.
</p>
</li>
<li>
<p>  Stern JNH, Yaari G, Vander Heiden JA, et al. B cells populating the multiple 
sclerosis brain mature in the draining cervical lymph nodes. 
Sci Transl Med. 2014 6(248):248ra107.
</p>
</li>
</ol>
<h3>See Also</h3>

<p>Takes as input a ChangeoClone. 
Temporary directories are created with makeTempDir.
Distance is calculated using seqDist. 
See [igraph](http://www.rdocumentation.org/packages/igraph/topics/aaa-igraph-package) 
and [igraph.plotting](http://www.rdocumentation.org/packages/igraph/topics/plot.common) 
for working  with igraph <code>graph</code> objects.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Preprocess clone
db &lt;- subset(ExampleDb, clone_id == 3138)
clone &lt;- makeChangeoClone(db, text_fields=c("sample_id", "c_call"), 
                          num_fields="duplicate_count")

# Run PHYLIP and process output
phylip_exec &lt;- "~/apps/phylip-3.695/bin/dnapars"
graph &lt;- buildPhylipLineage(clone, phylip_exec, rm_temp=TRUE)

# Plot graph with a tree layout
library(igraph)
plot(graph, layout=layout_as_tree, vertex.label=V(graph)$c_call, 
     vertex.size=50, edge.arrow.mode=0, vertex.color="grey80")

# To consider each indel event as a mutation, change the masking character 
# and distance matrix
clone &lt;- makeChangeoClone(db, text_fields=c("sample_id", "c_call"), 
                          num_fields="duplicate_count", mask_char="-")
graph &lt;- buildPhylipLineage(clone, phylip_exec, dist_mat=getDNAMatrix(gap=-1), 
                            rm_temp=TRUE)

## End(Not run)

</code></pre>


</div>