<div class="container">

<table style="width: 100%;"><tr>
<td>allelematch-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Identification of unique multilocus genotypes
</h2>

<h3>Description</h3>

<p>Tools for the identification of unique of multilocus genotypes when both genotyping error and 
missing data may be present; targeted for use with large datasets and databases containing
multiple samples of each individual (a common situation in conservation genetics, particularly
in non-invasive wildlife sampling applications). Functions explicitly incorporate missing data
and can tolerate allele mismatches created by genotyping error.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
		Package: </td>
<td style="text-align: left;"> allelematch</td>
</tr>
<tr>
<td style="text-align: left;">
		Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
		Version: </td>
<td style="text-align: left;"> 2.5.2</td>
</tr>
<tr>
<td style="text-align: left;">
		Date: </td>
<td style="text-align: left;"> 2023-05-18</td>
</tr>
<tr>
<td style="text-align: left;">
		License: </td>
<td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
<td style="text-align: left;">
		Requires: </td>
<td style="text-align: left;"> dynamicTreeCut</td>
</tr>
<tr>
<td style="text-align: left;">
		LazyLoad: </td>
<td style="text-align: left;"> yes</td>
</tr>
<tr>
<td style="text-align: left;">
	</td>
</tr>
</table>
<p>Supplementary documentation describing the operation of the software in detail and illustrating the
use of the software using tutorials is available as a vignette.
It is installed with the package and linked from the package index help page. An online version is
also available via the Data S1 Supplementary documentation and tutorials (PDF) located at
<a href="https://doi.org/10.1111/j.1755-0998.2012.03137.x">doi:10.1111/j.1755-0998.2012.03137.x</a>.<br></p>
<p>Simulations examining the performance of these tools have also been performed, and results are
available in the publication associated with this package. Please refer to the publication:<br></p>
<p>Galpern P, Manseau, M, Hettinga P, Smith K, and Wilson P. (2012) allelematch: an R package for
identifying unique multilocus genotypes where genotype error and missing data may be present.
<em>Molecular Ecology Resources</em> 12:771-778.<br></p>
<p>Use <code>citation("allelematch")</code> for the full citation. Please also use this publication when
citing the package.<br></p>
<p>An important core element of the package is dynamic tree cutting, and this is made possible via the
cutreeHybrid function within the dynamicTreeCut package for R (Langfelder et al., 2008). 
</p>


<h3>Author(s)</h3>

<p>Paul Galpern (<a href="mailto:pgalpern@gmail.com">pgalpern@gmail.com</a>)
</p>


<h3>References</h3>

<p>Langfelder P, Zhang B, Horvath S. (2008) Defining clusters from a hierarchical cluster tree: the
Dynamic Tree Cut package for R. <em>Bioinformatics</em>, 24, 719
</p>


</div>