<div class="container">

<table style="width: 100%;"><tr>
<td>alleImputer</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Imputation of missing alleles from a dataset composed by families.</h2>

<h3>Description</h3>

<p>By analyzing all possible combinations of a parent-offspring pedigree in which parental and/or offspring genotypes may be missing; as long as one child was genotyped, in certain cases it is possible an unequivocal imputation of the missing genotypes both in parents and children.
</p>


<h3>Usage</h3>

<pre><code class="language-R">alleImputer(data, invisibleOutput = TRUE, dataSummary = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Data containing the families' identifiers and the corresponding genetic data (or the path of the PED file).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>invisibleOutput</code></td>
<td>
<p>Data are not shown by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dataSummary</code></td>
<td>
<p>A summary of the data is shown by default.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Imputed markers, Homozygosity (HMZ) matrix, marker messages and number of unique alleles per marker.
</p>


<h3>References</h3>

<p>Medina-Rodriguez, N. Santana A. et al. (2014) alleHap: an efficient algorithm to reconstruct zero-recombinant haplotypes from parent-offspring pedigrees. BMC Bioinformatics, 15, A6 (S-3).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Imputation of families containing parental missing data
simulatedFams &lt;- alleSimulator(10,4,6,missParProb=0.2) 
famsAlls &lt;- simulatedFams[[1]]       # Original data 
alleImputer(famsAlls)                # Imputed alleles (genotypes)

## Imputation of families containing offspring missing data
datasetAlls &lt;- alleSimulator(10,4,6,missOffProb=0.2)
famsAlls &lt;- simulatedFams[[1]]       # Original data 
alleImputer(famsAlls)                # Imputed alleles (genotypes)

## Imputation of a family marker containing missing values in one parent and one child
infoFam &lt;- data.frame(famID="FAM03",indID=1:5,patID=c(0,0,1,1,1),
                      matID=c(0,0,2,2,2),sex=c(1,2,1,2,1),phenot=0)
mkr &lt;- rbind(father=c(NA,NA),mother=c(1,3),child1=c(1,1),child2=c(2,3),child3=c(NA,NA))
colnames(mkr) &lt;- c("Mkr1_1","Mkr1_2")
famMkr &lt;- cbind(infoFam,mkr)         # Original data 
alleImputer(famMkr)                  # Imputed alleles (genotypes)

</code></pre>


</div>