<div class="container">

<table style="width: 100%;"><tr>
<td>GenAnno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate Genotype Annotations</h2>

<h3>Description</h3>

<p>Returns a vector of length = the number of columns (SNPs) in the genotype
matrix. Each SNP is classified as a benign missense variant (0), a
deleterious missense variant (1), or a protein truncating variant (2).
</p>


<h3>Usage</h3>

<pre><code class="language-R">GenAnno(snps, p_dmv = 0.33, p_ptv = 0.33)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>snps</code></td>
<td>
<p>Number of SNPs in the gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_dmv</code></td>
<td>
<p>Frequency of deleterious missense variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_ptv</code></td>
<td>
<p>Frequency of protein truncating variants.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>(snps x 1) integer vector.
</p>


</div>