<div class="container">

<table style="width: 100%;"><tr>
<td>COAST</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>COding-variant Allelic Series Test</h2>

<h3>Description</h3>

<p>Main allelic series test. Performs both Burden and SKAT type tests, then
combines the results to calculate an omnibus p-value.
</p>


<h3>Usage</h3>

<pre><code class="language-R">COAST(
  anno,
  geno,
  pheno,
  apply_int = TRUE,
  covar = NULL,
  include_orig_skato_all = FALSE,
  include_orig_skato_ptv = FALSE,
  is_pheno_binary = FALSE,
  return_omni_only = FALSE,
  score_test = FALSE,
  weights = DEFAULT_WEIGHTS
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>anno</code></td>
<td>
<p>(snps x 1) annotation vector with values in c(0, 1, 2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>(n x 1) phenotype vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>apply_int</code></td>
<td>
<p>Apply rank-based inverse normal transform to the phenotype?
Default: TRUE. Ignored if phenotype is binary.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covar</code></td>
<td>
<p>(n x p) covariate matrix. Defaults to an (n x 1) intercept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include_orig_skato_all</code></td>
<td>
<p>Include the original version of SKAT-O applied
to all variants in the omnibus test? Default: FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include_orig_skato_ptv</code></td>
<td>
<p>Include the original version of SKAT-O applied
to PTV variants only in the omnibus test? Default: FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_pheno_binary</code></td>
<td>
<p>Is the phenotype binary? Default: FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_omni_only</code></td>
<td>
<p>Return only the omnibus p-value? Default: FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>score_test</code></td>
<td>
<p>Use a score test for burden analysis? If FALSE, uses a
Wald test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>(3 x 1) annotation category weights.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Numeric p-value.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Generate data.
data &lt;- DGP(n = 1e3, snps = 1e2)

# Run the COding-variant Allelic Series Test.
results &lt;- COAST(
  anno = data$anno,
  geno = data$geno,
  pheno = data$pheno,
  covar = data$covar
)
show(results)

</code></pre>


</div>