<div class="container">

<table style="width: 100%;"><tr>
<td>alleLoader</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Data loading of nuclear families (in .ped format)</h2>

<h3>Description</h3>

<p>The data to be loaded must be structured in .ped format and families must comprise by parent-offspring pedigrees.
</p>


<h3>Usage</h3>

<pre><code class="language-R">alleLoader(data, invisibleOutput = TRUE, dataSummary = TRUE,
  missingValues = c(-9, -99))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Data to be loaded.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>invisibleOutput</code></td>
<td>
<p>Data are not shown by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dataSummary</code></td>
<td>
<p>A summary of the data is shown by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missingValues</code></td>
<td>
<p>Specification of the character/numerical values which may be missing.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Loaded dataset.
</p>


<h3>References</h3>

<p>Medina-Rodriguez, N. Santana A. et al. (2014) alleHap: an efficient algorithm to reconstruct zero-recombinant haplotypes from parent-offspring pedigrees. BMC Bioinformatics, 15, A6 (S-3).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Loading of a dataset in .ped format with alphabetical alleles (A,C,G,T)
example1 &lt;- file.path(find.package("alleHap"), "examples", "example1.ped")
example1Alls &lt;- alleLoader(example1)
head(example1Alls)

## Loading of a dataset in .ped format with numerical alleles
example2 &lt;- file.path(find.package("alleHap"), "examples", "example2.ped")
example2Alls &lt;- alleLoader(example2)
head(example2Alls)

</code></pre>


</div>