<div class="container">

<table style="width: 100%;"><tr>
<td>ASKAT</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Allelic Series SKAT Test</h2>

<h3>Description</h3>

<p>Sequence kernel association test (SKAT) with allelic series weights.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ASKAT(
  anno,
  geno,
  pheno,
  apply_int = TRUE,
  covar = NULL,
  is_pheno_binary = FALSE,
  return_null_model = FALSE,
  weights = DEFAULT_WEIGHTS
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>anno</code></td>
<td>
<p>(snps x 1) annotation vector with values in c(0, 1, 2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>(n x snps) genotype matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>(n x 1) phenotype vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>apply_int</code></td>
<td>
<p>Apply rank-based inverse normal transform to the phenotype?
Default: TRUE. Ignored if phenotype is binary.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covar</code></td>
<td>
<p>(n x p) covariate matrix. Defaults to an (n x 1) intercept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_pheno_binary</code></td>
<td>
<p>Is the phenotype binary? Default: FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_null_model</code></td>
<td>
<p>Return the null model in addition to the p-value?
Useful if running additional SKAT tests. Default: FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>(3 x 1) annotation category weights.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If <code>return_null_model</code>, a list containing the p-value and the
SKAT null model. Otherwise, a numeric p-value.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Generate data.
data &lt;- DGP(n = 1e3, snps = 1e2)

# Run the Allelic Series SKAT Test.
# Note: the output is a scalar p-value.
results &lt;- ASKAT(
  anno = data$anno,
  geno = data$geno,
  pheno = data$pheno,
  covar = data$covar
)
</code></pre>


</div>