<div class="container">

<table style="width: 100%;"><tr>
<td>DGP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Data Generating Process</h2>

<h3>Description</h3>

<p>Generate a data set consisting of: </p>

<ul>
<li>
<p>"anno"A SNP-length annotation vector.
</p>
</li>
<li>
<p>"covar"A subject by 6 covariate matrix.
</p>
</li>
<li>
<p>"geno"A subject by SNP genotype matrix.
</p>
</li>
<li>
<p>"pheno"A subject-length phenotype vector.
</p>
</li>
</ul>
<h3>Usage</h3>

<pre><code class="language-R">DGP(
  anno = NULL,
  beta = c(0, 1, 2),
  binary = FALSE,
  geno = NULL,
  include_residual = TRUE,
  indicator = FALSE,
  maf_range = c(0.005, 0.01),
  method = "none",
  n = 100,
  p_dmv = 0.4,
  p_ptv = 0.1,
  prop_causal = 1,
  random_signs = FALSE,
  random_var = 0,
  snps = 100,
  weights = c(1, 2, 3)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>anno</code></td>
<td>
<p>Annotation vector, if providing genotypes. Should match the
number of columns in geno.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p>If method = "none", a (3 x 1) coefficient vector for bmvs, dmvs,
and ptvs respectively. If method != "none", a scalar effect size.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>binary</code></td>
<td>
<p>Generate binary phenotype? Default: FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>Genotype matrix, if providing genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include_residual</code></td>
<td>
<p>Include residual? If FALSE, returns the expected
value. Intended for testing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indicator</code></td>
<td>
<p>Convert raw counts to indicators? Default: FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf_range</code></td>
<td>
<p>Range of minor allele frequencies: c(MIN, MAX).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Genotype aggregation method. Default: "none".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>Sample size.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_dmv</code></td>
<td>
<p>Frequency of deleterious missense variants. Default of 40% is
based on the frequency of DMVs among rare coding variants in the UK
Biobank.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_ptv</code></td>
<td>
<p>Frequency of protein truncating variants. Default of 10% is
based on the frequency of PTVs among rare coding variants in the UK
Biobank.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop_causal</code></td>
<td>
<p>Proportion of variants which are causal. Default: 1.0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>random_signs</code></td>
<td>
<p>Randomize signs? FALSE for burden-type genetic
architecture, TRUE for SKAT-type.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>random_var</code></td>
<td>
<p>Frailty variance in the case of random signs. Default: 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snps</code></td>
<td>
<p>Number of SNP in the gene. Default: 100.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>Aggregation weights.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List containing: genotypes, annotations, covariates, phenotypes.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Generate data.
data &lt;- DGP(n = 100)

# View components.
table(data$anno)
head(data$covar)
head(data$geno[, 1:5])
hist(data$pheno)
</code></pre>


</div>