<div class="container">

<table style="width: 100%;"><tr>
<td>runMacs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create founder haplotypes using MaCS</h2>

<h3>Description</h3>

<p>Uses the MaCS software to produce founder haplotypes
(Chen et al. 2009).
</p>


<h3>Usage</h3>

<pre><code class="language-R">runMacs(
  nInd,
  nChr = 1,
  segSites = NULL,
  inbred = FALSE,
  species = "GENERIC",
  split = NULL,
  ploidy = 2L,
  manualCommand = NULL,
  manualGenLen = NULL,
  nThreads = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nInd</code></td>
<td>
<p>number of individuals to simulate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nChr</code></td>
<td>
<p>number of chromosomes to simulate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segSites</code></td>
<td>
<p>number of segregating sites to keep per chromosome. A
value of NULL results in all sites being retained.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inbred</code></td>
<td>
<p>should founder individuals be inbred</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>species history to simulate. See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split</code></td>
<td>
<p>an optional historic population split in terms of generations ago.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>ploidy level of organism</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>manualCommand</code></td>
<td>
<p>user provided MaCS options. For advanced users only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>manualGenLen</code></td>
<td>
<p>user provided genetic length. This must be supplied if using
manualCommand. If not using manualCommand, this value will replace the predefined
genetic length for the species. However, this the genetic length is only used by
AlphaSimR and is not passed to MaCS, so MaCS still uses the predefined genetic length.
For advanced users only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nThreads</code></td>
<td>
<p>if OpenMP is available, this will allow for simulating chromosomes in parallel.
If the value is NULL, the number of threads is automatically detected.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>There are currently three species histories available: GENERIC, CATTLE, WHEAT, and MAIZE.
</p>
<p>The GENERIC history is meant to be a reasonable all-purpose choice. It runs quickly and
models a population with an effective populations size that has gone through several historic
bottlenecks. This species history is used as the default arguments in the <code>runMacs2</code>
function, so the user should examine this function for the details of how the species is modeled.
</p>
<p>The CATTLE history is based off of real genome sequence data (MacLeod et al. 2013).
</p>
<p>The WHEAT (Gaynor et al. 2017) and MAIZE (Hickey et al. 2014)
histories have been included due to their use in previous simulations. However, it should
be noted that neither faithfully simulates its respective species. This is apparent by
the low number of segregating sites simulated by each history relative to their real-world
analogs. Adjusting these histories to better represent their real-world analogs would result
in a drastic increase to runtime.
</p>


<h3>Value</h3>

<p>an object of <code>MapPop-class</code>
</p>


<h3>References</h3>

<p>Chen GK, Marjoram P, Wall JD (2009).
“Fast and Flexible Simulation of DNA Sequence Data.”
<em>Genome Research</em>, <b>19</b>, 136-142.
<a href="https://genome.cshlp.org/content/19/1/136">https://genome.cshlp.org/content/19/1/136</a>.<br><br> Gaynor RC, Gorjanc G, Bentley AR, Ober ES, Howell P, Jackson R, Mackay IJ, Hickey JM (2017).
“A Two-Part Strategy for Using Genomic Selection to Develop Inbred Lines.”
<em>Crop Science</em>, <b>57</b>(5), 2372–2386.
ISSN 0011-183X, <a href="https://doi.org/10.2135/cropsci2016.09.0742">doi:10.2135/cropsci2016.09.0742</a>, <a href="https://acsess.onlinelibrary.wiley.com/doi/full/10.2135/cropsci2016.09.0742">https://acsess.onlinelibrary.wiley.com/doi/full/10.2135/cropsci2016.09.0742</a>.<br><br> Hickey JMDS, Crossa J, Hearne S, Babu R, Prasanna BM, Grondona M, Zambelli A, Windhausen VS, Mathews K, Gorjanc G (2014).
“Evaluation of Genomic Selection Training Population Designs and Genotyping Strategies in Plant Breeding Programs Using Simulation.”
<em>Crop Science</em>, <b>54</b>(4), 1476-1488.
<a href="https://doi.org/10.2135/cropsci2013.03.0195">doi:10.2135/cropsci2013.03.0195</a>.<br><br> MacLeod IM, Larkin DM, Lewin HAHBJ, Goddard ME (2013).
“Inferring Demography from Runs of Homozygosity in Whole-Genome Sequence, with Correction for Sequence Errors.”
<em>Molecular Biology and Evolution</em>, <b>30</b>(9), 2209–2223.
<a href="https://doi.org/10.1093/molbev/mst125">doi:10.1093/molbev/mst125</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Creates a populations of 10 outbred individuals
# Their genome consists of 1 chromosome and 100 segregating sites
## Not run: 
founderPop = runMacs(nInd=10,nChr=1,segSites=100)

## End(Not run)
</code></pre>


</div>