<div class="container">

<table style="width: 100%;"><tr>
<td>quickHaplo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Quick founder haplotype simulation</h2>

<h3>Description</h3>

<p>Rapidly simulates founder haplotypes by randomly
sampling 0s and 1s. This is equivalent to having all loci with
allele frequency 0.5 and being in linkage equilibrium.
</p>


<h3>Usage</h3>

<pre><code class="language-R">quickHaplo(nInd, nChr, segSites, genLen = 1, ploidy = 2L, inbred = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nInd</code></td>
<td>
<p>number of individuals to simulate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nChr</code></td>
<td>
<p>number of chromosomes to simulate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segSites</code></td>
<td>
<p>number of segregating sites per chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genLen</code></td>
<td>
<p>genetic length of chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>ploidy level of organism</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inbred</code></td>
<td>
<p>should founder individuals be inbred</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>an object of <code>MapPop-class</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Creates a populations of 10 outbred individuals
# Their genome consists of 1 chromosome and 100 segregating sites
founderPop = quickHaplo(nInd=10,nChr=1,segSites=100)

</code></pre>


</div>