<div class="container">

<table style="width: 100%;"><tr>
<td>pullSegSiteGeno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pull segregating site genotypes</h2>

<h3>Description</h3>

<p>Retrieves genotype data for all segregating sites
</p>


<h3>Usage</h3>

<pre><code class="language-R">pullSegSiteGeno(pop, chr = NULL, asRaw = FALSE, simParam = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>an object of <code>RawPop-class</code> or
<code>MapPop-class</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>a vector of chromosomes to retrieve. If NULL,
all chromosome are retrieved.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>asRaw</code></td>
<td>
<p>return in raw (byte) format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParam</code></td>
<td>
<p>an object of <code>SimParam</code>, not 
used if pop is <code>MapPop-class</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a matrix of genotypes
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Create founder haplotypes
founderPop = quickHaplo(nInd=10, nChr=1, segSites=15)

#Set simulation parameters
SP = SimParam$new(founderPop)
SP$addTraitA(10)
SP$addSnpChip(5)

#Create population
pop = newPop(founderPop, simParam=SP)
pullSegSiteGeno(pop, simParam=SP)

</code></pre>


</div>