<div class="container">

<table style="width: 100%;"><tr>
<td>importHaplo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Import haplotypes</h2>

<h3>Description</h3>

<p>Formats haplotype in a matrix format to an 
AlphaSimR population that can be used to 
initialize a simulation. This function serves 
as wrapper for <code>newMapPop</code> that 
utilizes a more user friendly input format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">importHaplo(haplo, genMap, ploidy = 2L, ped = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>haplo</code></td>
<td>
<p>a matrix of haplotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genMap</code></td>
<td>
<p>genetic map as a data.frame. The first
three columns must be: marker name, chromosome, and 
map position (Morgans). Marker name and chromosome are 
coerced using as.character. See <code>importGenMap</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>ploidy level of the organism</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ped</code></td>
<td>
<p>an optional pedigree for the supplied 
genotypes. See details.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The optional pedigree can be a data.frame, matrix or a vector. 
If the object is a data.frame or matrix, the first three 
columns must include information in the following order: id, 
mother, and father. All values are coerced using 
as.character. If the object is a vector, it is assumed to only 
include the id. In this case, the mother and father will be set 
to "0" for all individuals.
</p>


<h3>Value</h3>

<p>a <code>MapPop-class</code> if ped is NULL,
otherwise a <code>NamedMapPop-class</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">haplo = rbind(c(1,1,0,1,0),
              c(1,1,0,1,0),
              c(0,1,1,0,0),
              c(0,1,1,0,0))
colnames(haplo) = letters[1:5]

genMap = data.frame(markerName=letters[1:5],
                    chromosome=c(1,1,1,2,2),
                    position=c(0,0.5,1,0.15,0.4))

ped = data.frame(id=c("a","b"),
                 mother=c(0,0),
                 father=c(0,0))

founderPop = importHaplo(haplo=haplo, 
                         genMap=genMap,
                         ploidy=2L,
                         ped=ped)

</code></pre>


</div>