<div class="container">

<table style="width: 100%;"><tr>
<td>writeRecords</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Write data records</h2>

<h3>Description</h3>

<p>Saves a population's phenotypic and marker data to a directory.
</p>


<h3>Usage</h3>

<pre><code class="language-R">writeRecords(
  pop,
  dir,
  snpChip = 1,
  useQtl = FALSE,
  includeHaplo = FALSE,
  append = TRUE,
  simParam = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>an object of <code>Pop-class</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir</code></td>
<td>
<p>path to a directory for saving output</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpChip</code></td>
<td>
<p>which SNP chip genotype to save. If useQtl=TRUE, this
value will indicate which trait's QTL genotype to save. A value of
0 will skip writing a snpChip.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useQtl</code></td>
<td>
<p>should QTL genotype be written instead of SNP chip
genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>includeHaplo</code></td>
<td>
<p>should markers be separated by female and male
haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>append</code></td>
<td>
<p>if true, new records are added to any existing records.
If false, any existing records are deleted before writing new records.
Note that this will delete all files in the 'dir' directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParam</code></td>
<td>
<p>an object of <code>SimParam</code></p>
</td>
</tr>
</table>
</div>