<div class="container">

<table style="width: 100%;"><tr>
<td>addSegSite</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add segregating site to MapPop</h2>

<h3>Description</h3>

<p>This function allows for adding a new
segregating site with user supplied genotypes to a MapPop.
The position of the site is set using a genetic map position.
</p>


<h3>Usage</h3>

<pre><code class="language-R">addSegSite(mapPop, siteName, chr, mapPos, haplo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mapPop</code></td>
<td>
<p>an object of <code>MapPop-class</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>siteName</code></td>
<td>
<p>name to give the segregating site</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>which chromosome to add the site</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapPos</code></td>
<td>
<p>genetic map position of site in Morgans</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haplo</code></td>
<td>
<p>haplotypes for the site</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>an object of <code>MapPop-class</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Creates a populations of 10 outbred individuals
# Their genome consists of 1 chromosome and 2 segregating sites
founderPop = quickHaplo(nInd=10,nChr=1,segSites=2)

# Add a locus a the 0.5 Morgan map position
haplo = matrix(sample(x=0:1, size=20, replace=TRUE), ncol=1)

founderPop2 = addSegSite(founderPop, siteName="x", chr=1, mapPos=0.5, haplo=haplo)

pullSegSiteHaplo(founderPop2)

</code></pre>


</div>