<div class="container">

<table style="width: 100%;"><tr>
<td>pullMarkerHaplo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pull marker haplotypes</h2>

<h3>Description</h3>

<p>Retrieves haplotype data for user
specified loci
</p>


<h3>Usage</h3>

<pre><code class="language-R">pullMarkerHaplo(pop, markers, haplo = "all", asRaw = FALSE, simParam = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>an object of <code>RawPop-class</code> or
<code>MapPop-class</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markers</code></td>
<td>
<p>a character vector. Indicates the
names of the loci to be retrieved</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haplo</code></td>
<td>
<p>either "all" for all haplotypes or an integer
for a single set of haplotypes. Use a value of 1 for female
haplotypes and a value of 2 for male haplotypes in diploids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>asRaw</code></td>
<td>
<p>return in raw (byte) format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParam</code></td>
<td>
<p>an object of <code>SimParam</code>, not 
used if pop is <code>MapPop-class</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a matrix of genotypes.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Create founder haplotypes
founderPop = quickHaplo(nInd=10, nChr=1, segSites=15)

#Set simulation parameters
SP = SimParam$new(founderPop)

SP$addTraitA(10)
SP$addSnpChip(5)
SP$setTrackRec(TRUE)

#Create population
pop = newPop(founderPop, simParam=SP)

#Pull haplotype data for first two markers on chromosome one.
#Marker name is consistent with default naming in AlphaSimR.
pullMarkerHaplo(pop, markers=c("1_1","1_2"), simParam=SP)

</code></pre>


</div>