<div class="container">

<table style="width: 100%;"><tr>
<td>RRBLUP_D2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>RR-BLUP with Dominance Model 2</h2>

<h3>Description</h3>

<p>Fits an RR-BLUP model for genomic predictions that includes
dominance effects. This implementation is meant for situations where
<code>RRBLUP_D</code> is too slow. Note that RRBLUP_D2
is only faster in certain situations. Most users should use
<code>RRBLUP_D</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">RRBLUP_D2(
  pop,
  traits = 1,
  use = "pheno",
  snpChip = 1,
  useQtl = FALSE,
  maxIter = 10,
  Va = NULL,
  Vd = NULL,
  Ve = NULL,
  useEM = TRUE,
  tol = 1e-06,
  simParam = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>a <code>Pop-class</code> to serve as the training population</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>traits</code></td>
<td>
<p>an integer indicating the trait to model, a trait name, or a
function of the traits returning a single value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use</code></td>
<td>
<p>train model using phenotypes "pheno", genetic values "gv",
estimated breeding values "ebv", breeding values "bv", or randomly "rand"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpChip</code></td>
<td>
<p>an integer indicating which SNP chip genotype
to use</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useQtl</code></td>
<td>
<p>should QTL genotypes be used instead of a SNP chip.
If TRUE, snpChip specifies which trait's QTL to use, and thus these
QTL may not match the QTL underlying the phenotype supplied in traits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxIter</code></td>
<td>
<p>maximum number of iterations. Only used
when number of traits is greater than 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Va</code></td>
<td>
<p>marker effect variance for additive effects. If value is NULL,
a reasonable starting point is chosen automatically.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Vd</code></td>
<td>
<p>marker effect variance for dominance effects. If value is NULL,
a reasonable starting point is chosen automatically.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Ve</code></td>
<td>
<p>error variance. If value is NULL, a
reasonable starting point is chosen automatically.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useEM</code></td>
<td>
<p>use EM to solve variance components. If false,
the initial values are considered true.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>tolerance for EM algorithm convergence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParam</code></td>
<td>
<p>an object of <code>SimParam</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments if using a function for
traits</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">#Create founder haplotypes
founderPop = quickHaplo(nInd=10, nChr=1, segSites=20)

#Set simulation parameters
SP = SimParam$new(founderPop)

SP$addTraitAD(10, meanDD=0.5)
SP$setVarE(h2=0.5)
SP$addSnpChip(10)

#Create population
pop = newPop(founderPop, simParam=SP)

#Run GS model and set EBV
ans = RRBLUP_D2(pop, simParam=SP)
pop = setEBV(pop, ans, simParam=SP)

#Evaluate accuracy
cor(gv(pop), ebv(pop))

</code></pre>


</div>