<div class="container">

<table style="width: 100%;"><tr>
<td>getQtlMap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get QTL genetic map</h2>

<h3>Description</h3>

<p>Retrieves the genetic map for the 
QTL of a given trait.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getQtlMap(trait = 1, sex = "A", simParam = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>trait</code></td>
<td>
<p>an integer for the</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex</code></td>
<td>
<p>determines which sex specific map 
is returned. Options are "A" for average map, "F" 
for female map, and "M" for male map. All options are 
equivalent if not using sex specific maps.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParam</code></td>
<td>
<p>an object of <code>SimParam</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data.frame with:
</p>

<dl>
<dt>id</dt>
<dd>
<p>Unique identifier for the QTL</p>
</dd>
<dt>chr</dt>
<dd>
<p>Chromosome containing the QTL</p>
</dd>
<dt>site</dt>
<dd>
<p>Segregating site on the chromosome</p>
</dd>
<dt>pos</dt>
<dd>
<p>Genetic map position</p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R">#Create founder haplotypes
founderPop = quickHaplo(nInd=10, nChr=1, segSites=10)

#Set simulation parameters
SP = SimParam$new(founderPop)

SP$addTraitA(5)

#Pull SNP map
getQtlMap(trait=1, simParam=SP)

</code></pre>


</div>