<div class="container">

<table style="width: 100%;"><tr>
<td>getCAI</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate the Codon Adaptation Index</h2>

<h3>Description</h3>

<p><code>getCAI</code> returns the Codon Adaptation Index for a 
genome based on a provided reference.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getCAI(referenceGenome, testGenome, default.weight = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>referenceGenome</code></td>
<td>
<p>A genome object initialized with <code>initializeGenomeObject</code>.
Serves as reference set to calculate the necessary codon weights.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>testGenome</code></td>
<td>
<p>A genome object initialized with <code>initializeGenomeObject</code>.
The genome for which the CAI is supposed to be calculated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>default.weight</code></td>
<td>
<p>Default weight to use if codon is missing from referenceGenome</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a named vector with the CAI for each gene
</p>


<h3>Examples</h3>

<pre><code class="language-R">
genome_file1 &lt;- system.file("extdata", "more_genes.fasta", package = "AnaCoDa")
genome_file2 &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

## reading genome
referenceGenome &lt;- initializeGenomeObject(file = genome_file1)
testGenome &lt;- initializeGenomeObject(file = genome_file2)

cai &lt;- getCAI(referenceGenome, testGenome)

</code></pre>


</div>