<div class="container">

<table style="width: 100%;"><tr>
<td>initializeGenomeObject</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genome Initialization</h2>

<h3>Description</h3>

<p><code>initializeGenomeObject</code> initializes the Rcpp Genome object
</p>


<h3>Usage</h3>

<pre><code class="language-R">initializeGenomeObject(
  file,
  genome = NULL,
  observed.expression.file = NULL,
  fasta = TRUE,
  positional = FALSE,
  match.expression.by.id = TRUE,
  append = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>A file of coding sequences in fasta or RFPData format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>A genome object can be passed in to concatenate the input file to it (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>observed.expression.file</code></td>
<td>
<p>String containing the location of a file containing
empirical expression rates (optional). Default value is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fasta</code></td>
<td>
<p>Boolean value indicating whether <code>file</code> argument is a
fasta file (TRUE) or an RFPData file (FALSE). Default value is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>positional</code></td>
<td>
<p>Boolean indicating if the positional information in the RFPData file is necessary. Default value is FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>match.expression.by.id</code></td>
<td>
<p>If TRUE, observed expression values will be assigned by matching sequence identifier. 
If FALSE, observed expression values will be assigned by order. Default value is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>append</code></td>
<td>
<p>If TRUE, function will read in additional genome data to append to an existing genome.
If FALSE, genome data is cleared before reading in data (no preexisting data). Default value is FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function returns the initialized Genome object.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
genes_file &lt;- system.file("extdata", "more_genes.fasta", package = "AnaCoDa")
expression_file &lt;- system.file("extdata", "expression.csv", package = "AnaCoDa")
 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)

## reading genome and observed expression data
genome &lt;- initializeGenomeObject(file = genome_file, observed.expression.file = expression_file)
 
## add aditional genes to existing genome
genome &lt;- initializeGenomeObject(file = genome_file)
genome &lt;- initializeGenomeObject(file = genes_file, genome = genome, append = TRUE)   

</code></pre>


</div>