<div class="container">

<table style="width: 100%;"><tr>
<td>getCodonCountsForAA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Codon Counts For a specific Amino Acid</h2>

<h3>Description</h3>

<p>provides the codon counts for a fiven amino acid across all genes
</p>


<h3>Usage</h3>

<pre><code class="language-R">getCodonCountsForAA(aa, genome)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>aa</code></td>
<td>
<p>One letter code of the amino acid for which the codon counts should be returned</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>A genome object from which the counts of each
codon can be obtained.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The returned matrix containes a row for each gene and a coloumn 
for each synonymous codon of <code>aa</code>.
</p>


<h3>Value</h3>

<p>Returns a data.frame storing the codon counts for the specified amino acid.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)
counts &lt;- getCodonCountsForAA("A", genome)

</code></pre>


</div>