<div class="container">

<table style="width: 100%;"><tr>
<td>initializeCovarianceMatrices</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Initialize Covariance Matrices</h2>

<h3>Description</h3>

<p>Initialize Covariance Matrices
</p>


<h3>Usage</h3>

<pre><code class="language-R">initializeCovarianceMatrices(
  parameter,
  genome,
  numMixtures,
  geneAssignment,
  init.csp.variance = 0.0025
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>parameter</code></td>
<td>
<p>A Parameter object that corresponds to one of the model types. 
Valid values are "ROC", "PA", and "FONSE".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>An object of type Genome necessary for the initialization of the Parameter object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numMixtures</code></td>
<td>
<p>The number of mixture elements for the underlying mixture distribution (numMixtures &gt; 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneAssignment</code></td>
<td>
<p>A vector holding the initial mixture assignment for each gene.
The vector length has to equal the number of genes in the genome.
Valid values for the vector range from 1 to numMixtures.
It is possible but not advised to leave a mixture element empty.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>init.csp.variance</code></td>
<td>
<p>initial proposal variance for codon specific parameter, default is 0.0025.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>parameter Returns the Parameter argument, now modified with initialized mutation, selection, and covariance matrices.
</p>


</div>