<div class="container">

<table style="width: 100%;"><tr>
<td>getSelectionCoefficients</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate Selection coefficients</h2>

<h3>Description</h3>

<p><code>getSelectionCoefficients</code> calculates the selection coefficient of each codon in each gene.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getSelectionCoefficients(genome, parameter, samples = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>A genome object initialized with 
<code>initializeGenomeObject</code> to add observed expression data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parameter</code></td>
<td>
<p>an object created by <code>initializeParameterObject</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samples</code></td>
<td>
<p>The number of samples used for the posterior estimates.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix with selection coefficients.
</p>


<h3>Examples</h3>

<pre><code class="language-R">genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- 1
numMixtures &lt;- 1
geneAssignment &lt;- rep(1, length(genome))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")
model &lt;- initializeModelObject(parameter = parameter, model = "ROC")
samples &lt;- 2500
thinning &lt;- 50
adaptiveWidth &lt;- 25
mcmc &lt;- initializeMCMCObject(samples = samples, thinning = thinning, 
                             adaptive.width=adaptiveWidth, est.expression=TRUE, 
                             est.csp=TRUE, est.hyper=TRUE, est.mix = TRUE) 
divergence.iteration &lt;- 10
## Not run: 
runMCMC(mcmc = mcmc, genome = genome, model = model, 
        ncores = 4, divergence.iteration = divergence.iteration)

## return estimates for selection coefficients s for each codon in each gene
selection.coefficients &lt;- getSelectionCoefficients(genome = genome, 
                                                   parameter = parameter, samples = 1000)

## End(Not run)

</code></pre>


</div>