<div class="container">

<table style="width: 100%;"><tr>
<td>addObservedSynthesisRateSet</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add gene observed synthesis rates</h2>

<h3>Description</h3>

<p><code>addObservedSynthesisRateSet</code> returns the observed 
synthesis rates of the genes within the genome specified.
</p>


<h3>Usage</h3>

<pre><code class="language-R">addObservedSynthesisRateSet(
  genome,
  observed.expression.file,
  match.expression.by.id = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>A genome object initialized with 
<code>initializeGenomeObject</code> to add observed expression data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>observed.expression.file</code></td>
<td>
<p>A string containing 
the location of a file containing empirical expression rates (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>match.expression.by.id</code></td>
<td>
<p>If TRUE (default) observed expression 
values will be assigned by matching sequence identifier.
If FALSE observed expression values will be assigned by order</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns the genome after adding the new gene expression values
</p>


<h3>Examples</h3>

<pre><code class="language-R">
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
expression_file &lt;- system.file("extdata", "expression.csv", package = "AnaCoDa") 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)


## add expression values after the genome was initiallized, 
## or adding an additional set of expression values
genome &lt;- addObservedSynthesisRateSet(genome = genome, 
                   observed.expression.file = expression_file)

</code></pre>


</div>