<div class="container">

<table style="width: 100%;"><tr>
<td>initializeModelObject</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Model Initialization</h2>

<h3>Description</h3>

<p>initializes the model object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">initializeModelObject(
  parameter,
  model = "ROC",
  with.phi = FALSE,
  fix.observation.noise = FALSE,
  rfp.count.column = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>parameter</code></td>
<td>
<p>An object created with <code>initializeParameterObject</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>A string containing the model to run (ROC, FONSE, or PA), has to match parameter object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>with.phi</code></td>
<td>
<p>(ROC only) A boolean that determines whether or not to include empirical
phi values (expression rates) for the calculations. Default value is FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fix.observation.noise</code></td>
<td>
<p>(ROC only) Allows fixing the noise term sepsilon in the observed expression dataset to its initial condition.  This value should override the est.hyper=TRUE setting in <code>initializeMCMCObject()</code>
The initial condition for the observed expression noise is set in the parameter object. Default value is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rfp.count.column</code></td>
<td>
<p>(PA and PANSE only) A number representing the RFP count column to use. Default value is 1.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>initializeModelObject initializes a model. The type of model is determined based on the string passed to the <code>model</code> argument.
The Parameter object has to match the model that is initialized. E.g. to initialize a ROC model, 
it is required that a ROC parameter object is passed to the function.
</p>


<h3>Value</h3>

<p>This function returns the model object created.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#initializing a model object

genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
expression_file &lt;- system.file("extdata", "expression.csv", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file, 
                                 observed.expression.file = expression_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")

# initializing a model object assuming we have observed expression (phi) 
# values stored in the genome object.
initializeModelObject(parameter = parameter, model = "ROC", with.phi = TRUE)

# initializing a model object ignoring observed expression (phi) 
# values stored in the genome object.
initializeModelObject(parameter = parameter, model = "ROC", with.phi = FALSE)

</code></pre>


</div>