<div class="container">

<table style="width: 100%;"><tr>
<td>getObservedSynthesisRateSet</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get gene observed synthesis rates</h2>

<h3>Description</h3>

<p><code>getObservedSynthesisRateSet</code> returns the observed 
synthesis rates of the genes within the genome specified.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getObservedSynthesisRateSet(genome, simulated = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>A genome object initialized with <code>initializeGenomeObject</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simulated</code></td>
<td>
<p>A logical value denoting if the synthesis 
rates to be listed are simulated or not. The default value is FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data.frame with the observed expression values in genome
</p>


<h3>Examples</h3>

<pre><code class="language-R">
genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")
expression_file &lt;- system.file("extdata", "expression.csv", package = "AnaCoDa") 
## reading genome
genome &lt;- initializeGenomeObject(file = genome_file)


## return expression values as a data.frame with gene ids in the first column.
expressionValues &lt;- getObservedSynthesisRateSet(genome = genome)

</code></pre>


</div>