<div class="container">

<table style="width: 100%;"><tr>
<td>writeParameterObject</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Write Parameter Object to a File</h2>

<h3>Description</h3>

<p><code>writeParameterObject</code> will write the parameter object as binary to the filesystem
</p>


<h3>Usage</h3>

<pre><code class="language-R">writeParameterObject(parameter, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>parameter</code></td>
<td>
<p>parameter on object created by <code>initializeParameterObject</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>A filename that where the data will be stored.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>As Rcpp object are not serializable with the default R <code>save</code> function, 
therefore this custom save function is provided (see loadParameterObject).
</p>


<h3>Value</h3>

<p>This function has no return value.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 

genome_file &lt;- system.file("extdata", "genome.fasta", package = "AnaCoDa")

genome &lt;- initializeGenomeObject(file = genome_file)
sphi_init &lt;- c(1,1)
numMixtures &lt;- 2
geneAssignment &lt;- c(rep(1,floor(length(genome)/2)),rep(2,ceiling(length(genome)/2)))
parameter &lt;- initializeParameterObject(genome = genome, sphi = sphi_init, 
                                       num.mixtures = numMixtures, 
                                       gene.assignment = geneAssignment, 
                                       mixture.definition = "allUnique")

## writing an empty parameter object as the runMCMC routine was not called yet
writeParameterObject(parameter = parameter, file = file.path(tempdir(), "file.Rda"))


## End(Not run)

</code></pre>


</div>