<div class="container">

<table style="width: 100%;"><tr>
<td>Maelstrom_Motif2TF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Maelstrom_Motif2TF</h2>

<h3>Description</h3>

<p>create motif-factor links &amp; export tables for printing motif score alongside its binding factor
</p>


<h3>Usage</h3>

<pre><code class="language-R">Maelstrom_Motif2TF(
  seurat_object,
  mot_mat = NULL,
  m2f_df = NULL,
  cluster_id = "seurat_clusters",
  maelstrom_dir = "./maelstrom/",
  combine_motifs = "means",
  RNA_expression_assay = "RNA",
  RNA_expression_slot = "data",
  expr_tresh = 10,
  cor_tresh = 0.3,
  curated_motifs = FALSE,
  cor_method = "pearson",
  return_df = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seurat_object</code></td>
<td>
<p>object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mot_mat</code></td>
<td>
<p>motif_matrix, if not provided extracts one from the single cell object from the maelstrom assay</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m2f_df</code></td>
<td>
<p>motif to factor dataframe, if not provided extracts from the maelstrom directory</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cluster_id</code></td>
<td>
<p>ID used for finding clusters of cells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maelstrom_dir</code></td>
<td>
<p>directory where the GimmeMotifs m2f table is stored</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>combine_motifs</code></td>
<td>
<p>means (take mean multiple motifscores), max_var (take motif with highest variance), or max_cor (take motif with best correlation to gene expression)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>RNA_expression_assay</code></td>
<td>
<p>Seurat assay containing factor expression info</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>RNA_expression_slot</code></td>
<td>
<p>slot within assay used for calculating mean factor expression per cluster</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expr_tresh</code></td>
<td>
<p>minimum sum of gene counts over all cells in RNA_expression_assay to filter genes by</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor_tresh</code></td>
<td>
<p>minimum value of to filter the cor() output by</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>curated_motifs</code></td>
<td>
<p>use only curated motifs (T), or all motifs in the database (F)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor_method</code></td>
<td>
<p>specify one of the cor() methods</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_df</code></td>
<td>
<p>return both the seurat object and two dataframes with maelstrom scores and expression values as a list</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>seurat object with two assays added, MotifTFcor for TFs with positive correlation to the linked motif, and MotifTFanticor for TFs with positive correlation to the linked motif
</p>


<h3>Examples</h3>

<pre><code class="language-R">sce_small &lt;- readRDS(system.file("extdata","sce_small.Rds",package = 'AnanseSeurat'))
maelstrom_dir_path &lt;- system.file("extdata","maelstrom",package = 'AnanseSeurat')
sce_small &lt;- Maelstrom_Motif2TF(sce_small, maelstrom_dir = maelstrom_dir_path)
</code></pre>


</div>