<div class="container">

<table style="width: 100%;"><tr>
<td>DiffAnalysis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Differential Analysis for paired data </h2>

<h3>Description</h3>

<p>Performs the differential analysis from normalised paired microarray data according to different ways of variance modelling
and computes lists of differentially expressed genes according different multiple testing procedures.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DiffAnalysis(fileIN = "resNorm.txt", n = 3, ind.array = 1:2, name.A = "A",
name.M = "M.norm", fileOUT = "ListOfGenes.txt",
fileDelete = "GenesOutOfAnalysis.txt", procs = c("bonferroni", "BH"),
alpha = c(0.05, 0.05), dyeswap = TRUE, indDS = c(2), fileID = NULL,
function.trt = NULL, by.var = "ID", varmixt.meth = TRUE, header = TRUE,
sep = "\t", sep.write = "\t", dec.write = ".", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fileIN          </code></td>
<td>
<p> normalized data set. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n               </code></td>
<td>
<p> number of identificant columns. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.array       </code></td>
<td>
<p> the indices of arrays to analyze. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.A          </code></td>
<td>
<p> character string containing a regular expression contained in the columnames corresponding to the A values. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.M          </code></td>
<td>
<p> character string containing a regular expression contained in the columnames corresponding to the M values. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileOUT         </code></td>
<td>
<p> output data file. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileDelete      </code></td>
<td>
<p> output data file containing the list of withdrawn genes. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>procs           </code></td>
<td>
<p> adjusting p-values procedures to be used. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha           </code></td>
<td>
<p> values of the first type error to be used in the different procedures. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dyeswap         </code></td>
<td>
<p> logical indicating whether the design is a dye swap. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indDS           </code></td>
<td>
<p> index of arrays related to dye swap. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileID          </code></td>
<td>
<p> file giving information about genes. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>function.trt    </code></td>
<td>
<p> function to be applied before differential analysis. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by.var          </code></td>
<td>
<p> argument passed to function.trt. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>varmixt.meth    </code></td>
<td>
<p> logical indicating whether to perform the variance modelling proposed by Delmar et al. (2005). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header          </code></td>
<td>
<p> a logical value indicating whether the file contains the names of the variables as its first line. Used in the read.table function. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep             </code></td>
<td>
<p> the field separator string to use in the read.table function. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep.write       </code></td>
<td>
<p> the field separator string to use in the write.table function. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dec.write       </code></td>
<td>
<p> the string to use for decimal points in the write.table function. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...           </code></td>
<td>
<p> Further arguments to be passed to read.table. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function performs a differential analysis in the gene-specific and homoscedastic cases.
If <em>varmixt.meth = TRUE</em>, the method used is the method proposed by Delmar et al. (2005), (the same as coded in the vm.analysis.paired function from the varmixt package.)
</p>


<h3>Value</h3>

<p>By default these following files are created
</p>
<p>- one list of differentially expressed genes by procedure
</p>
<p>- The list of withdrawn genes
</p>
<p>- The list of all the genes
</p>
<p>An R object is returned if function return value is assigned.
</p>


<h3>Author(s)</h3>

<p> J. Aubert </p>


<h3>References</h3>

<p> Delmar, P., Robin, S. and Daudin, J.J., (2005), VarMixt: efficient variance modelling for the differential analysis of replicated gene expression data, <em>Bioinformatics</em>, <b>21</b>,(4), 502–8
</p>
<p>Dudoit, S., Yang, Y. H., Callow, M. J. and Speed, T.P., (2002), Statistical methods for identifying differentially expressed genes in replicated cdna microarray experiments, <em>Statisticia Sinica</em>, <b>12</b>, 111–139
</p>


<h3>See Also</h3>

 <p><code>p.adjust</code>,  <code>MeanBySpot</code>,  <code>DiffAnalysis.unpaired</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(spleen)
# Analysis on the first 100 genes
resSpleen &lt;- DiffAnalysis(spleen[1:100, ], n = 1, ind.array = 1:6, name.A = "A.", name.M = "M.",
varmixt.meth = TRUE, dyeswap = TRUE, indDS = c(2,4,6))
# Histogramm of the raw-pvalues
# hist(resSpleen$PValueVM)
</code></pre>


</div>