<div class="container">

<table style="width: 100%;"><tr>
<td>MergeSearchTerms</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Merging of two tables containing search terms to expand search term database for the AnnotationBust function.</h2>

<h3>Description</h3>

<p>This function merges two data frames with search terms. This allows users to easily add search terms to data frames (either their
own or ones included in this package using data() as GenBank annotations for the same genes may vary in gene name.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MergeSearchTerms(..., SortGenes = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>the data frames of search terms you want to combine into a single data frame The Data frame(s) should have stringsAsFactors=FALSE listed if you want to sort them.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SortGenes</code></td>
<td>
<p>Should the final data frame be sorted by gene name? Default is FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A new merged data frame with all the search terms combined from the lists supplied. If sort.gene=TRUE, genes will be sorted by name.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#load the list of search terms for mitochondrial genes
data(mtDNAterms) 

#Make a data.frame of new terms to add.
#This is a dummy example for a non-real annoation of COI, but lets pretend it is real.
add.name&lt;-data.frame("COI","CDS", "CX1")

# make the column names the same for combination.
colnames(add.name)&lt;-colnames(mtDNAterms)

#Run the merge search term function without sorting based on gene name.
new.terms&lt;-MergeSearchTerms(add.name, mtDNAterms, SortGenes=FALSE)

#Run the merge search term function with sorting based on gene name.
new.terms&lt;-MergeSearchTerms(add.name, mtDNAterms, SortGenes=TRUE)

#Merge search terms and create an additional column for introns and/or exons to
#In this example, add the trnK intron to the terms
#create empty IntornExonNumber column for non-intron/exons
cp.terms&lt;-cbind(cpDNAterms,rep(NA,length(cpDNAterms$Name)))
colnames(cp.terms)[4]&lt;-"IntronExonNumber"#Name the column IntronExonNumber
trnK.intron.terms&lt;-subset(cpDNAterms,cpDNAterms$Locus=="trnK")#subset trnK
#Create a vector of 1's the same length as the number of rows for trnK
trnK.terms&lt;-cbind(trnK.intron.terms,rep(1,length(trnK.intron.terms$Name)))
colnames(trnK.terms)[4]&lt;-"IntronExonNumber"#Name the column IntronExonNumber
#Use MergeSearchTerms to merge the modified cpDNAterms and new intron terms
all.terms&lt;-MergeSearchTerms(cp.terms,trnK.terms)
</code></pre>


</div>