<div class="container">

<table style="width: 100%;"><tr>
<td>getAnnotationsGenBank</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read Annotations from GenBank</h2>

<h3>Description</h3>

<p>This function connects to the GenBank database and reads sequence annotations
using accession number(s) given as argument.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getAnnotationsGenBank(access.nb, quiet = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>access.nb</code></td>
<td>
<p>a vector of mode character giving the accession
numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>a logical value indicating whether to show the progress
of the downloads.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The sequence annotations (a.k.a. feature list) are returned in a data
frame with five or six columns: start, end, type, product, others, and
gene (the last being optional). This is the same information that can
be downloaded from NCBI's Web interface by clicking on ‘Send to:’,
‘File’, and then selecting ‘Feature Table’ under ‘Format’.
</p>
<p>A warning is given if some features are incomplete (this information
is then dropped from the returned object).
</p>
<p>A warning is given if some accession numbers are not found on GenBank.
</p>


<h3>Value</h3>

<p>On of the followings: (i) a data frame if <code>access.nb</code> contains a
single accession number; (ii) a list of data frames if
<code>access.nb</code> contains several accession numbers, the names are set
with <code>access.nb</code> (if some accession numbers are not found on
GenBank, the corresponding entries are set to <code>NULL</code>); (iii)
<code>NULL</code> if all accession numbers are not found on GenBank.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>References</h3>

<p>https://www.ncbi.nlm.nih.gov/Sequin/table.html (Note: it
seems this URL is broken; 2022-01-03)</p>


<h3>See Also</h3>

<p><code>read.GenBank</code>, <code>read.gff</code>,
<code>DNAbin</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## The 8 sequences of tanagers (Ramphocelus):
ref &lt;- c("U15717", "U15718", "U15719", "U15720",
         "U15721", "U15722", "U15723", "U15724")
## Copy/paste or type the following commands if you
## want to try them.
## Not run: 
annot.rampho &lt;- getAnnotationsGenBank(ref)
annot.rampho
## check all annotations are the same:
unique(do.call(rbind, annot.rampho)[, -5])

## End(Not run)
</code></pre>


</div>