<div class="container">

<table style="width: 100%;"><tr>
<td>image.DNAbin</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot of DNA Sequence Alignement</h2>

<h3>Description</h3>

<p>This function plots an image of an alignment of nucleotide sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'DNAbin'
image(x, what, col, bg = "white", xlab = "", ylab = "",
      show.labels = TRUE, cex.lab = 1, legend = TRUE,
      grid = FALSE, show.bases = FALSE, base.cex = 1,
      base.font = 1, base.col = "black", scheme = "Ape_NT", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a matrix of DNA sequences (class <code>"DNAbin"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>what</code></td>
<td>
<p>a vector of characters specifying the bases to
visualize. If missing, this is set to “a”, “g”, “c”, “t”,
“n”, and “-” (in this order).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>a vector of colours. If missing, this is set to “red”,
“yellow”, “green”, “blue”, “grey”, and “black”. If it is
shorter (or longer) than <code>what</code>, it is recycled (or shortened).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bg</code></td>
<td>
<p>the colour used for nucleotides whose base is not among
<code>what</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>the label for the <em>x</em>-axis; none by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>Idem for the <em>y</em>-axis. Note that by default, the
labels of the sequences are printed on the <em>y</em>-axis (see next option).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.labels</code></td>
<td>
<p>a logical controlling whether the sequence labels
are printed (<code>TRUE</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab</code></td>
<td>
<p>a single numeric controlling the size of the sequence labels.
Use <code>cex.axis</code> to control the size of the annotations on the
<em>x</em>-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p>a logical controlling whether the legend is plotted
(<code>TRUE</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid</code></td>
<td>
<p>a logical controlling whether to draw a grid (<code>FALSE</code>
by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.bases</code></td>
<td>
<p>a logical controlling whether to show the base symbols
(<code>FALSE</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base.cex, base.font, base.col</code></td>
<td>
<p>control the aspect of the base
symbols (ignored if the previous is <code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scheme</code></td>
<td>
<p>a predefined color scheme. For amino acid options are "Ape_AA",
"Zappo_AA", "Clustal", "Polarity" and "Transmembrane_tendency", for
nucleotides "Ape_NT" and"RY_NT".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments passed to
<code>image.default</code> (e.g., <code>xlab</code>,
<code>cex.axis</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The idea of this function is to allow flexible plotting and colouring
of a nucleotide alignment. By default, the most common bases (a, g, c,
t, and n) and alignment gap are plotted using a standard colour
scheme.
</p>
<p>It is possible to plot only one base specified as <code>what</code> with a
chosen colour: this might be useful to check, for instance, the
distribution of alignment gaps (<code>image(x, "-")</code>) or missing data
(see examples).
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis, Klaus Schliep</p>


<h3>See Also</h3>

<p><code>DNAbin</code>, <code>del.gaps</code>, <code>alex</code>,
<code>alview</code>, <code>all.equal.DNAbin</code>,
<code>clustal</code>, <code>grid</code>,
<code>image.AAbin</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(woodmouse)
image(woodmouse)
rug(seg.sites(woodmouse), -0.02, 3, 1)
image(woodmouse, "n", "blue") # show missing data
image(woodmouse, c("g", "c"), "green") # G+C
par(mfcol = c(2, 2))
### barcoding style:
for (x in c("a", "g", "c", "t"))
    image(woodmouse, x, "black", cex.lab = 0.5, cex.axis = 0.7)
par(mfcol = c(1, 1))
### zoom on a portion of the data:
image(woodmouse[11:15, 1:50], c("a", "n"), c("blue", "grey"))
grid(50, 5, col = "black")
### see the guanines on a black background:
image(woodmouse, "g", "yellow", "black")
### Amino acid
X &lt;- trans(woodmouse, 2)
image(X) # default ape colors
image(X, scheme="Clustal") # Clustal coloring
</code></pre>


</div>