<div class="container">

<table style="width: 100%;"><tr>
<td>read.GenBank</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read DNA Sequences from GenBank via Internet</h2>

<h3>Description</h3>

<p>This function connects to the GenBank database, and reads nucleotide
sequences using accession numbers given as arguments.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.GenBank(access.nb, seq.names = access.nb, species.names = TRUE,
             as.character = FALSE, chunk.size = 400, quiet = TRUE,
             type = "DNA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>access.nb</code></td>
<td>
<p>a vector of mode character giving the accession numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq.names</code></td>
<td>
<p>the names to give to each sequence; by default the
accession numbers are used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species.names</code></td>
<td>
<p>a logical indicating whether to attribute the
species names to the returned object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>as.character</code></td>
<td>
<p>a logical controlling whether to return the
sequences as an object of class <code>"DNAbin"</code> (the default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chunk.size</code></td>
<td>
<p>the number of sequences downloaded together (see
details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>a logical value indicating whether to show the progress
of the downloads. If <code>TRUE</code>, will also print the (full) name of
the FASTA file containing the downloaded sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>a character specifying to download "DNA" (nucleotide) or
"AA" (amino acid) sequences.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function uses the site <a href="https://www.ncbi.nlm.nih.gov/">https://www.ncbi.nlm.nih.gov/</a> from
where the sequences are retrieved.
</p>
<p>If <code>species.names = TRUE</code>, the returned list has an attribute
<code>"species"</code> containing the names of the species taken from the
field “ORGANISM” in GenBank.
</p>
<p>Since <span class="pkg">ape</span> 3.6, this function retrieves the sequences in FASTA
format: this is more efficient and more flexible (scaffolds and
contigs can be read) than what was done in previous versions. The
option <code>gene.names</code> has been removed in <span class="pkg">ape</span> 5.4; this
information is also present in the description.
</p>
<p>Setting <code>species.names = FALSE</code> is much faster (could be useful
if you read a series of scaffolds or contigs, or if you already have
the species names).
</p>
<p>The argument <code>chunk.size</code> is set by default to 400 which is
likely to work in many cases. If an error occurs such as “Cannot open
file ...” showing the list of the accession numbers, then you may
try decreasing <code>chunk.size</code> to 200 or 300.
</p>
<p>If <code>quiet = FALSE</code>, the display is done chunk by chunk, so the
message “Downloading sequences: 400 / 400 ...” means that the
download from sequence 1 to sequence 400 is under progress (it is not
possible to display a more accurate message because the download
method depends on the platform).
</p>


<h3>Value</h3>

<p>A list of DNA sequences made of vectors of class <code>"DNAbin"</code>, or
of single characters (if <code>as.character = TRUE</code>) with two
attributes (species and description).
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis and Klaus Schliep</p>


<h3>See Also</h3>

<p><code>read.dna</code>, <code>write.dna</code>,
<code>dist.dna</code>, <code>DNAbin</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## This won't work if your computer is not connected
## to the Internet

## Get the 8 sequences of tanagers (Ramphocelus)
## as used in Paradis (1997)
ref &lt;- c("U15717", "U15718", "U15719", "U15720",
         "U15721", "U15722", "U15723", "U15724")
## Copy/paste or type the following commands if you
## want to try them.
## Not run: 
Rampho &lt;- read.GenBank(ref)
## get the species names:
attr(Rampho, "species")
## build a matrix with the species names and the accession numbers:
cbind(attr(Rampho, "species"), names(Rampho))
## print the first sequence
## (can be done with `Rampho$U15717' as well)
Rampho[[1]]
## the description from each FASTA sequence:
attr(Rampho, "description")

## End(Not run)
</code></pre>


</div>