<div class="container">

<table style="width: 100%;"><tr>
<td>read.gff</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read GFF Files</h2>

<h3>Description</h3>

<p>This function reads a file in general feature format version 3 (GFF3)
and returns a data frame.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.gff(file, na.strings = c(".", "?"), GFF3 = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>a file name specified by a character string.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.strings</code></td>
<td>
<p>the strings in the GFF file that will be converted
as NA's (missing values).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GFF3</code></td>
<td>
<p>a logical value specifying whether if the file is
formatted according to version 3 of GFF.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The returned data frame has its (column) names correctly set (see
References) and the categorical variables (seqid, source, type,
strand, and phase) set as factors.
</p>
<p>This function should be more efficient than using <code>read.delim</code>.
</p>
<p>GFF2 (aka GTF) files can also be read: use <code>GFF3 = FALSE</code> to have
the correct field names. Note that GFF2 files and GFF3 files have the
same structure, although some fields are slightly different (see
reference).
</p>
<p>The file can be gz-compressed (see examples), but not zipped.
</p>


<h3>Value</h3>

<p>NULL</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>References</h3>

<p><a href="https://en.wikipedia.org/wiki/General_feature_format">https://en.wikipedia.org/wiki/General_feature_format</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## requires to be connected on Internet
d &lt;- "https://ftp.ensembl.org/pub/release-86/gff3/homo_sapiens/"
f &lt;- "Homo_sapiens.GRCh38.86.chromosome.MT.gff3.gz"
download.file(paste0(d, f), "mt_gff3.gz")
## If the above command doesn't work, you may copy/paste the full URL in
## a Web browser instead.
gff.mito &lt;- read.gff("mt_gff3.gz")
## the lengths of the sequence features:
gff.mito$end - (gff.mito$start - 1)
table(gff.mito$type)
## where the exons start:
gff.mito$start[gff.mito$type == "exon"]

## End(Not run)
</code></pre>


</div>