<div class="container">

<table style="width: 100%;"><tr>
<td>aphid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The <span class="pkg">aphid</span> package for analysis with profile hidden Markov models.</h2>

<h3>Description</h3>

<p><span class="pkg">aphid</span> is an R package for the development and application of
hidden Markov models and profile HMMs for biological sequence analysis.
Functions are included for multiple and pairwise sequence alignment,
model construction and parameter optimization, calculation of conditional
probabilities (using the forward, backward and Viterbi algorithms),
tree-based sequence weighting, sequence simulation, and file import/export
compatible with the <a href="http://www.hmmer.org">HMMER</a> software package.
The package has a wide variety of uses including database searching,
gene-finding and annotation, phylogenetic analysis and sequence classification.
</p>


<h3>Details</h3>

<p>The <span class="pkg">aphid</span> package is based on the algorithms outlined in the book 'Biological
sequence analysis: probabilistic models of proteins and nucleic acids' by Richard Durbin,
Sean Eddy, Anders Krogh and Graeme Mitchison. This book is highly recommended
for those wishing to develop a better understanding of HMMs and PHMMs, regardless of
prior experience. Many of the examples in the function help pages are taken directly
from the book, so that readers can learn to use the package as they work through the
chapters.
</p>
<p>There are also excellent rescources available for those wishing to use profile hidden
Markov models outside of the R environment. The <span class="pkg">aphid</span> package maintains
compatibility with the <a href="http://www.hmmer.org">HMMER</a> software suite
through the file input and output functions <code>readPHMM</code> and
<code>writePHMM</code>.
</p>
<p>The <span class="pkg">aphid</span> package is designed to work in conjunction with the "DNAbin"
and "AAbin" object types produced by the <code>ape</code> package
(Paradis et al 2004, 2012). This is an essential piece of software for those
using R for biological sequence analysis, and provides a binary coding format
for nucleotides and amino acids that maximizes memory and speed efficiency.
While <span class="pkg">aphid</span> also works with standard character vectors and matrices,
it may not recognize the DNA and amino acid amibguity codes and therefore is not
guaranteed to treat them appropriately.
</p>
<p>To maximize speed, the low-level dynamic programming functions such
as <code>Viterbi</code>, <code>forward</code> and <code>backward</code>
are written in C++ with the help of the <code>Rcpp</code>
package (Eddelbuettel &amp; Francois 2011).
Note that R versions of these functions are also maintained
for the purposes of debugging, experimentation and code interpretation.
</p>


<h3>Classes</h3>

<p>The <span class="pkg">aphid</span> package creates two primary object classes, <code>"HMM"</code>
(hidden Markov models) and <code>"PHMM"</code> (profile hidden Markov models)
with the functions <code>deriveHMM</code> and <code>derivePHMM</code>, respectively.
These objects are lists consisting of emission and transition probability matrices
(denoted E and A), vectors of non-position-specific background emission and transition
probabilies (denoted qe and qa) and other model metadata.
Objects of class <code>"DPA"</code> (dynammic programming array) are also generated
by the Viterbi and forward/backward functions.
These are primarily created for succinct console printing.
</p>


<h3>Functions</h3>

<p>A breif description of the primary <span class="pkg">aphid</span> functions are provided with links
to their help pages below.
</p>


<h3>File import and export</h3>


<ul>
<li> <p><code>readPHMM</code> parses a <a href="http://www.hmmer.org">HMMER</a> text file
into R and creates an object of class <code>"PHMM"</code>
</p>
</li>
<li> <p><code>writePHMM</code> writes a <code>"PHMM"</code> object to a text file in
<a href="http://www.hmmer.org">HMMER</a> v3 format
</p>
</li>
</ul>
<h3>Visualization</h3>


<ul>
<li> <p><code>plot.HMM</code> plots a <code>"PHMM"</code> object as a cyclic directed graph
</p>
</li>
<li> <p><code>plot.PHMM</code> plots a <code>"PHMM"</code> object as a directed graph with
sequential modules consisting of match, insert and delete states
</p>
</li>
</ul>
<h3>Model building and training</h3>


<ul>
<li> <p><code>deriveHMM</code> builds a <code>"HMM"</code> object from a list of training
sequences
</p>
</li>
<li> <p><code>derivePHMM</code> builds a <code>"PHMM"</code> object from a multiple sequence
alignment or a list of non-aligned sequences
</p>
</li>
<li> <p><code>map</code> optimizes profile hidden Markov model construction
using the maximum <em>a posteriori</em> algorithm
</p>
</li>
<li> <p><code>train</code> optimizes the parameters of a <code>"HMM"</code> or
<code>"PHMM"</code> object using a list of training sequences
</p>
</li>
</ul>
<h3>Sequence alignment and weighting</h3>


<ul>
<li> <p><code>align</code> performs a multiple sequence alignment
</p>
</li>
<li> <p><code>weight</code> assigns weights to sequences
</p>
</li>
</ul>
<h3>Conditional probabilities</h3>


<ul>
<li> <p><code>Viterbi</code> finds the optimal path of a sequence through a HMM
or PHMM, and returns its log odds or probability given the model
</p>
</li>
<li> <p><code>forward</code> finds the full probability of a sequence
given a HMM or PHMM using the forward algorithm
</p>
</li>
<li> <p><code>backward</code> finds the full probability of a sequence
given a HMM or PHMM using the backward algorithm
</p>
</li>
<li> <p><code>posterior</code> finds the position-specific posterior probability
of a sequence given a HMM or PHMM
</p>
</li>
</ul>
<h3>Sequence simulation</h3>


<ul>
<li> <p><code>generate.HMM</code> simulates a random sequence from an HMM
</p>
</li>
<li> <p><code>generate.PHMM</code> simulates a random sequence from a PHMM
</p>
</li>
</ul>
<h3>Datasets</h3>


<ul>
<li> <p><code>substitution</code> a collection of DNA and amino acid
substitution matrices from <a href="ftp://ftp.ncbi.nih.gov/blast/matrices/">NCBI</a>
including the PAM, BLOSUM, GONNET, DAYHOFF and NUC matrices
</p>
</li>
<li> <p><code>casino</code> data from the dishonest casino example of
Durbin et al (1998) chapter 3.2
</p>
</li>
<li> <p><code>globins</code> Small globin alignment data from
Durbin et al (1998) Figure 5.3
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>References</h3>

<p>Durbin R, Eddy SR, Krogh A, Mitchison G (1998) Biological
sequence analysis: probabilistic models of proteins and nucleic acids.
Cambridge University Press, Cambridge, United Kingdom.
</p>
<p>Eddelbuettel D, Francois R (2011) Rcpp: seamless R and C++ integration.
<em>Journal of Statistical Software</em> <strong>40</strong>, 1-18.
</p>
<p>Finn RD, Clements J &amp; Eddy SR (2011) HMMER web server: interactive sequence
similarity searching.
<em>Nucleic Acids Research</em>. <strong>39</strong>, W29-W37. <a href="http://hmmer.org/">http://hmmer.org/</a>.
</p>
<p>HMMER: biosequence analysis using profile hidden Markov models.
<a href="http://www.hmmer.org">http://www.hmmer.org</a>.
</p>
<p>NCBI index of substitution matrices.
<a href="ftp://ftp.ncbi.nih.gov/blast/matrices/">ftp://ftp.ncbi.nih.gov/blast/matrices/</a>.
</p>
<p>Paradis E, Claude J, Strimmer K, (2004) APE: analyses of phylogenetics
and evolution in R language. <em>Bioinformatics</em> <strong>20</strong>, 289-290.
</p>
<p>Paradis E (2012) Analysis of Phylogenetics and Evolution with R
(Second Edition). Springer, New York.
</p>


</div>