<div class="container">

<table style="width: 100%;"><tr>
<td>generate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate random sequences from a model.</h2>

<h3>Description</h3>

<p>The <code>generate</code> function outputs a random sequence from a HMM or PHMM.
</p>


<h3>Usage</h3>

<pre><code class="language-R">generate(x, size, ...)

## S3 method for class 'HMM'
generate(x, size, logspace = "autodetect", random = TRUE, ...)

## S3 method for class 'PHMM'
generate(
  x,
  size,
  logspace = "autodetect",
  gap = "-",
  random = TRUE,
  DNA = FALSE,
  AA = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>'HMM'</code> or <code>'PHMM'</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p>a non-negative integer representing the length of the output
sequence if x is a <code>"HMM"</code> object with zero probability of
transitioning to the begin/end state, or the maximum length of the
output sequence otherwise (this acts as a safeguard against overflow).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments to be passed between methods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logspace</code></td>
<td>
<p>logical indicating whether the emission and transition
probabilities of x are logged. If <code>logspace = "autodetect"</code>
(the default setting), the function will automatically detect
if the probabilities are logged, returning an error if
inconsistencies are found. Note that choosing the latter option
increases the computational overhead; therefore specifying
<code>TRUE</code> or <code>FALSE</code> can reduce the running time.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>random</code></td>
<td>
<p>logical indicating whether residues should be emitted randomly
with probabilities defined by the emission probabilities in the model
(TRUE; default), or deterministically, whereby each residue is emitted
and each transition taken based on the maximum emission/transition
probability in the current state.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap</code></td>
<td>
<p>the character used to represent gaps (delete states)
in the output sequence (only applicable for <code>PHMM</code> objects).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DNA</code></td>
<td>
<p>logical indicating whether the returned sequence should be a
<code>"DNAbin"</code> object. Only applicable if the matrix of emission
probabilities in the model has four residues corresponding to the nucleotide
alphabet (A, T, G, and C).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>AA</code></td>
<td>
<p>logical indicating whether the returned sequence should be a
<code>"AAbin"</code> object. Only applicable if the matrix of emission
probabilities in the model has 20 residues corresponding to the amino acid
alphabet.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This simple function generates a single sequence
from a HMM or profile HMM by recursively simulating a path through
the model. The function is fairly slow in its current state, but a
faster C++ function may be made available in a future version depending
on demand.
</p>


<h3>Value</h3>

<p>a named vector giving the sequence of residues emitted by the model,
with the "names" attribute representing the hidden states.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>References</h3>

<p>Durbin R, Eddy SR, Krogh A, Mitchison G (1998) Biological
sequence analysis: probabilistic models of proteins and nucleic acids.
Cambridge University Press, Cambridge, United Kingdom.
</p>


<h3>Examples</h3>

<pre><code class="language-R">  ## Generate a random sequence from a standard HMM
  ## The dishonest casino example from Durbin et al (1998) chapter 3.2
  states &lt;- c("Begin", "Fair", "Loaded")
  residues &lt;- paste(1:6)
  ### Define the transition probability matrix
  A &lt;- matrix(c(0, 0, 0, 0.99, 0.95, 0.1, 0.01, 0.05, 0.9), nrow = 3)
  dimnames(A) &lt;- list(from = states, to = states)
  ### Define the emission probability matrix
  E &lt;- matrix(c(rep(1/6, 6), rep(1/10, 5), 1/2), nrow = 2, byrow = TRUE)
  dimnames(E) &lt;- list(states = states[-1], residues = residues)
  ### Build and plot the HMM object
  x &lt;- structure(list(A = A, E = E), class = "HMM")
  plot(x, main = "Dishonest casino HMM")
  ### Generate a random sequence from the model
  generate(x, size = 300)
  ##
  ## Generate a random sequence from a profile HMM:
  ## Small globin alignment data from Durbin et al (1998) Figure 5.3
  data(globins)
  ### Derive a profile hidden Markov model from the alignment
  globins.PHMM &lt;- derivePHMM(globins, residues = "AMINO", seqweights = NULL)
  plot(globins.PHMM, main = "Profile hidden Markov model for globins")
  ### Simulate a random sequence from the model
  suppressWarnings(RNGversion("3.5.0"))
  set.seed(999)
  simulation &lt;- generate(globins.PHMM, size = 20)
  simulation ## "F" "S" "A" "N" "N" "D" "W" "E"
  ### Names attribute indicates that all residues came from "match" states
</code></pre>


</div>