<div class="container">

<table style="width: 100%;"><tr>
<td>aSPU</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sum of Powered Score (SPU) tests and adaptive SPU (aSPU) test for single trait - SNP set association.</h2>

<h3>Description</h3>

<p>It gives p-values of the SPU tests and aSPU test.
</p>


<h3>Usage</h3>

<pre><code class="language-R">aSPU(
  Y,
  X,
  cov = NULL,
  resample = c("perm", "boot", "sim"),
  model = c("gaussian", "binomial"),
  pow = c(1:8, Inf),
  n.perm = 1000,
  threshold = 10^5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>Response or phenotype data. It can be a disease indicator; =0 for controls, =1 for cases.
Or it can be a quantitative trait. A vector with length n (number of observations).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>Genotype or other data; each row for a subject, and each column
for an SNP (or a predictor). The value of each SNP is the # of the copies
for an allele. A matrix with dimension n by k (n : number of observation, k : number of SNPs (or predictors) ).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by p (n :number of observation, p : number of covariates).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resample</code></td>
<td>
<p>Use "perm" for residual permutations, "sim" for simulations from the null distribution, and "boot" for parametric bootstrap.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>Use "gaussian" for a quantitative trait, and use "binomial" for a binary trait.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow</code></td>
<td>
<p>power used in SPU test. A vector of the powers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>number of permutations or bootstraps.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>When n.perm is less than the threshold, we use faster version of aSPU but use more memory. (default is 10^5)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list object, Ts : test statistics for the SPU tests (in the order of the specified pow) and finally for the aSPU test.
pvs : p-values for the SPU and aSPU tests.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak, Junghi Kim, Yiwei Zhang and Wei Pan
</p>


<h3>References</h3>

<p>Wei Pan, Junghi Kim, Yiwei Zhang, Xiaotong Shen and Peng Wei (2014)
A powerful and adaptive association test for rare variants,
Genetics, 197(4), 1081-95
</p>
<p>Junghi Kim, Jeffrey R Wozniak, Bryon A Mueller, Xiaotong Shen and Wei Pan (2014) Comparison of statistical tests for group differences in brain functional networks, NeuroImage, 1;101:681-694
</p>


<h3>See Also</h3>

<p><code>aSPUw</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(exdat)

## example analysis using aSPU test on exdat data.
# increase n.perm for better p.val 
## Not run: out &lt;- aSPU(exdat$Y, exdat$X, cov = NULL, resample = "boot",
            model = "binomial", pow = c(1:8, Inf), n.perm = 1000) 
## End(Not run)


out$Ts
# This is a vector of Test Statistics for SPU and aSPU tests.
# SPU1 to SPUInf corresponds with the option pow=c(1:8, Inf)
# They are SPU test statistics.
# The last element aSPU is minimum of them, aSPU statistic.

out$pvs
# This is a vector of p-values for SPU and aSPU tests.
# SPU1 to SPUInf corresponds with the option pow=c(1:8, Inf)
# They are p-values for corresponding SPU tests.
# The last element is p-value of aSPU test.

</code></pre>


</div>