<div class="container">

<table style="width: 100%;"><tr>
<td>aSPUpath2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pathway based Sum of Powered Score tests (SPUpath) and adaptive SPUpath (aSPUpath) test for single trait - pathway association. (vector version, fast when n is large)</h2>

<h3>Description</h3>

<p>It gives p-values of the SPUpath tests and aSPUpath test. Faster than aSPUsPath function when n is large (N &gt; 10^4).
</p>


<h3>Usage</h3>

<pre><code class="language-R">aSPUpath2(
  Y,
  X,
  cov = NULL,
  model = c("binomial", "gaussian"),
  snp.info,
  gene.info,
  pow = c(1:8, Inf),
  pow2 = c(1, 2, 4, 8),
  n.perm = 200,
  usePCs = F,
  varprop = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>Response or phenotype data. It can be a disease indicator; =0 for controls, =1 for cases.
Or it can be a quantitative trait. A vector with length n (number of observations).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>Genotype or other data; each row for a subject, and each column
for an SNP (or a predictor). The value of each SNP is the # of the copies
for an allele. A matrix with dimension n by k (n : number of observation, k : number of SNPs (or predictors) ).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by p (n :number of observation, p : number of covariates).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>Use "gaussian" for a quantitative trait, and use "binomial" for a binary trait.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow</code></td>
<td>
<p>SNP specific power(gamma values) used in SPUpath test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in SPUpath test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>number of permutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>usePCs</code></td>
<td>
<p>indicating whether to extract PCs and then use PCs of X.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>varprop</code></td>
<td>
<p>the proportion of the variations explained (cutoff) that
determines how many top PCs to use.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>P-values for SPUpath tests and aSPUpath test.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Wei Pan, Il-Youp Kwak and Peng Wei (2015)
A Powerful and Pathway-Based Adaptive Test for Genetic Association With Common or Rare Variants, The American Journal of Human Genetics, 97, 86-98
</p>


<h3>See Also</h3>

<p><code>simPathAR1Snp</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: dat1&lt;-simPathAR1Snp(nGenes=20, nGenes1=5, nSNPlim=c(1, 20),
	       nSNP0=1, LOR=.2, n=100, MAFlim=c(0.05, 0.4), p0=0.05 ) 
## End(Not run)

# p-values of SPUpath and aSPUpath tests.
## Not run: p.pathaspu&lt;- aSPUpath2(dat1$Y, dat1$X, snp.info = dat1$snp.info,
         gene.info = dat1$gene.info,
         model = "binomial", pow=1:8, pow2=c(1, 2, 4, 8), n.perm=1000) 
## End(Not run)
#p.pathaspu
## pow = 1:8 and pow2 = 1,2,4,8
## So, there are 8*4 = 32 SPUpath p-values.
## SPUpathi,j corresponds pow = i , pow2 = j
## The last element, aSPUpath gives aSPUpath p-value.

</code></pre>


</div>