<div class="container">

<table style="width: 100%;"><tr>
<td>aSPUsPath2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pathway based Sum of Powered Score tests (SPUsPath) and adaptive SPUpath (aSPUsPath) test for single trait - pathway association with GWAS summary statistics. (vector version, fast when n is large)</h2>

<h3>Description</h3>

<p>It gives p-values of the SPUsPath tests and aSPUsPath test with GWAS summary statistics. Faster than aSPUsPath function when n is large (N &gt; 10^4).
</p>


<h3>Usage</h3>

<pre><code class="language-R">aSPUsPath2(
  Zs,
  corSNP,
  pow = c(1, 2, 4, 8, Inf),
  pow2 = c(1, 2, 4, 8),
  snp.info,
  gene.info,
  n.perm = 1000,
  Ps = FALSE,
  prune = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Zs</code></td>
<td>
<p>Z-scores for each SNPs. It could be P-values if the Ps option is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow</code></td>
<td>
<p>SNP specific power(gamma values) used in SPUpath test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in SPUpath test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>number of permutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prune</code></td>
<td>
<p>if it is TRUE, do pruing before the test using pruneSNP function.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>P-values for SPUMpath tests and aSPUMpath test.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2015)
Adaptive Gene- and Pathway-Trait Association Testing with GWAS Summary Statistics,
Bioinformatics, 32(8):1178-1184
</p>


<h3>See Also</h3>

<p><code>aSPUs</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(kegg9)

# p-values of SPUpath and aSPUpath tests.
## Not run: out.a &lt;- aSPUsPath2(kegg9$nP, corSNP = kegg9$ldmatrix, pow=c(1:8, Inf),
                  pow2 = c(1,2,4,8), 
                  snp.info=kegg9$snp.info, gene.info = kegg9$gene.info,
                  n.perm=1000, Ps = TRUE) 
## End(Not run)

#out.a

</code></pre>


</div>