<div class="container">

<table style="width: 100%;"><tr>
<td>pruneSNP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Prune SNPs (related to aSPUs, aSPUsPath, MTaSPUsSet)</h2>

<h3>Description</h3>

<p>When correlation matirx have highly correlated SNPs, the performance of aSPUs, aSPUsPath and MTaSPUsSet are not very good. Do pruning using this function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pruneSNP(corSNP, rup = 0.95, rdown = -0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rup</code></td>
<td>
<p>pruning criteria, erase one SNP when correlation between two SNPs are larger than this value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rdown</code></td>
<td>
<p>pruning criteria, erase one SNP when correlation between two SNPs are smaller than this value.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list object pruned.corSNP and to.erase. pruned.corSNP is pruned correlation matrix. to.erase is SNP index to erase to get purned object. (i.e. corSNP[-to.erase, -to.erase] = pruned.corSNP )
</p>


<h3>Author(s)</h3>

<p>Il Youp Kwak
</p>


<h3>See Also</h3>

<p><code>aSPUs</code> <code>aSPUsPath</code> <code>MTaSPUsSet</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">

data(kegg9)
## example analysis using aSPUM test.
g &lt;- kegg9$gene.info[1,1]  #  SOAT1 
## Take snps mapped on gene "SOAT1" from the information of gene.info and snp.info. 
snps &lt;- which( ( kegg9$snp.info[,2] == kegg9$gene.info[kegg9$gene.info[,1] == g, 2] ) &amp;
                 (kegg9$snp.info[,3] &gt; kegg9$gene.info[kegg9$gene.info[,1] == g, 3] ) &amp;
                 (kegg9$snp.info[,3] &lt; kegg9$gene.info[kegg9$gene.info[,1] == g, 4] )  )
## Take subsets
newP &lt;- kegg9$nP[snps] ;
ldsub &lt;- kegg9$ldmatrix[snps, snps];

prSNP &lt;- pruneSNP(ldsub) 
newP &lt;- newP[-prSNP$to.erase]
ldsub &lt;- ldsub[-prSNP$to.erase, -prSNP$to.erase ]

## Get p-value for gene SOAT1. Read vignette for details.
out &lt;- aSPUs(newP, corSNP=ldsub , pow=c(1:8, Inf), n.perm=100, Ps=TRUE)

out$Ts
# This is a vector of Test Statistics for SPUM and aSPUM tests.
# SPUs1 to SPUsInf corresponds with the option pow=c(1:8, Inf)
# They are SPUs test statistics.
# The last element aSPUs is minimum of them, aSPUs statistic.

out$pvs
# This is a vector of p-values for SPUs and aSPUs tests.
# SPUs1 to SPUsInf corresponds with the option pow=c(1:8, Inf)
# They are p-values for corresponding SPUs tests.
# The last element is p-value of aSPUs test.

</code></pre>


</div>