<div class="container">

<table style="width: 100%;"><tr>
<td>getlogitp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get the p-values of the multiple SNPs based on the Wald test in a logistic regresion model.</h2>

<h3>Description</h3>

<p>Get p-value using logistic regresion for each of the multiple SNPs
</p>


<h3>Usage</h3>

<pre><code class="language-R">getlogitp(Y, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>Response or phenotype data. It can be a disease indicator; =0 for controls, =1 for cases.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>Genotype or other data; each row for a subject, and each column
for an SNP (or a predictor). The value of each SNP is the # of the copies
for an allele. A matrix with dimension n by p (n : number of observation, p : number of SNPs (or predictors) ).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>p-values for each SNPs.
</p>


<h3>See Also</h3>

<p><code>GATES2</code> <code>GatesSimes</code> <code>Hyst</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#simula &lt;- simPathAR1Snp(nGenes=20, nGenes1=1, nSNPlim=c(1, 20), nSNP0=1:3,
#                           LOR=.2, rholim=c(0,0),
#                           n=10, MAFlim=c(0.05, 0.4), p0=0.05)
#logitp &lt;- getlogitp(simula$Y, simula$X)


</code></pre>


</div>