<div class="container">

<table style="width: 100%;"><tr>
<td>aSPUs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sum of Powered Score (SPUs) tests and adaptive SPU (aSPUs) test for single trait - SNP set association with GWAS summary statistics.</h2>

<h3>Description</h3>

<p>It gives p-values of the SPUs tests and aSPUs test with GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class="language-R">aSPUs(Zs, corSNP, pow = c(1:8, Inf), n.perm = 1000, Ps = FALSE, prune = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Zs</code></td>
<td>
<p>Z-scores for each SNPs. It could be P-values if the Ps option is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow</code></td>
<td>
<p>power used in SPU test. A vector of the powers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>number of permutations or bootstraps.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prune</code></td>
<td>
<p>if it is TRUE, do pruing before the test using pruneSNP function.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list object, Ts : test statistics for the SPU tests (in the order of the specified pow) and finally for the aSPU test.
pvs : p-values for the SPUs and aSPUs tests.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2015)
Adaptive Gene- and Pathway-Trait Association Testing with GWAS Summary Statistics,
Bioinformatics, 32(8), 1178-1184.
</p>


<h3>See Also</h3>

<p><code>aSPUw</code> <code>aSPU</code> <code>aSPUsPath</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(kegg9)
## example analysis using aSPUM test.
g &lt;- kegg9$gene.info[1,1]  #  SOAT1 
## Take snps mapped on gene "SOAT1" from the information of gene.info and snp.info. 
snps &lt;- which( ( kegg9$snp.info[,2] == kegg9$gene.info[kegg9$gene.info[,1] == g, 2] ) &amp;
                 (kegg9$snp.info[,3] &gt; kegg9$gene.info[kegg9$gene.info[,1] == g, 3] ) &amp;
                 (kegg9$snp.info[,3] &lt; kegg9$gene.info[kegg9$gene.info[,1] == g, 4] )  )
## Take subsets
newP &lt;- kegg9$nP[snps] ;
ldsub &lt;- kegg9$ldmatrix[snps, snps];

## Get p-value for gene SOAT1. Read vignette for details.
out &lt;- aSPUs(newP, corSNP=ldsub , pow=c(1:8, Inf), n.perm=100, Ps=TRUE)

out$Ts
# This is a vector of Test Statistics for SPUM and aSPUM tests.
# SPUs1 to SPUsInf corresponds with the option pow=c(1:8, Inf)
# They are SPUs test statistics.
# The last element aSPUs is minimum of them, aSPUs statistic.

out$pvs
# This is a vector of p-values for SPUs and aSPUs tests.
# SPUs1 to SPUsInf corresponds with the option pow=c(1:8, Inf)
# They are p-values for corresponding SPUs tests.
# The last element is p-value of aSPUs test.

</code></pre>


</div>