<div class="container">

<table style="width: 100%;"><tr>
<td>MTaSPUsSetPath</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Multitraits - Pathway based Sum of Powered Score tests (MTSPUsSetPath) and adaptive MTSPUsSetpath (MTaSPUsSetPath) test for multi trait - pathway association with GWAS summary statistics.</h2>

<h3>Description</h3>

<p>It gives p-values of the MTSPUsSetPath tests and MTaSPUsSetPath test with GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MTaSPUsSetPath(
  Zs,
  corPhe,
  corSNP,
  pow1 = c(1, 2, 4, 8),
  pow2 = c(1, 2, 4, 8),
  pow3 = c(1, 2, 4, 8),
  snp.info,
  gene.info,
  n.perm = 1000,
  Ps = FALSE,
  prune = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Zs</code></td>
<td>
<p>Z-score matrix. row represent SNPs and column represent traits. It could be P-values if the Ps option is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corPhe</code></td>
<td>
<p>Correlation matirx of phenotypes to be tested; Estimated from Z-scores.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corSNP</code></td>
<td>
<p>Correlation matirx of the SNPs to be tested; estimated from a
reference panel (based on the same set of the reference alleles as
used in calculating Z-scores).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow1</code></td>
<td>
<p>SNP specific power(gamma values) used in MTSPUsSetPath test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow2</code></td>
<td>
<p>GENE specific power(gamma values) used in MTSPUsSetPath test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow3</code></td>
<td>
<p>Trait specific power(gamma values) used in MTSPUsSetPath test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>number of permutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Ps</code></td>
<td>
<p>TRUE if input is p-value, FALSE if input is Z-scores. The default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prune</code></td>
<td>
<p>if it is TRUE, do pruing before the test using pruneSNP function.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>P-values for MTSPUsSetpath tests and MTaSPUsSetpPath test.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2017)
Gene- and pathway-based association tests for multiple
traits with GWAS summary statistics, Bioinformatics, 33(1), 64-71
</p>


<h3>See Also</h3>

<p><code>MTaSPUsSet</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
Zs &lt;- cbind ( c( 0.3, 0.2, 0.1,0.5,1.2),
          c(-.1, .3,-.1,.1,1.2) )

varSNP = cbind( c( 1, .1,0, 0, .11),
               c(.1,1, 0, 0, 0),
               c(0,0, 1, 0, 0),
               c(0,0, 0, 1, 0),
               c(.11,0,0,0,1) )

varPhe = cbind( c( 1, -.1),
               c(-.1,1) )

gene.info = data.frame( Gnm = c( "G1", "G2"), chr = c(1,3),
                    loc1 = c(0, 0), loc2 = c(10,10) )
snp.info = data.frame( rsid = c("rs1", "rs2", "rs3", "rs4", "rs5"),
                chr = c(1,1,3,3,3), loc = c(1,2,1,2,3) )
 out &lt;- MTaSPUsSetPath(Zs, corPhe = varPhe, corSNP=varSNP,
            n.perm = 100, snp.info = snp.info, gene.info = gene.info)
out

</code></pre>


</div>