<div class="container">

<table style="width: 100%;"><tr>
<td>plotPmat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Image plot of P-value matrix.</h2>

<h3>Description</h3>

<p>It gives a P-value image for a given P-value matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotPmat(
  Ps,
  zlim = NULL,
  main = NULL,
  yt = NULL,
  xlab = "SNPs",
  thresh = -log(5e-08, 10),
  trait.names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Ps</code></td>
<td>
<p>P-value matrix. Row represent traits, column represent SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zlim</code></td>
<td>
<p>-log 10 transformed p value range.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>main title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yt</code></td>
<td>
<p>it controls the location of trait names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>lable for x axis. Default is "SNPs".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thresh</code></td>
<td>
<p>Default is set to widely used genome wise threshold -log(5e-8,10).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait.names</code></td>
<td>
<p>A vector of trait names.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Image of P-values matrix.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wei Pan (2017)
Gene- and pathway-based association tests for multiple traits with GWAS summary statistics, Bioinformatics. 33(1), 64-71
</p>


<h3>See Also</h3>

<p><code>aSPUs</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Say we have 3 traits and their p-values at 5 SNPs. 
Ps &lt;- rbind( c(0.001, 0.4, 0.5, 0.00000001, .1),
           c(0.03, 0.3, 0.3, 0.00001, .2),
           c(0.01, 0.2, 0.4, 0.001, .0001) )

## We can visualize it using plotPmat function.
plotPmat(Ps)


</code></pre>


</div>