<div class="container">

<table style="width: 100%;"><tr>
<td>GatesSimes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GATES-Simes test for single trait - pathway association.</h2>

<h3>Description</h3>

<p>Get the p-value of GATES-Simes. It uses an extended Simes procedure to combine GATES p-values across multiple genes in a pathway.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GatesSimes(pvec, ldmatrix, snp.info, gene.info)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pvec</code></td>
<td>
<p>p-values for each SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ldmatrix</code></td>
<td>
<p>numeric. A correlation matrix of SNPs with dimensions matching the length of pvec (the number of SNPs).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.info</code></td>
<td>
<p>SNP information matrix, the 1st column is SNP id, 2nd column is chromosome #, 3rd column indicates SNP location.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.info</code></td>
<td>
<p>GENE information matrix, The 1st column is GENE id, 2nd column is chromosome #, 3rd and 4th column indicate start and end positions of the gene.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A p-value.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak and Wei Pan
</p>


<h3>References</h3>

<p>Hongsheng Gui, Miaoxin Li, Pak C Sham and Stacey S Cherny (2011)
Comparisons of seven algorithms for pathway analysis using the WTCCC Crohn's Disease
BMC Research Notes, 4:386
</p>


<h3>See Also</h3>

<p><code>Hyst</code>  <code>GATES2</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#simula &lt;- simPathAR1Snp(nGenes=20, nGenes1=1, nSNPlim=c(1, 20), nSNP0=1:3,
#                           LOR=.2, rholim=c(0,0),
#                           n=30, MAFlim=c(0.05, 0.4), p0=0.05)
#logitp &lt;- getlogitp(simula$Y, simula$X)

## get correlation of SNPs using controls
#ldmat &lt;- cor(simula$X[ simula$Y == 0, ])
#out &lt;- GatesSimes(pvec = logitp, ldmatrix = ldmat, snp.info = simula$snp.info,
#                  gene.info = simula$gene.info)
#out


</code></pre>


</div>