<div class="container">

<table style="width: 100%;"><tr>
<td>aSPUd</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Adaptive Sum of powered score (SPU) tests (SPU and aSPU) based on asymptotic distribution.</h2>

<h3>Description</h3>

<p>It gives the p-values of the SPU(1), SPU(2) and minP tests and aSPUd test based on the asymptotic distribution.
</p>


<h3>Usage</h3>

<pre><code class="language-R">aSPUd(Y, X, cov = NULL, model = c("gaussian", "binomial"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>phenotype data. It can be disease lables; =0 for controls, =1 for cases.
or It can be any quantitative traits. Vector with length n (number of observations)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>genotype data; each row for a subject, and each column
for an SNP. The value of each element is the # of the copies
for an allele. Matrix with dimension n by k (n : number of observation, k : number of genotype data)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cov</code></td>
<td>
<p>covariates. Matrix with dimension n by p (n :number of observation, p : number of covariates)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>Use "gaussian" for quantitative trait (Default)
, and Use "binomial" for binary trait.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>p-values for SPU(1), SPU(2), minP tests and aSPU test.
</p>


<h3>References</h3>

<p>Gongjun Xu, Lifeng Lin, Peng Wei and Wei Pan (2016) An adaptive two-sample test for high-dimensional means, Biometrika (2016) 103 (3): 609-624.
</p>


<h3>See Also</h3>

<p><code>aSPU</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(exdat)
out &lt;- aSPUd(exdat$Y, exdat$X, cov = NULL, model = "binomial")
out

</code></pre>


</div>