<div class="container">

<table style="width: 100%;"><tr>
<td>H.inverse</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generates the inverse of the hybrid H matrix</h2>

<h3>Description</h3>

<p>The single-step GBLUP approach combines the information from the pedigree relationship matrix
<code class="reqn">\boldsymbol{A}</code> and the genomic relationship matrix <code class="reqn">\boldsymbol{G}</code> in one
hybrid relationship matrix called <code class="reqn">\boldsymbol{H}</code>.
This function will calculate directly the inverse of this matrix <code class="reqn">\boldsymbol{H}</code>.
The user should provide the matrices <code class="reqn">\boldsymbol{A}</code> or
its inverse (only one of these is required) and the
inverse of the matrix <code class="reqn">\boldsymbol{G}</code> (<code class="reqn">\boldsymbol{G_{inv}}</code>) in its full form. Individual names should
be assigned to  <code>rownames</code> and <code>colnames</code>, and individuals from
<code class="reqn">\boldsymbol{G_{inv}}</code> are verified to be all a subset within individuals from
<code class="reqn">\boldsymbol{A}</code> (or <code class="reqn">\boldsymbol{A_{inv}}</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">H.inverse(
  A = NULL,
  Ainv = NULL,
  Ginv = NULL,
  lambda = NULL,
  tau = 1,
  omega = 1,
  sparseform = FALSE,
  keep.order = TRUE,
  digits = 8,
  inverse = TRUE,
  message = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>A</code></td>
<td>
<p>Input of the pedigree relationship matrix <code class="reqn">\boldsymbol{A}</code>
in full form (<code class="reqn">na \times na</code>) (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Ainv</code></td>
<td>
<p>Input of the inverse of the pedigree relationship matrix
<code class="reqn">\boldsymbol{A}^{-1}</code> in full form (<code class="reqn">na \times na</code>) (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Ginv</code></td>
<td>
<p>Input of the inverse of the genomic relationship matrix
<code class="reqn">\boldsymbol{G}^{-1}</code> in full form (<code class="reqn">ng \times ng</code>) (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>
<p>The scaling factor for <code class="reqn">(\boldsymbol{G}^{-1}-\boldsymbol{A}^{-1}_{22})</code> (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau</code></td>
<td>
<p>The scaling factor for <code class="reqn">\boldsymbol{G}^{-1}</code> (default = <code>1</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>omega</code></td>
<td>
<p>The scaling factor for <code class="reqn">\boldsymbol{A}^{-1}_{22}</code> (default = <code>1</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sparseform</code></td>
<td>
<p>If <code>TRUE</code> it generates the requested matrix in sparse form to be used
directly in <span class="pkg">asreml</span> with required attributes (default = <code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.order</code></td>
<td>
<p>If <code>TRUE</code> the original order of the individuals from the
<code class="reqn">\boldsymbol{A}</code> or <code class="reqn">\boldsymbol{A_{inv}}</code> matrix is kept.
Otherwise the non-genotyped individuals are placed first and
then genotyped individuals (default = <code>TRUE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>digits</code></td>
<td>
<p>Set up the number of digits used to round the output matrix (default = <code>8</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inverse</code></td>
<td>
<p>If <code>TRUE</code> it generates the inverse of <code class="reqn">\boldsymbol{H}</code> matrix (default = <code>TRUE</code>)
(to be deprecated).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>message</code></td>
<td>
<p>If <code>TRUE</code> diagnostic messages are printed on screen (default = <code>TRUE</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The generation of the <code class="reqn">\boldsymbol{H^{-1}}</code> matrix contains a few scaling factors
to help with the calculation of this inverse and
to allow further exploration of the combination of the
information from the <code class="reqn">\boldsymbol{A^{-1}}</code> and <code class="reqn">\boldsymbol{G^{-1}}</code>.
We follow the specifications described by Martini <em>et. al</em> (2018),
which is done by specifying the parameters <code class="reqn">\lambda</code>, or the pair
<code class="reqn">\tau</code> and <code class="reqn">\omega</code>.
</p>

<p>The general expression used is:
</p>
<p style="text-align: center;"><code class="reqn">\boldsymbol{H^{-1}}=\boldsymbol{A^{-1}}+\begin{bmatrix}\boldsymbol{0}&amp;\boldsymbol{0}\\\boldsymbol{0}&amp;(\tau\boldsymbol{G^{-1}}-\omega\boldsymbol{{A_{22}^{-1}}})\end{bmatrix}</code>
</p>



<p>and a more common representation of the above expression is found when <code class="reqn">\tau = \omega = \lambda</code>, as shown below:
</p>
<p style="text-align: center;"><code class="reqn">\boldsymbol{H^{-1}}=\boldsymbol{A^{-1}}+\begin{bmatrix}\boldsymbol{0}&amp;\boldsymbol{0}\\\boldsymbol{0}&amp;\lambda(\boldsymbol{G^{-1}}-\boldsymbol{{A_{22}^{-1}}})\end{bmatrix}</code>
</p>


<p>If <code>inverse = FALSE</code> the <code class="reqn">\boldsymbol{H}</code>
matrix is provided instead of its inverse. This option will be deprecated and
it is better to use the function H.matrix.
</p>

<p>The <code class="reqn">\boldsymbol{H}</code> matrix is obtained with the following equations:
</p>
<p style="text-align: center;"><code class="reqn">\boldsymbol{H}=\boldsymbol{A}+\begin{bmatrix}\boldsymbol{A}_{12}\boldsymbol{A}_{22}^{-1}(\boldsymbol{G}-\boldsymbol{A}_{22})\boldsymbol{A}_{22}^{-1}\boldsymbol{A}_{21}&amp;\boldsymbol{A}_{12}\boldsymbol{A}_{22}^{-1}(\boldsymbol{G}-\boldsymbol{A}_{22})\\(\boldsymbol{G}-\boldsymbol{A}_{22})\boldsymbol{A}_{22}^{-1}\boldsymbol{A}_{21}&amp;(\boldsymbol{G}-\boldsymbol{A}_{22})\end{bmatrix}</code>
</p>




<h3>Value</h3>

<p>The inverse of the hybrid matrix <code class="reqn">\boldsymbol{H}</code> matrix, in full or sparse form with
required attributes to be used in <span class="pkg">asreml</span>.
</p>


<h3>References</h3>

<p>Christensen, O.F., Lund, M.S. 2010. Genomic prediction matrix when some animals
are not genotyped. Gen. Sel. Evol. 42(2):1–8.
</p>
<p>Christensen, O., Madsen, P., Nielsen, B., Ostersen, T., and Su, G. 2012. Single-step methods
for genomic evaluation in pigs. Animal 6(10):1565–1571.
</p>
<p>Legarra, A., Aguilar, I., and Misztal, I. 2009. A relationship matrix including full
pedigree and genomic information. J. Dairy Sci. 92:4656-4663.
</p>
<p>Martini, J.W.R., Schrauf, M.F., Garcia-Baccino, C.A., Pimentel, E.C.G., Munilla, S.,
Rogberg-Muñoz, A., Cantet, R.J.C., Reimer, C., Gao, N., Wimmer, V., and Simianer, H. 2018.
The effect of the <code class="reqn">H^{-1}</code> scaling factors <code class="reqn">\tau</code> and <code class="reqn">\omega</code>
on the structure of <code class="reqn">H</code> in the single-step procedure.
Genet. Sel. Evol. 50:1-9.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Get A matrix.
A &lt;- AGHmatrix::Amatrix(data = ped.pine)
A[1:5,1:5]
dim(A)

# Read and filter genotypic data.
M.clean &lt;- qc.filtering(
 M = geno.pine655,
 maf = 0.05,
 marker.callrate = 0.2, ind.callrate = 0.20,
 na.string = "-9",
 plots = FALSE)$M.clean

# Get G matrix.
G &lt;- G.matrix(M.clean, method = "VanRaden", na.string = "-9")$G
G[1:5, 1:5]
dim(G)

# Match G and A.
check &lt;- match.G2A(
 A = A, G = G,
 clean = TRUE, ord = TRUE, mism = TRUE, RMdiff = TRUE)

# Align G matrix with A.
G_align &lt;- G.tuneup(G = check$Gclean, A = check$Aclean, align = TRUE, sparseform = FALSE)$Gb

# Get Ginverse using the G aligned.
Ginv &lt;- G.inverse(G = G_align, sparseform = FALSE)$Ginv
Ginv[1:5, 1:5]
dim(Ginv)

# Obtain Hinv.
Hinv &lt;- H.inverse(A = A, Ginv = Ginv, lambda = 0.90, sparseform = TRUE)
head(Hinv)
attr(Hinv, "INVERSE")


</code></pre>


</div>