<div class="container">

<table style="width: 100%;"><tr>
<td>snp.pca</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Performs a Principal Component Analysis (PCA) based on a molecular matrix M</h2>

<h3>Description</h3>

<p>Generates a PCA and summary statistics from a given molecular matrix
for population structure. Matrix
provided is of full form (<code class="reqn">n \times p</code>), with n individuals and p markers. Individual and
marker names are assigned to <code>rownames</code> and <code>colnames</code>, respectively.
SNP data is coded as 0, 1, 2 (integers or decimal numbers). Missing values are
not accepted and these need to be imputed (see function <code>qc.filtering()</code>
for implementing mean imputation). There is additional output such as plots and
other data frames
to be used on other downstream analyses (such as GWAS).
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp.pca(M = NULL, label = FALSE, ncp = 10, groups = NULL, ellipses = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>
<p>A matrix with SNP data of full form (<code class="reqn">n \times p</code>), with <code class="reqn">n</code>
individuals and <code class="reqn">p</code> markers (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>If <code>TRUE</code> then includes in output individuals names (default = <code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncp</code></td>
<td>
<p>The number of PC dimensions to be shown in the screeplot, and to provide
in the output data frame (default = <code>10</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groups</code></td>
<td>
<p>Specifies a vector of class factor that will be used to define different
colors for individuals in the PCA plot. It must be presented in the same order as the individuals
in the molecular <code class="reqn">\boldsymbol{M}</code> matrix (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ellipses</code></td>
<td>
<p>If <code>TRUE</code>, ellipses will will be drawn around each of the define levels in
<code>groups</code> (default = <code>FALSE</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>It calls function <code>prcomp()</code> to generate the PCA and the
<code>factoextra</code> R package to extract and visualize results.
Methodology uses normalized allele frequencies as proposed by Patterson <em>et al.</em> (2006).
</p>


<h3>Value</h3>

<p>A list with the following four elements:
</p>

<ul>
<li>
<p><code>eigenvalues</code>: a data frame with the eigenvalues and its variances associated with each dimension
including only the first <code>ncp</code> dimensions.
</p>
</li>
<li>
<p><code>pca.scores</code>: a data frame with scores (rotated observations on the new components) including
only the first <code>ncp</code> dimensions.
</p>
</li>
<li>
<p><code>plot.pca</code>: a scatterplot with the first two-dimensions (PC1 and PC2) and their scores.
</p>
</li>
<li>
<p><code>plot.scree</code>: a barchart with the percentage of variances explained by the <code>ncp</code> dimensions.
</p>
</li>
</ul>
<h3>References</h3>

<p>Patterson N., Price A.L., and Reich, D. 2006. Population structure and eigenanalysis.
PLoS Genet 2(12):e190. doi:10.1371/journal.pgen.0020190
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Perform the PCA.
SNP_pca &lt;- snp.pca(M = geno.apple, ncp = 10)
ls(SNP_pca)
SNP_pca$eigenvalues
head(SNP_pca$pca.scores)
SNP_pca$plot.pca
SNP_pca$plot.scree

# PCA plot by family (17 groups).
grp &lt;- as.factor(pheno.apple$Family)
SNP_pca_grp &lt;- snp.pca(M = geno.apple, groups = grp, label = FALSE)
SNP_pca_grp$plot.pca

</code></pre>


</div>