<div class="container">

<table style="width: 100%;"><tr>
<td>snp.pruning</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reduces the number of redundant markers on a molecular matrix M by pruning</h2>

<h3>Description</h3>

<p>For a given molecular dataset <code class="reqn">\boldsymbol{M}</code> (in the format 0, 1 and 2)
it produces a reduced molecular matrix by eliminating "redundant"
markers using pruning techniques. This function finds and drops some of the
SNPs in high linkage disequilibrium (LD).
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp.pruning(
  M = NULL,
  map = NULL,
  marker = NULL,
  chrom = NULL,
  pos = NULL,
  method = c("correlation"),
  criteria = c("callrate", "maf"),
  pruning.thr = 0.95,
  by.chrom = FALSE,
  window.n = 50,
  overlap.n = 5,
  iterations = 10,
  seed = NULL,
  message = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>
<p>A matrix with marker data of full form (<code class="reqn">n \times p</code>), with <code class="reqn">n</code> individuals
and <code class="reqn">p</code> markers. Individual and marker names are assigned to <code>rownames</code> and <code>colnames</code>, respectively.
Data in matrix is coded as 0, 1, 2 (integer or numeric) (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>(Optional) A data frame with the map information with <code class="reqn">p</code> rows.
If <code>NULL</code> a dummy map is generated considering a single chromosome and sequential positions
for markers. A <code>map</code> is mandatory if <code>by.chrom = TRUE</code>, where also option <code>chrom</code>
must also be non-null.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker</code></td>
<td>
<p>A character indicating the name of the column in data frame <code>map</code>
with the identification
of markers. This is mandatory if <code>map</code> is provided (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom</code></td>
<td>
<p>A character indicating the name of the column in data frame <code>map</code> with the identification
of chromosomes. This is mandatory if <code>map</code> is provided (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>A character indicating the name of the column in data frame <code>map</code> with the identification
of marker positions (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>A character indicating the method (or algorithm) to be used as reference for
identifying redundant markers.
The only method currently available is based on correlations (default = <code>"correlation"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>criteria</code></td>
<td>
<p>A character indicating the criteria to choose which marker to drop
from a detected redundant pair.
Options are: <code>"callrate"</code> (the marker with fewer missing values will be kept) and
<code>"maf"</code> (the marker with higher minor allele frequency will be kept) (default = <code>"callrate"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pruning.thr</code></td>
<td>
<p>A threshold value to identify redundant markers with Pearson's correlation larger than the
value provided (default = <code>0.95</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by.chrom</code></td>
<td>
<p>If TRUE the pruning is performed independently by chromosome (default = <code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window.n</code></td>
<td>
<p>A numeric value with number of markers to consider in each
window to perform pruning (default = <code>50</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overlap.n</code></td>
<td>
<p>A numeric value with number of markers to overlap between consecutive windows
(default = <code>5</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iterations</code></td>
<td>
<p>An integer indicating the number of sequential times the pruning procedure
should be executed on remaining markers.
If no markers are dropped in a given iteration/run, the algorithm will stop (default = <code>10</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>An integer to be used as seed for reproducibility. In case the criteria has the
same values for a given pair of markers, one will be dropped at random (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>message</code></td>
<td>
<p>If <code>TRUE</code> diagnostic messages are printed on screen (default = <code>TRUE</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Pruning is recommended as redundancies can affect
the quality of matrices used for downstream analyses.
The algorithm used is based on the Pearson's correlation between markers
as a <em>proxy</em> for LD. In the event of a pairwise correlation higher
than the selected threshold markers will be eliminated as specified by: call rate,
minor allele frequency. In case of tie, one marker will be dropped at random.
</p>
<p>Filtering markers (qc.filtering) is of high relevance before pruning.
Poor quality markers (<em>e.g.</em>, monomorphic markers) may prevent correlations from being
calculated and may affect eliminations.
</p>


<h3>Value</h3>


<ul>
<li>
<p><code>Mpruned</code>: a matrix containing the pruned marker <em>M</em> matrix.
</p>
</li>
<li>
<p><code>map</code>: an data frame containing the pruned map.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R"># Read and filter genotypic data.
M.clean &lt;- qc.filtering(
 M = geno.pine655,
 maf = 0.05,
 marker.callrate = 0.20, ind.callrate = 0.20,
 Fis = 1, heterozygosity = 0.98,
 na.string = "-9",
 plots = FALSE)$M.clean

# Prune correlations &gt; 0.9.
Mpr &lt;- snp.pruning(
 M = M.clean, pruning.thr = 0.90,
 by.chrom = FALSE, window.n = 40, overlap.n = 10)
head(Mpr$map)
Mpr$Mpruned[1:5, 1:5]

</code></pre>


</div>