<div class="container">

<table style="width: 100%;"><tr>
<td>smt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Single-marker test</h2>

<h3>Description</h3>

<p>Conduct the single-marker test in an association study to test for
the association between the genotype at a biallelic marker and a
trait.
</p>


<h3>Usage</h3>

<pre><code class="language-R">smt(
  y,
  g,
  covariates = NULL,
  min.count = 5,
  missing.rate = 0.2,
  y.continuous = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a numeric vector of the observed trait values in which
the <em>i</em>th element is for the <em>i</em>th subject. The elements
could be discrete (<code>0</code> or <code>1</code>) or continuous. The missing value is
represented by <code>NA</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>a numeric vector of the observed genotype values (<code>0</code>, <code>1</code>,
or <code>2</code> denotes the number of risk alleles) in which the <em>i</em>th
element is for the <em>i</em>th subject. The missing value is
represented by <code>NA</code>. <code>g</code> has the same length as <code>y</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covariates</code></td>
<td>
<p>an optional data frame, list or environment
containing the covariates used in the model. The default is <code>NULL</code>,
that is, there are no covariates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.count</code></td>
<td>
<p>a critical value to decide which method is used
to calculate the p-value when the trait is discrete and <code>covariates
= NULL</code>. If the minimum number of the elements given a specific
trait value and a specific genotype value is less than
<code>min.count</code>, the Fisher's exact test is adopted; otherwise, the
Wald test is adopted. The default is <code>5</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing.rate</code></td>
<td>
<p>the highest missing value rate of the genotype
values that this function can tolerate. The default is <code>0.2</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y.continuous</code></td>
<td>
<p>logical. If <code>TRUE</code>, <code>y</code> is continuous;
otherwise, <code>y</code> is discrete. The default is <code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Single-marker analysis is a core in many gene-based or
pathway-based procedures, such as the truncated p-value
combination and the minimal p-value.
</p>


<h3>Value</h3>

<p><code>smt</code> returns a list with class "<code>htest</code>".
</p>
<p>If y is continuous, the list contains the following components:
</p>

<table>
<tr>
<td style="text-align: left;">
<code>statistic</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>p.value</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>alternative</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>method</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>data.name</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string giving the names of the data. </td>
</tr>
<tr>
<td style="text-align: left;">
<code>sample.size</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a vector giving the numbers of the subjects with the genotypes <code>0</code>, <code>1</code>, and <code>2</code> (<code>n0</code>, </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> <code>n1</code>, and <code>n2</code>, respectively).
</td>
</tr>
</table>
<p>If y is discrete, the list contains the following components:
</p>

<table>
<tr>
<td style="text-align: left;">
<code>statistic</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>p.value</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>alternative</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>method</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>data.name</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string giving the names of the data. </td>
</tr>
<tr>
<td style="text-align: left;">
<code>sample.size</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a vector giving </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of subjects with the trait value <code>1</code> and the genotype <code>0</code> (<code>r0</code>), </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of subjects with the trait value <code>1</code> and the genotype <code>1</code> (<code>r1</code>), </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of subjects with the trait value <code>1</code> and the genotype <code>2</code> (<code>r2</code>), </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of subjects with the trait value <code>0</code> and the genotype <code>0</code> (<code>s0</code>), </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of subjects with the trait value <code>0</code> and the genotype <code>1</code> (<code>s1</code>), </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> and the number of subjects with the trait value <code>0</code> and the genotype <code>2</code> (<code>s2</code>).</td>
</tr>
<tr>
<td style="text-align: left;">
<code>bad.obs</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a vector giving the number of missing genotype values with the trait value <code>1</code> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> (<code>r.miss</code>), the number of missing genotype values with the trait value <code>0</code> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> (<code>s.miss</code>), and the total number of the missing genotype values (<code>n.miss</code>).
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>


<h3>Examples</h3>

<pre><code class="language-R">y &lt;- rep(c(0, 1), 25)
g &lt;- sample(c(0, 1, 2), 50, replace = TRUE)
smt(y, g, covariates = NULL, min.count=5,
        missing.rate=0.20, y.continuous = FALSE)
</code></pre>


</div>