<div class="container">

<table style="width: 100%;"><tr>
<td>dr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Distance regression</h2>

<h3>Description</h3>

<p>Conduct the distance regression with or without the adjustment of
the covariates to detect the association between a distance matrix
and some independent variants of interest.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dr(
  simi.mat,
  null.space,
  x.mat,
  permute = TRUE,
  n.MonteCarlo = 1000,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>simi.mat</code></td>
<td>
<p>a similarity matrix among the subjects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>null.space</code></td>
<td>
<p>a numeric vector to show the column numbers of
the null space in <code>x.mat</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x.mat</code></td>
<td>
<p>the covariate matrix which combines the null space
and the matrix of interest.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permute</code></td>
<td>
<p>logical. If <code>TRUE</code>, the Monte Carlo sampling is used
without replacement; otherwise, with replacement. The default is
<code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.MonteCarlo</code></td>
<td>
<p>the number of times for the Monte Carlo
procedure. The default is <code>1000</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>if it is not <code>NULL</code>, set the random number generator
state for random number generation. The default is <code>NULL</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The pseudo <em>F</em> statistic based on the distance regression with
or without the adjustment of the covariates detects the
association between a distance matrix and some independent
variants of interest. A distance matrix can be transformed into a
similarity matrix easily.
</p>


<h3>Value</h3>

<p>A list with class "<code>htest</code>" containing the following components:
</p>

<table>
<tr>
<td style="text-align: left;">
<code>statistic</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>p.value</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>alternative</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>method</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>data.name</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string giving the names of the data.
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>
<p>Q Li, S Wacholder, DJ Hunter, RN Hoover, S Chanock, G
Thomas, and K Yu. Genetic Background Comparison Using
Distance-Based Regression, with Applications in Population
Stratification Evaluation and Adjustment. <em>Genetic
Epidemiology</em>. 2009; 33(5): 432-441.
</p>
<p>J Wessel and NJ Schork. Generalized Genomic
Distance-Based Regression Methodology for Multilocus Association
Analysis. <em>American Journal of Human Genetics</em>. 2006; 79(5):
792-806.
</p>
<p>MA Zapala and NJ Schork. Multivariate Regression
Analysis of Distance Matrices for Testing Associations Between
Gene Expression Patterns and Related Variables. <em>Proceedings
of the National Academy of Sciences of the United States of
America</em>. 2006; 103(51): 19430-19435.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(drS.eg)
null.space &lt;- 1
x.mat &lt;- matrix(c(rep(1, 600), rep(0, 200)), ncol=2)
dr(drS.eg, null.space, x.mat, permute = TRUE, n.MonteCarlo = 50, seed = NULL)
</code></pre>


</div>