<div class="container">

<table style="width: 100%;"><tr>
<td>AssocAFC</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Allele Frequency Comparison
</h2>

<h3>Description</h3>

<p>When doing association analysis one does not always have the 
genotypes for the control population.  In such cases it may be 
necessary to fall back on frequency based tests using well known sources 
for the frequencies in the control population, for instance, from the 
1000 Genomes Project.  The Allele Frequency Comparison ('AssocAFC') package 
performs multiple rare variant association analyses in both population and 
family-based GWAS (Genome-Wide Association Study) designs. It includes 
three score tests that are based on the difference of the sum of allele 
frequencies between cases and controls.  Two of these tests, Wcorrected() 
and Wqls(), are collapsing-based tests and suffer from having protective 
and risk variants. The third test, afcSKAT(), is a score test that 
overcomes the mix of SNP (Single-Nucleotide Polymorphism) effect 
directions. For more details see 
Saad M and Wijsman EM (2017) &lt;doi:10.1093/bib/bbx107&gt;.
</p>


<h3>Author(s)</h3>

<p>Khalid B. Kunji [aut, cre], Mohamad Saad [aut]
</p>
<p>Maintainer: Khalid B. Kunji &lt;kkunji@hbku.edu.qa&gt;
</p>


<h3>References</h3>

<p>Mohamad Saad, Ellen M. Wijsman; Association score testing for rare variants and binary traits in family data with shared controls, Briefings in Bioinformatics, , bbx107, https://doi.org/10.1093/bib/bbx107
</p>


<h3>See Also</h3>

<p><code>Wcorrected</code>
<code>Wqls</code>
<code>afcSKAT</code>
</p>


</div>