<div class="container">

<table style="width: 100%;"><tr>
<td>max3</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Maximum Test: maximum value of the three Cochran-Armitage
trend tests under the recessive, additive, and dominant models</h2>

<h3>Description</h3>

<p>Conduct MAX3 (the maximal value of the three Cochran-Armitage
trend tests derived for the recessive, additive, and dominant
models) based on the trend tests without the adjustment of the
covariates or based on the Wald tests with the adjustment of the
covariates to test for the association between a single-nucleotide
polymorphism and the binary phenotype.
</p>


<h3>Usage</h3>

<pre><code class="language-R">max3(
  y,
  g,
  covariates = NULL,
  Score.test = TRUE,
  Wald.test = FALSE,
  rhombus.formula = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a numeric vector of the observed trait values in which
the <em>i</em>th element is for the <em>i</em>th subject. The elements
should be <code>0</code> or <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>a numeric vector of the observed genotype values (<code>0</code>, <code>1</code>,
or <code>2</code> denotes the number of risk alleles) in which the <em>i</em>th
element is for the <em>i</em>th subject. The missing value is
represented by <code>NA</code>. <code>g</code> has the same length as <code>y</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covariates</code></td>
<td>
<p>a numeric matrix for the covariates used in the
model. Each column is for one covariate. The default is <code>NULL</code>, that
is, there are no covariates to be adjusted for.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Score.test</code></td>
<td>
<p>logical. If <code>TRUE</code>, the score tests are used. One
of <code>Score.test</code> and <code>Wald.test</code> should be <code>FALSE</code>,
and the other should be <code>TRUE</code>. The default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Wald.test</code></td>
<td>
<p>logical. If <code>TRUE</code>, the Wald tests are used. One of
<code>Score.test</code> and <code>Wald.test</code> should be <code>FALSE</code>,
and the other should be <code>TRUE</code>. The default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rhombus.formula</code></td>
<td>
<p>logical. If <code>TRUE</code>, the p-value for the MAX3
is approximated by the rhombus formula. IF <code>FALSE</code>, the 2-fold
integration is used to calculate the p-value. The default is
<code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In an association study, the genetic inheritance models
(recessive, additive, or dominant) are unknown beforehand. This
function can account for the uncertainty of the underlying genetic
models and test for the association between a single-nucleotide
polymorphism and a binary phenotype with or without correcting for
the covariates.
</p>


<h3>Value</h3>

<p>A list with class "<code>htest</code>" containing the following components:
</p>

<table>
<tr>
<td style="text-align: left;">
<code>statistic</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>p.value</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>alternative</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>method</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>data.name</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string giving the names of the data.
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>
<p>Q Li, G Zheng, Z Li, and K Yu. Efficient Approximation
of P Value of the Maximum of Correlated Tests, with Applications
to Genome-Wide Association Studies. <em>Annals of Human
Genetics</em>. 2008; 72(3): 397-406.
</p>


<h3>Examples</h3>

<pre><code class="language-R">y &lt;- rep(c(0, 1), 5)
g &lt;- sample(c(0, 1, 2), 10, replace = TRUE)
max3(y, g, covariates = NULL, Score.test = TRUE, Wald.test = FALSE,
       rhombus.formula = FALSE)
max3(y, g, covariates = matrix(sample(c(0,1), 20, replace = TRUE), ncol=2),
       Score.test = TRUE, Wald.test = FALSE, rhombus.formula = FALSE)
</code></pre>


</div>