<div class="container">

<table style="width: 100%;"><tr>
<td>npt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Nonparametric trend test based on the nonparametric
risk under a given genetic model</h2>

<h3>Description</h3>

<p>Test for the association between a genetic variant and a
non-normal distributed quantitative trait based on the
nonparametric risk under a specific genetic model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">npt(y, g, varphi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a numeric vector of the observed quantitative trait
values in which the <em>i</em>th element corresponds to the trait
value of the <em>i</em>th subject.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>a numeric vector of the observed genotype values (<code>0</code>, <code>1</code>,
or <code>2</code> denotes the number of risk alleles) in which the <em>i</em>th
element is the genotype value of the <em>i</em>th subject for a
biallelic SNP. <code>g</code> has the same length as <code>y</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>varphi</code></td>
<td>
<p>a numeric value which represents the genetic
model. It should be <code>0</code>, <code>0.5</code>, or <code>1</code>, which indicates that the
calculation is performed under the recessive, additive, or
dominant model, respectively.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For a non-normal distributed quantitative trait, three genetic
models (recessive, additive and dominant) used commonly are
defined in terms of the nonparametric risk (NR). The recessive,
additive, and dominant models can be classified based on the
nonparametric risks. More specifically, the recessive, additive,
and dominant models refer to NR20 &gt; NR10 = 1/2, NR12 = NR10&gt;1/2, and
NR10 = NR20 &gt; 1/2, respectively, where NR10 and NR20 are the
nonparametric risks of the groups with the genotypes 1 and 2
relative to the group with the genotype 0, respectively, and NR12
is the nonparametric risk of the group with the genotype 2
relative to the group with the genotype 1.
</p>
<p><code>varphi</code> can be <code>0</code>, <code>0.5</code>, or <code>1</code> for the recessive, additive, or
dominant model, respectively. When <code>varphi</code> is <code>0</code>, the test is
constructed under the recessive model by pooling together the
subjects with the genotypes 0 and 1. Similarly, when <code>varphi</code>
is <code>1</code>, the test is constructed under the dominant model by pooling
together the subjects with the genotypes 1 and 2. When
<code>varphi</code> is <code>0.5</code>, the test is based on the weighted sum of
NR10 and NR12.
</p>


<h3>Value</h3>

<p>A list with class "<code>htest</code>" containing the following components:
</p>

<table>
<tr>
<td style="text-align: left;">
<code>statistic</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the observed value of the test statistic.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>p.value</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the p-value for the test.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>alternative</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string describing the alternative hypothesis.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>method</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string indicating the type of test performed.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>data.name</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> a character string giving the names of the data.
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>


<h3>Examples</h3>

<pre><code class="language-R">g &lt;- rbinom(1500, 2, 0.3)
y &lt;- 0.5 + 0.25 * g + rgev(1500, 0, 0, 5)
npt(y, g, 0.5)
</code></pre>


</div>