<div class="container">

<table style="width: 100%;"><tr>
<td>eigenstrat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>EIGENSTRAT for correcting for population stratification</h2>

<h3>Description</h3>

<p>Find the eigenvectors of the similarity matrix among the subjects
used for correcting for population stratification in the
population-based genetic association studies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">eigenstrat(
  genoFile,
  outFile.Robj = "out.list",
  outFile.txt = "out.txt",
  rm.marker.index = NULL,
  rm.subject.index = NULL,
  miss.val = 9,
  num.splits = 10,
  topK = NULL,
  signt.eigen.level = 0.01,
  signal.outlier = FALSE,
  iter.outlier = 5,
  sigma.thresh = 6
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genoFile</code></td>
<td>
<p>a txt file containing the genotypes (0, 1, 2, or
9). The element of the file in Row <em>i</em> and Column <em>j</em>
represents the genotype at the <em>i</em>th marker of the <em>j</em>th
subject. 0, 1, and 2 denote the number of risk alleles, and 9
(default) is for the missing genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outFile.Robj</code></td>
<td>
<p>the name of an R object for saving the list of
the results which is the same as the return value of this
function. The default is "<code>out.list</code>".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outFile.txt</code></td>
<td>
<p>a txt file for saving the eigenvectors
corresponding to the top significant eigenvalues.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rm.marker.index</code></td>
<td>
<p>a numeric vector for the indices of the
removed markers. The default is <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rm.subject.index</code></td>
<td>
<p>a numeric vector for the indices of the
removed subjects. The default is <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miss.val</code></td>
<td>
<p>the number representing the missing data in the
input data. The default is <code>9</code>. The element 9 for the missing data
in the <code>genoFile</code> should be changed according to the value of
<code>miss.val</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.splits</code></td>
<td>
<p>the number of groups into which the markers are
split. The default is <code>10</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>topK</code></td>
<td>
<p>the number of eigenvectors to return. If <code>NULL</code>, it is
calculated by the Tracy-Widom test. The default is <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>signt.eigen.level</code></td>
<td>
<p>a numeric value which is the significance
level of the Tracy-Widom test. It should be <code>0.05</code>, <code>0.01</code>, <code>0.005</code>, or
<code>0.001</code>. The default is <code>0.01</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>signal.outlier</code></td>
<td>
<p>logical. If <code>TRUE</code>, delete the outliers of the
subjects; otherwise, do not search for the outliers. The default
is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iter.outlier</code></td>
<td>
<p>a numeric value that is the iteration time for
finding the outliers of the subjects. The default is <code>5</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma.thresh</code></td>
<td>
<p>a numeric value that is the lower limit for
eliminating the outliers. The default is <code>6</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Suppose that a total of <em>n</em> cases and controls are randomly
enrolled in the source population and a panel of <em>m</em>
single-nucleotide polymorphisms are genotyped. The genotype at a
marker locus is coded as 0, 1, or 2, with the value corresponding
to the copy number of risk alleles. All the genotypes are given in
the form of a <em>m*n</em> matrix, in which the element in the
<em>i</em>th row and the <em>j</em>th column represents the genotype
of the <em>j</em>th subject at the <em>i</em>th marker. This function
calculates the top eigenvectors or the eigenvectors with
significant eigenvalues of the similarity matrix among the
subjects to infer the potential population structure. See also
tw.
</p>


<h3>Value</h3>

<p><code>eigenstrat</code> returns a list, which contains the following components:
</p>

<table>
<tr>
<td style="text-align: left;">
<code>num.markers</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of markers excluding the removed markers.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>num.subjects</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of subjects excluding the outliers.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>rm.marker.index</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the indices of the removed markers.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>rm.subject.index</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the indices of the removed subjects.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>TW.level</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the significance level of the Tracy-Widom test.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>signal.outlier</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> dealing with the outliers in the subjects or not.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>iter.outlier</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the iteration time for finding the outliers.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>sigma.thresh</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the lower limit for eliminating the outliers.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>num.outliers</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of outliers.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>outliers.index</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the indices of the outliers.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>num.used.subjects</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of the used subjects.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>used.subjects.index</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the indices of the used subjects.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>similarity.matrix</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the similarity matrix among the subjects.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>eigenvalues</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the eigenvalues of the similarity matrix.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>eigenvectors</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the eigenvectors corresponding to the eigenvalues.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>topK</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the number of significant eigenvalues.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>TW.stat</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the observed values of the Tracy-Widom statistics.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>topK.eigenvalues</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the top eigenvalues.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>topK.eigenvectors</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the eigenvectors corresponding to the top eigenvalues.</td>
</tr>
<tr>
<td style="text-align: left;">
<code>runtime</code> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> the running time of this function.
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li.
</p>


<h3>References</h3>

<p>Lin Wang, Wei Zhang, and Qizhai Li. AssocTests: An R Package 
for Genetic Association Studies. <em>Journal of Statistical Software</em>. 
2020; 94(5): 1-26.
</p>
<p>AL Price, NJ Patterson, RM Plenge, ME Weinblatt, NA
Shadick, and D Reich. Principal Components Analysis Corrects for
Stratification in Genome-Wide Association Studies. <em>Nature
Genetics</em>. 2006; 38(8): 904-909.
</p>
<p>N Patterson, AL Price, and D Reich. Population
Structure and Eigenanalysis. <em>PloS Genetics</em>. 2006; 2(12):
2074-2093.
</p>
<p>CA Tracy and H Widom. Level-Spacing Distributions and
the Airy Kernel. <em>Communications in Mathematical
Physics</em>. 1994; 159(1): 151-174.
</p>


<h3>Examples</h3>

<pre><code class="language-R">eigenstratG.eg &lt;- matrix(rbinom(3000, 2, 0.5), ncol = 30)
write.table(eigenstratG.eg, file = "eigenstratG.eg.txt", quote = FALSE,
            sep = "", row.names = FALSE, col.names = FALSE)
eigenstrat(genoFile = "eigenstratG.eg.txt", outFile.Robj = "eigenstrat.result.list",
             outFile.txt = "eigenstrat.result.txt", rm.marker.index = NULL,
             rm.subject.index = NULL, miss.val = 9, num.splits = 10,
             topK = NULL, signt.eigen.level = 0.01, signal.outlier = FALSE,
             iter.outlier = 5, sigma.thresh = 6)
file.remove("eigenstratG.eg.txt", "eigenstrat.result.list", "eigenstrat.result.txt")
</code></pre>


</div>