<div class="container">

<table style="width: 100%;"><tr>
<td>compute.ALD</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute ALD.</h2>

<h3>Description</h3>

<p>A function to compute asymmetric Linkage Disequilibrium measures (ALD) for polymorphic genetic
data. These measures are identical to the correlation measure (r) for bi-allelic data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">compute.ALD(dat, tolerance = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p>A data.frame with 5 required variables (having the names listed below):
</p>

<table>
<tr>
<td style="text-align: left;">
     <code>haplo.freq</code> </td>
<td style="text-align: left;"> A numeric vector of haplotype frequencies.</td>
</tr>
<tr>
<td style="text-align: left;">
      <code>locus1</code> </td>
<td style="text-align: left;"> A character vector indentifying the first locus.</td>
</tr>
<tr>
<td style="text-align: left;">
      <code>locus2</code> </td>
<td style="text-align: left;"> A character vector indentifying the second locus.</td>
</tr>
<tr>
<td style="text-align: left;">
      <code>allele1</code> </td>
<td style="text-align: left;"> A character vector indentifying the allele at locus 1.</td>
</tr>
<tr>
<td style="text-align: left;">
      <code>allele2</code> </td>
<td style="text-align: left;"> A character vector indentifying the allele at locus 2.</td>
</tr>
<tr>
<td style="text-align: left;">
      </td>
</tr>
</table>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tolerance</code></td>
<td>
<p>A threshold for the sum of the haplotype frequencies.
If the sum of the haplotype frequencies is greater than 1+tolerance or less
than 1-tolerance an error is returned. The default is 0.01.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The return value is a dataframe with the following components:
</p>

<table>
<tr>
<td style="text-align: left;">
 <code>locus1</code>	</td>
<td style="text-align: left;"> The name of the first locus.</td>
</tr>
<tr>
<td style="text-align: left;">
 <code>locus2</code>	</td>
<td style="text-align: left;"> The name of the second locus.</td>
</tr>
<tr>
<td style="text-align: left;">
 <code>F.1</code>	</td>
<td style="text-align: left;"> Homozygosity (expected under HWP) for locus 1.</td>
</tr>
<tr>
<td style="text-align: left;">
 <code>F.1.2</code>	</td>
<td style="text-align: left;"> Conditional homozygosity* for locus1 given locus2.</td>
</tr>
<tr>
<td style="text-align: left;">
 <code>F.2</code>	</td>
<td style="text-align: left;"> Homozygosity (expected under HWP) for locus 2.</td>
</tr>
<tr>
<td style="text-align: left;">
 <code>F.2.1</code>	</td>
<td style="text-align: left;"> Conditional homozygosity* for locus2 given locus1.</td>
</tr>
<tr>
<td style="text-align: left;">
 <code>ALD.1.2</code> </td>
<td style="text-align: left;"> Asymmetric LD for locus1 given locus2.</td>
</tr>
<tr>
<td style="text-align: left;">
 <code>ALD.2.1</code> </td>
<td style="text-align: left;">  Asymmetric LD for locus2 given locus1.</td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
</tr>
</table>
<p>*Overall weighted haplotype-specific homozygosity for the first locus given the second locus.
</p>


<h3>Details</h3>

<p>A warning message is given if the sum of the haplotype frequencies is greater than 1.01 or less
than 0.99 (regardless of the <code>tolerance</code> setting). The haplotype frequencies that are passed
to the function are normalized within the function to sum to 1.0 by dividing each frequency by
the sum of the passed frequencies.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(asymLD)

# An example using haplotype frequencies from Wilson(2010)
data(hla.freqs)
hla.a_b &lt;- hla.freqs[hla.freqs$locus1=="A" &amp; hla.freqs$locus2=="B",]
compute.ALD(hla.a_b)
hla.freqs$locus &lt;- paste(hla.freqs$locus1, hla.freqs$locus2, sep="-")
compute.ALD(hla.freqs[hla.freqs$locus=="C-B",])
# Note: additonal columns on the input dataframe (e.g., "locus" above) are allowed, but
# ignored by the function.

# An example using genotype data from the haplo.stats package
require(haplo.stats)
data(hla.demo)
geno &lt;- hla.demo[,5:8]  #DPB-DPA
label &lt;- unique(gsub(".a(1|2)", "", colnames(geno)))
label &lt;- paste("HLA*",label,sep="")
keep &lt;- !apply(is.na(geno) | geno==0, 1, any)
em.keep  &lt;- haplo.em(geno=geno[keep,], locus.label=label)
hapfreqs.df &lt;- cbind(em.keep$haplotype, em.keep$hap.prob)
#format dataframe for ALD function
names(hapfreqs.df)[dim(hapfreqs.df)[2]] &lt;- "haplo.freq"
names(hapfreqs.df)[1] &lt;- "allele1"
names(hapfreqs.df)[2] &lt;- "allele2"
hapfreqs.df$locus1 &lt;- label[1]
hapfreqs.df$locus2 &lt;- label[2]
head(hapfreqs.df)
compute.ALD(hapfreqs.df)
# Note that there is substantially less variablity (higher ALD) for HLA*DPA1
# conditional on HLA*DPB1 than for HLA*DPB1 conditional on HLA*DPA1, indicating
# that the overall variation for DPA1 is relatively low given specific DPB1 alleles

# An example using SNP data where results are symmetric and equal to the ordinary
# correlation measure (r)
data(snp.freqs)
snps &lt;- c("rs1548306", "rs6923504", "rs4434496", "rs7766854")
compute.ALD(snp.freqs[snp.freqs$locus1==snps[2] &amp; snp.freqs$locus2==snps[3],])

snp.freqs$locus &lt;- paste(snp.freqs$locus1, snp.freqs$locus2, sep="-")
by(snp.freqs,list(snp.freqs$locus),compute.ALD)

# SNP1 &amp; SNP2 : the r correlation &amp; ALD measures are equivalent due to symmetry for
# bi-allelic SNPs
p.AB &lt;- snp.freqs$haplo.freq[1]
p.Ab &lt;- snp.freqs$haplo.freq[2]
p.aB &lt;- snp.freqs$haplo.freq[3]
p.ab &lt;- snp.freqs$haplo.freq[4]
p.A &lt;- p.AB + p.Ab
p.B &lt;- p.AB + p.aB
r.squared &lt;- (p.AB - p.A*p.B)^2 / (p.A*(1-p.A)*p.B*(1-p.B))
sqrt(r.squared) #the r correlation measure
compute.ALD(snp.freqs[snp.freqs$locus1==snps[1] &amp; snp.freqs$locus2==snps[2],])
</code></pre>


</div>