<div class="container">

<table style="width: 100%;"><tr>
<td>heatmapGO</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>HeatmapGO</h2>

<h3>Description</h3>

<p>If the analysis has been performed on more conditions it is interest to have a look at the difference in the enrichment results between the groups. This can be performed by heatmapGO().
</p>
<p>The function automatically reads all the enrichment results of the chosen database. A heatmap is produced for each database, all the terms are merged together and a filter is applied as follows: only terms with a significant pvalue (i.e. less than padj_threshold) in at least one comparison will be retained and plotted. These plots will be saved in the "Comparison_Heatmap" folder. In order to have readable plots, if many terms are enriched for a database several images will be created (indexed _1, _2, ...).
</p>


<h3>Usage</h3>

<pre><code class="language-R">heatmapGO(
  db,
  outfolder = "./results",
  log2FC_threshold = 0,
  padj_threshold = 0.05,
  min_term_per_row = 2,
  which_list = c("up_genes", "down_genes", "up_down_genes", "not_from_DE")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>Database of choice to plot the heatmap. It has to be one for which the enrichment analysis has been performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfolder</code></td>
<td>
<p>The name to assign to the folder for output saving. (Default = "./results"). NOTE: please add "/" at the end.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2FC_threshold</code></td>
<td>
<p>Threshold value for log2(Fold Change) for considering genes as differentially expressed (Default = 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>padj_threshold</code></td>
<td>
<p>Threshold value for adjusted p-value significance (Defaults to 0.05).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_term_per_row</code></td>
<td>
<p>Minimum of comparisons or enriched lists on which a certain term must be significant (Defaults to 2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>which_list</code></td>
<td>
<p>One of c("up_genes", "down_genes","up_down_genes", "not_from_DE"): select data to plot. Respectively, only up regulated genes (up_genes), only down regulated genes ("down_genes"), enrichment on both up and down regulated genes (up_down_genes) or select "not_from_DE" if the enrichment will be made on a list of genes that does not come from a differential expression analysis.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>No return value. Files will be produced as part of normal execution.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
heatmapGO(
  db = "GO_Biological_Process_2021",
  outfolder = "./results",
  log2FC_threshold = 0,
  padj_threshold = 0.05,
  min_term_per_row = 3,
  which_list = "down_genes"
)

## End(Not run)
</code></pre>


</div>